// auth0/dist/cjs/management/tests/wire/roles/permissions.test.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const MockServerPool_js_1 = require("../../mock-server/MockServerPool.js");
const Client_js_1 = require("../../../Client.js");
const Management = __importStar(require("../../../api/index.js"));
describe("Permissions", () => {
    test("list (13f93d13)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = {
            start: 1.1,
            limit: 1.1,
            total: 1.1,
            permissions: [
                {
                    resource_server_identifier: "resource_server_identifier",
                    permission_name: "permission_name",
                    resource_server_name: "resource_server_name",
                    description: "description",
                },
            ],
        };
        server
            .mockEndpoint()
            .get("/roles/id/permissions")
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const expected = {
            start: 1.1,
            limit: 1.1,
            total: 1.1,
            permissions: [
                {
                    resource_server_identifier: "resource_server_identifier",
                    permission_name: "permission_name",
                    resource_server_name: "resource_server_name",
                    description: "description",
                },
            ],
        };
        const page = yield client.roles.permissions.list("id");
        expect(expected.permissions).toEqual(page.data);
        expect(page.hasNextPage()).toBe(true);
        const nextPage = yield page.getNextPage();
        expect(expected.permissions).toEqual(nextPage.data);
    }));
    test("list (fcf9dbd1)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/roles/id/permissions")
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.roles.permissions.list("id");
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("list (49d52691)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/roles/id/permissions")
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.roles.permissions.list("id");
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("list (2428808d)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/roles/id/permissions")
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.roles.permissions.list("id");
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("list (e55ce3fd)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/roles/id/permissions")
            .respondWith()
            .statusCode(404)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.roles.permissions.list("id");
        })).rejects.toThrow(new Management.NotFoundError({
            key: "value",
        }));
    }));
    test("list (27b44cb5)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/roles/id/permissions")
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.roles.permissions.list("id");
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("add (7044cab6)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            permissions: [
                { resource_server_identifier: "resource_server_identifier", permission_name: "permission_name" },
            ],
        };
        server
            .mockEndpoint()
            .post("/roles/id/permissions")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(200)
            .build();
        const response = yield client.roles.permissions.add("id", {
            permissions: [
                {
                    resource_server_identifier: "resource_server_identifier",
                    permission_name: "permission_name",
                },
            ],
        });
        expect(response).toEqual(undefined);
    }));
    test("add (4cfc2db5)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            permissions: [
                { resource_server_identifier: "resource_server_identifier", permission_name: "permission_name" },
                { resource_server_identifier: "resource_server_identifier", permission_name: "permission_name" },
            ],
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/roles/id/permissions")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.roles.permissions.add("id", {
                permissions: [
                    {
                        resource_server_identifier: "resource_server_identifier",
                        permission_name: "permission_name",
                    },
                    {
                        resource_server_identifier: "resource_server_identifier",
                        permission_name: "permission_name",
                    },
                ],
            });
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("add (dd1ed2a5)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            permissions: [
                { resource_server_identifier: "resource_server_identifier", permission_name: "permission_name" },
                { resource_server_identifier: "resource_server_identifier", permission_name: "permission_name" },
            ],
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/roles/id/permissions")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.roles.permissions.add("id", {
                permissions: [
                    {
                        resource_server_identifier: "resource_server_identifier",
                        permission_name: "permission_name",
                    },
                    {
                        resource_server_identifier: "resource_server_identifier",
                        permission_name: "permission_name",
                    },
                ],
            });
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("add (9d65d271)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            permissions: [
                { resource_server_identifier: "resource_server_identifier", permission_name: "permission_name" },
                { resource_server_identifier: "resource_server_identifier", permission_name: "permission_name" },
            ],
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/roles/id/permissions")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.roles.permissions.add("id", {
                permissions: [
                    {
                        resource_server_identifier: "resource_server_identifier",
                        permission_name: "permission_name",
                    },
                    {
                        resource_server_identifier: "resource_server_identifier",
                        permission_name: "permission_name",
                    },
                ],
            });
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("add (fa8483c9)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            permissions: [
                { resource_server_identifier: "resource_server_identifier", permission_name: "permission_name" },
                { resource_server_identifier: "resource_server_identifier", permission_name: "permission_name" },
            ],
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/roles/id/permissions")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.roles.permissions.add("id", {
                permissions: [
                    {
                        resource_server_identifier: "resource_server_identifier",
                        permission_name: "permission_name",
                    },
                    {
                        resource_server_identifier: "resource_server_identifier",
                        permission_name: "permission_name",
                    },
                ],
            });
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("delete (7044cab6)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            permissions: [
                { resource_server_identifier: "resource_server_identifier", permission_name: "permission_name" },
            ],
        };
        server
            .mockEndpoint()
            .delete("/roles/id/permissions")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(200)
            .build();
        const response = yield client.roles.permissions.delete("id", {
            permissions: [
                {
                    resource_server_identifier: "resource_server_identifier",
                    permission_name: "permission_name",
                },
            ],
        });
        expect(response).toEqual(undefined);
    }));
    test("delete (4cfc2db5)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            permissions: [
                { resource_server_identifier: "resource_server_identifier", permission_name: "permission_name" },
                { resource_server_identifier: "resource_server_identifier", permission_name: "permission_name" },
            ],
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/roles/id/permissions")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.roles.permissions.delete("id", {
                permissions: [
                    {
                        resource_server_identifier: "resource_server_identifier",
                        permission_name: "permission_name",
                    },
                    {
                        resource_server_identifier: "resource_server_identifier",
                        permission_name: "permission_name",
                    },
                ],
            });
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("delete (dd1ed2a5)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            permissions: [
                { resource_server_identifier: "resource_server_identifier", permission_name: "permission_name" },
                { resource_server_identifier: "resource_server_identifier", permission_name: "permission_name" },
            ],
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/roles/id/permissions")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.roles.permissions.delete("id", {
                permissions: [
                    {
                        resource_server_identifier: "resource_server_identifier",
                        permission_name: "permission_name",
                    },
                    {
                        resource_server_identifier: "resource_server_identifier",
                        permission_name: "permission_name",
                    },
                ],
            });
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("delete (9d65d271)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            permissions: [
                { resource_server_identifier: "resource_server_identifier", permission_name: "permission_name" },
                { resource_server_identifier: "resource_server_identifier", permission_name: "permission_name" },
            ],
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/roles/id/permissions")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.roles.permissions.delete("id", {
                permissions: [
                    {
                        resource_server_identifier: "resource_server_identifier",
                        permission_name: "permission_name",
                    },
                    {
                        resource_server_identifier: "resource_server_identifier",
                        permission_name: "permission_name",
                    },
                ],
            });
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("delete (fa8483c9)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            permissions: [
                { resource_server_identifier: "resource_server_identifier", permission_name: "permission_name" },
                { resource_server_identifier: "resource_server_identifier", permission_name: "permission_name" },
            ],
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/roles/id/permissions")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.roles.permissions.delete("id", {
                permissions: [
                    {
                        resource_server_identifier: "resource_server_identifier",
                        permission_name: "permission_name",
                    },
                    {
                        resource_server_identifier: "resource_server_identifier",
                        permission_name: "permission_name",
                    },
                ],
            });
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
});
