// auth0/dist/cjs/management/tests/wire/keys/customSigning.test.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const MockServerPool_js_1 = require("../../mock-server/MockServerPool.js");
const Client_js_1 = require("../../../Client.js");
const Management = __importStar(require("../../../api/index.js"));
describe("CustomSigning", () => {
    test("get (4382a1a8)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = {
            keys: [
                {
                    kty: "EC",
                    kid: "kid",
                    use: "sig",
                    key_ops: ["verify"],
                    alg: "RS256",
                    n: "n",
                    e: "e",
                    crv: "P-256",
                    x: "x",
                    y: "y",
                    x5u: "x5u",
                    x5c: ["x5c"],
                    x5t: "x5t",
                    "x5t#S256": "x5t#S256",
                },
            ],
        };
        server
            .mockEndpoint()
            .get("/keys/custom-signing")
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.keys.customSigning.get();
        expect(response).toEqual({
            keys: [
                {
                    kty: "EC",
                    kid: "kid",
                    use: "sig",
                    key_ops: ["verify"],
                    alg: "RS256",
                    n: "n",
                    e: "e",
                    crv: "P-256",
                    x: "x",
                    y: "y",
                    x5u: "x5u",
                    x5c: ["x5c"],
                    x5t: "x5t",
                    "x5t#S256": "x5t#S256",
                },
            ],
        });
    }));
    test("get (1e230aeb)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/keys/custom-signing")
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.keys.customSigning.get();
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("get (af841397)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/keys/custom-signing")
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.keys.customSigning.get();
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("get (c29c5807)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/keys/custom-signing")
            .respondWith()
            .statusCode(404)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.keys.customSigning.get();
        })).rejects.toThrow(new Management.NotFoundError({
            key: "value",
        }));
    }));
    test("get (ee1e23bf)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/keys/custom-signing")
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.keys.customSigning.get();
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("set (861bb62b)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { keys: [{ kty: "EC" }] };
        const rawResponseBody = {
            keys: [
                {
                    kty: "EC",
                    kid: "kid",
                    use: "sig",
                    key_ops: ["verify"],
                    alg: "RS256",
                    n: "n",
                    e: "e",
                    crv: "P-256",
                    x: "x",
                    y: "y",
                    x5u: "x5u",
                    x5c: ["x5c"],
                    x5t: "x5t",
                    "x5t#S256": "x5t#S256",
                },
            ],
        };
        server
            .mockEndpoint()
            .put("/keys/custom-signing")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.keys.customSigning.set({
            keys: [
                {
                    kty: "EC",
                },
            ],
        });
        expect(response).toEqual({
            keys: [
                {
                    kty: "EC",
                    kid: "kid",
                    use: "sig",
                    key_ops: ["verify"],
                    alg: "RS256",
                    n: "n",
                    e: "e",
                    crv: "P-256",
                    x: "x",
                    y: "y",
                    x5u: "x5u",
                    x5c: ["x5c"],
                    x5t: "x5t",
                    "x5t#S256": "x5t#S256",
                },
            ],
        });
    }));
    test("set (70e3d361)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            keys: [
                {
                    kty: "EC",
                    kid: undefined,
                    use: undefined,
                    key_ops: undefined,
                    alg: undefined,
                    n: undefined,
                    e: undefined,
                    crv: undefined,
                    x: undefined,
                    y: undefined,
                    x5u: undefined,
                    x5c: undefined,
                    x5t: undefined,
                    "x5t#S256": undefined,
                },
                {
                    kty: "EC",
                    kid: undefined,
                    use: undefined,
                    key_ops: undefined,
                    alg: undefined,
                    n: undefined,
                    e: undefined,
                    crv: undefined,
                    x: undefined,
                    y: undefined,
                    x5u: undefined,
                    x5c: undefined,
                    x5t: undefined,
                    "x5t#S256": undefined,
                },
            ],
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .put("/keys/custom-signing")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.keys.customSigning.set({
                keys: [
                    {
                        kty: "EC",
                        kid: undefined,
                        use: undefined,
                        key_ops: undefined,
                        alg: undefined,
                        n: undefined,
                        e: undefined,
                        crv: undefined,
                        x: undefined,
                        y: undefined,
                        x5u: undefined,
                        x5c: undefined,
                        x5t: undefined,
                        "x5t#S256": undefined,
                    },
                    {
                        kty: "EC",
                        kid: undefined,
                        use: undefined,
                        key_ops: undefined,
                        alg: undefined,
                        n: undefined,
                        e: undefined,
                        crv: undefined,
                        x: undefined,
                        y: undefined,
                        x5u: undefined,
                        x5c: undefined,
                        x5t: undefined,
                        "x5t#S256": undefined,
                    },
                ],
            });
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("set (5a1d1961)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            keys: [
                {
                    kty: "EC",
                    kid: undefined,
                    use: undefined,
                    key_ops: undefined,
                    alg: undefined,
                    n: undefined,
                    e: undefined,
                    crv: undefined,
                    x: undefined,
                    y: undefined,
                    x5u: undefined,
                    x5c: undefined,
                    x5t: undefined,
                    "x5t#S256": undefined,
                },
                {
                    kty: "EC",
                    kid: undefined,
                    use: undefined,
                    key_ops: undefined,
                    alg: undefined,
                    n: undefined,
                    e: undefined,
                    crv: undefined,
                    x: undefined,
                    y: undefined,
                    x5u: undefined,
                    x5c: undefined,
                    x5t: undefined,
                    "x5t#S256": undefined,
                },
            ],
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .put("/keys/custom-signing")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.keys.customSigning.set({
                keys: [
                    {
                        kty: "EC",
                        kid: undefined,
                        use: undefined,
                        key_ops: undefined,
                        alg: undefined,
                        n: undefined,
                        e: undefined,
                        crv: undefined,
                        x: undefined,
                        y: undefined,
                        x5u: undefined,
                        x5c: undefined,
                        x5t: undefined,
                        "x5t#S256": undefined,
                    },
                    {
                        kty: "EC",
                        kid: undefined,
                        use: undefined,
                        key_ops: undefined,
                        alg: undefined,
                        n: undefined,
                        e: undefined,
                        crv: undefined,
                        x: undefined,
                        y: undefined,
                        x5u: undefined,
                        x5c: undefined,
                        x5t: undefined,
                        "x5t#S256": undefined,
                    },
                ],
            });
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("set (5ef29f1d)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            keys: [
                {
                    kty: "EC",
                    kid: undefined,
                    use: undefined,
                    key_ops: undefined,
                    alg: undefined,
                    n: undefined,
                    e: undefined,
                    crv: undefined,
                    x: undefined,
                    y: undefined,
                    x5u: undefined,
                    x5c: undefined,
                    x5t: undefined,
                    "x5t#S256": undefined,
                },
                {
                    kty: "EC",
                    kid: undefined,
                    use: undefined,
                    key_ops: undefined,
                    alg: undefined,
                    n: undefined,
                    e: undefined,
                    crv: undefined,
                    x: undefined,
                    y: undefined,
                    x5u: undefined,
                    x5c: undefined,
                    x5t: undefined,
                    "x5t#S256": undefined,
                },
            ],
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .put("/keys/custom-signing")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.keys.customSigning.set({
                keys: [
                    {
                        kty: "EC",
                        kid: undefined,
                        use: undefined,
                        key_ops: undefined,
                        alg: undefined,
                        n: undefined,
                        e: undefined,
                        crv: undefined,
                        x: undefined,
                        y: undefined,
                        x5u: undefined,
                        x5c: undefined,
                        x5t: undefined,
                        "x5t#S256": undefined,
                    },
                    {
                        kty: "EC",
                        kid: undefined,
                        use: undefined,
                        key_ops: undefined,
                        alg: undefined,
                        n: undefined,
                        e: undefined,
                        crv: undefined,
                        x: undefined,
                        y: undefined,
                        x5u: undefined,
                        x5c: undefined,
                        x5t: undefined,
                        "x5t#S256": undefined,
                    },
                ],
            });
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("set (7a4b7805)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            keys: [
                {
                    kty: "EC",
                    kid: undefined,
                    use: undefined,
                    key_ops: undefined,
                    alg: undefined,
                    n: undefined,
                    e: undefined,
                    crv: undefined,
                    x: undefined,
                    y: undefined,
                    x5u: undefined,
                    x5c: undefined,
                    x5t: undefined,
                    "x5t#S256": undefined,
                },
                {
                    kty: "EC",
                    kid: undefined,
                    use: undefined,
                    key_ops: undefined,
                    alg: undefined,
                    n: undefined,
                    e: undefined,
                    crv: undefined,
                    x: undefined,
                    y: undefined,
                    x5u: undefined,
                    x5c: undefined,
                    x5t: undefined,
                    "x5t#S256": undefined,
                },
            ],
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .put("/keys/custom-signing")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.keys.customSigning.set({
                keys: [
                    {
                        kty: "EC",
                        kid: undefined,
                        use: undefined,
                        key_ops: undefined,
                        alg: undefined,
                        n: undefined,
                        e: undefined,
                        crv: undefined,
                        x: undefined,
                        y: undefined,
                        x5u: undefined,
                        x5c: undefined,
                        x5t: undefined,
                        "x5t#S256": undefined,
                    },
                    {
                        kty: "EC",
                        kid: undefined,
                        use: undefined,
                        key_ops: undefined,
                        alg: undefined,
                        n: undefined,
                        e: undefined,
                        crv: undefined,
                        x: undefined,
                        y: undefined,
                        x5u: undefined,
                        x5c: undefined,
                        x5t: undefined,
                        "x5t#S256": undefined,
                    },
                ],
            });
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("delete (5465b825)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        server.mockEndpoint().delete("/keys/custom-signing").respondWith().statusCode(200).build();
        const response = yield client.keys.customSigning.delete();
        expect(response).toEqual(undefined);
    }));
    test("delete (1e230aeb)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/keys/custom-signing")
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.keys.customSigning.delete();
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("delete (af841397)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/keys/custom-signing")
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.keys.customSigning.delete();
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("delete (ee1e23bf)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/keys/custom-signing")
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.keys.customSigning.delete();
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
});
