// auth0/dist/cjs/management/tests/wire/guardian/factors/pushNotification.test.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const MockServerPool_js_1 = require("../../../mock-server/MockServerPool.js");
const Client_js_1 = require("../../../../Client.js");
const Management = __importStar(require("../../../../api/index.js"));
describe("PushNotification", () => {
    test("getApnsProvider (e71df9aa)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { bundle_id: "bundle_id", sandbox: true, enabled: true };
        server
            .mockEndpoint()
            .get("/guardian/factors/push-notification/providers/apns")
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.guardian.factors.pushNotification.getApnsProvider();
        expect(response).toEqual({
            bundle_id: "bundle_id",
            sandbox: true,
            enabled: true,
        });
    }));
    test("getApnsProvider (c60dd33b)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/guardian/factors/push-notification/providers/apns")
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.guardian.factors.pushNotification.getApnsProvider();
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("getApnsProvider (1e230aeb)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/guardian/factors/push-notification/providers/apns")
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.guardian.factors.pushNotification.getApnsProvider();
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("getApnsProvider (af841397)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/guardian/factors/push-notification/providers/apns")
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.guardian.factors.pushNotification.getApnsProvider();
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("setApnsProvider (d64d78d4)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {};
        const rawResponseBody = { sandbox: true, bundle_id: "bundle_id" };
        server
            .mockEndpoint()
            .patch("/guardian/factors/push-notification/providers/apns")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.guardian.factors.pushNotification.setApnsProvider({});
        expect(response).toEqual({
            sandbox: true,
            bundle_id: "bundle_id",
        });
    }));
    test("setApnsProvider (18893c3f)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { sandbox: undefined, bundle_id: undefined, p12: undefined };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/guardian/factors/push-notification/providers/apns")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.guardian.factors.pushNotification.setApnsProvider({
                sandbox: undefined,
                bundle_id: undefined,
                p12: undefined,
            });
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("setApnsProvider (2639abcf)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { sandbox: undefined, bundle_id: undefined, p12: undefined };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/guardian/factors/push-notification/providers/apns")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.guardian.factors.pushNotification.setApnsProvider({
                sandbox: undefined,
                bundle_id: undefined,
                p12: undefined,
            });
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("setApnsProvider (8bf2f71b)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { sandbox: undefined, bundle_id: undefined, p12: undefined };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/guardian/factors/push-notification/providers/apns")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.guardian.factors.pushNotification.setApnsProvider({
                sandbox: undefined,
                bundle_id: undefined,
                p12: undefined,
            });
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("setFcmProvider (8093684b)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {};
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/guardian/factors/push-notification/providers/fcm")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.guardian.factors.pushNotification.setFcmProvider({});
        expect(response).toEqual({
            key: "value",
        });
    }));
    test("setFcmProvider (e13a3115)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { server_key: undefined };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/guardian/factors/push-notification/providers/fcm")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.guardian.factors.pushNotification.setFcmProvider({
                server_key: undefined,
            });
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("setFcmProvider (7c48e605)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { server_key: undefined };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/guardian/factors/push-notification/providers/fcm")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.guardian.factors.pushNotification.setFcmProvider({
                server_key: undefined,
            });
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("setFcmProvider (875fafd1)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { server_key: undefined };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/guardian/factors/push-notification/providers/fcm")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.guardian.factors.pushNotification.setFcmProvider({
                server_key: undefined,
            });
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("setFcmv1Provider (8093684b)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {};
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/guardian/factors/push-notification/providers/fcmv1")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.guardian.factors.pushNotification.setFcmv1Provider({});
        expect(response).toEqual({
            key: "value",
        });
    }));
    test("setFcmv1Provider (d2005a53)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { server_credentials: undefined };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/guardian/factors/push-notification/providers/fcmv1")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.guardian.factors.pushNotification.setFcmv1Provider({
                server_credentials: undefined,
            });
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("setFcmv1Provider (847052a3)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { server_credentials: undefined };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/guardian/factors/push-notification/providers/fcmv1")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.guardian.factors.pushNotification.setFcmv1Provider({
                server_credentials: undefined,
            });
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("setFcmv1Provider (66e3c94f)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { server_credentials: undefined };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/guardian/factors/push-notification/providers/fcmv1")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.guardian.factors.pushNotification.setFcmv1Provider({
                server_credentials: undefined,
            });
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("getSnsProvider (37487858)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = {
            aws_access_key_id: "aws_access_key_id",
            aws_secret_access_key: "aws_secret_access_key",
            aws_region: "aws_region",
            sns_apns_platform_application_arn: "sns_apns_platform_application_arn",
            sns_gcm_platform_application_arn: "sns_gcm_platform_application_arn",
        };
        server
            .mockEndpoint()
            .get("/guardian/factors/push-notification/providers/sns")
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.guardian.factors.pushNotification.getSnsProvider();
        expect(response).toEqual({
            aws_access_key_id: "aws_access_key_id",
            aws_secret_access_key: "aws_secret_access_key",
            aws_region: "aws_region",
            sns_apns_platform_application_arn: "sns_apns_platform_application_arn",
            sns_gcm_platform_application_arn: "sns_gcm_platform_application_arn",
        });
    }));
    test("getSnsProvider (c60dd33b)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/guardian/factors/push-notification/providers/sns")
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.guardian.factors.pushNotification.getSnsProvider();
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("getSnsProvider (1e230aeb)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/guardian/factors/push-notification/providers/sns")
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.guardian.factors.pushNotification.getSnsProvider();
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("getSnsProvider (af841397)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/guardian/factors/push-notification/providers/sns")
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.guardian.factors.pushNotification.getSnsProvider();
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("setSnsProvider (283b6363)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {};
        const rawResponseBody = {
            aws_access_key_id: "aws_access_key_id",
            aws_secret_access_key: "aws_secret_access_key",
            aws_region: "aws_region",
            sns_apns_platform_application_arn: "sns_apns_platform_application_arn",
            sns_gcm_platform_application_arn: "sns_gcm_platform_application_arn",
        };
        server
            .mockEndpoint()
            .put("/guardian/factors/push-notification/providers/sns")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.guardian.factors.pushNotification.setSnsProvider();
        expect(response).toEqual({
            aws_access_key_id: "aws_access_key_id",
            aws_secret_access_key: "aws_secret_access_key",
            aws_region: "aws_region",
            sns_apns_platform_application_arn: "sns_apns_platform_application_arn",
            sns_gcm_platform_application_arn: "sns_gcm_platform_application_arn",
        });
    }));
    test("setSnsProvider (836aca22)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            aws_access_key_id: undefined,
            aws_secret_access_key: undefined,
            aws_region: undefined,
            sns_apns_platform_application_arn: undefined,
            sns_gcm_platform_application_arn: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .put("/guardian/factors/push-notification/providers/sns")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.guardian.factors.pushNotification.setSnsProvider({
                aws_access_key_id: undefined,
                aws_secret_access_key: undefined,
                aws_region: undefined,
                sns_apns_platform_application_arn: undefined,
                sns_gcm_platform_application_arn: undefined,
            });
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("setSnsProvider (70d3f13a)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            aws_access_key_id: undefined,
            aws_secret_access_key: undefined,
            aws_region: undefined,
            sns_apns_platform_application_arn: undefined,
            sns_gcm_platform_application_arn: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .put("/guardian/factors/push-notification/providers/sns")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.guardian.factors.pushNotification.setSnsProvider({
                aws_access_key_id: undefined,
                aws_secret_access_key: undefined,
                aws_region: undefined,
                sns_apns_platform_application_arn: undefined,
                sns_gcm_platform_application_arn: undefined,
            });
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("setSnsProvider (d9c789fe)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            aws_access_key_id: undefined,
            aws_secret_access_key: undefined,
            aws_region: undefined,
            sns_apns_platform_application_arn: undefined,
            sns_gcm_platform_application_arn: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .put("/guardian/factors/push-notification/providers/sns")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.guardian.factors.pushNotification.setSnsProvider({
                aws_access_key_id: undefined,
                aws_secret_access_key: undefined,
                aws_region: undefined,
                sns_apns_platform_application_arn: undefined,
                sns_gcm_platform_application_arn: undefined,
            });
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("updateSnsProvider (283b6363)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {};
        const rawResponseBody = {
            aws_access_key_id: "aws_access_key_id",
            aws_secret_access_key: "aws_secret_access_key",
            aws_region: "aws_region",
            sns_apns_platform_application_arn: "sns_apns_platform_application_arn",
            sns_gcm_platform_application_arn: "sns_gcm_platform_application_arn",
        };
        server
            .mockEndpoint()
            .patch("/guardian/factors/push-notification/providers/sns")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.guardian.factors.pushNotification.updateSnsProvider();
        expect(response).toEqual({
            aws_access_key_id: "aws_access_key_id",
            aws_secret_access_key: "aws_secret_access_key",
            aws_region: "aws_region",
            sns_apns_platform_application_arn: "sns_apns_platform_application_arn",
            sns_gcm_platform_application_arn: "sns_gcm_platform_application_arn",
        });
    }));
    test("updateSnsProvider (836aca22)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            aws_access_key_id: undefined,
            aws_secret_access_key: undefined,
            aws_region: undefined,
            sns_apns_platform_application_arn: undefined,
            sns_gcm_platform_application_arn: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/guardian/factors/push-notification/providers/sns")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.guardian.factors.pushNotification.updateSnsProvider({
                aws_access_key_id: undefined,
                aws_secret_access_key: undefined,
                aws_region: undefined,
                sns_apns_platform_application_arn: undefined,
                sns_gcm_platform_application_arn: undefined,
            });
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("updateSnsProvider (70d3f13a)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            aws_access_key_id: undefined,
            aws_secret_access_key: undefined,
            aws_region: undefined,
            sns_apns_platform_application_arn: undefined,
            sns_gcm_platform_application_arn: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/guardian/factors/push-notification/providers/sns")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.guardian.factors.pushNotification.updateSnsProvider({
                aws_access_key_id: undefined,
                aws_secret_access_key: undefined,
                aws_region: undefined,
                sns_apns_platform_application_arn: undefined,
                sns_gcm_platform_application_arn: undefined,
            });
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("updateSnsProvider (d9c789fe)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            aws_access_key_id: undefined,
            aws_secret_access_key: undefined,
            aws_region: undefined,
            sns_apns_platform_application_arn: undefined,
            sns_gcm_platform_application_arn: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/guardian/factors/push-notification/providers/sns")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.guardian.factors.pushNotification.updateSnsProvider({
                aws_access_key_id: undefined,
                aws_secret_access_key: undefined,
                aws_region: undefined,
                sns_apns_platform_application_arn: undefined,
                sns_gcm_platform_application_arn: undefined,
            });
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("getSelectedProvider (23f6cc2)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { provider: "guardian" };
        server
            .mockEndpoint()
            .get("/guardian/factors/push-notification/selected-provider")
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.guardian.factors.pushNotification.getSelectedProvider();
        expect(response).toEqual({
            provider: "guardian",
        });
    }));
    test("getSelectedProvider (c60dd33b)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/guardian/factors/push-notification/selected-provider")
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.guardian.factors.pushNotification.getSelectedProvider();
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("getSelectedProvider (1e230aeb)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/guardian/factors/push-notification/selected-provider")
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.guardian.factors.pushNotification.getSelectedProvider();
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("getSelectedProvider (af841397)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/guardian/factors/push-notification/selected-provider")
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.guardian.factors.pushNotification.getSelectedProvider();
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("setProvider (277ecd47)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { provider: "guardian" };
        const rawResponseBody = { provider: "guardian" };
        server
            .mockEndpoint()
            .put("/guardian/factors/push-notification/selected-provider")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.guardian.factors.pushNotification.setProvider({
            provider: "guardian",
        });
        expect(response).toEqual({
            provider: "guardian",
        });
    }));
    test("setProvider (ccb41335)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { provider: "guardian" };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .put("/guardian/factors/push-notification/selected-provider")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.guardian.factors.pushNotification.setProvider({
                provider: "guardian",
            });
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("setProvider (41db1825)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { provider: "guardian" };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .put("/guardian/factors/push-notification/selected-provider")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.guardian.factors.pushNotification.setProvider({
                provider: "guardian",
            });
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("setProvider (9d3e93f1)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { provider: "guardian" };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .put("/guardian/factors/push-notification/selected-provider")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.guardian.factors.pushNotification.setProvider({
                provider: "guardian",
            });
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
});
