// auth0/dist/cjs/management/tests/wire/organizations/clientGrants.test.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const MockServerPool_js_1 = require("../../mock-server/MockServerPool.js");
const Client_js_1 = require("../../../Client.js");
const Management = __importStar(require("../../../api/index.js"));
describe("ClientGrants", () => {
    test("list (712badd5)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = {
            start: 1.1,
            limit: 1.1,
            total: 1.1,
            client_grants: [
                {
                    id: "id",
                    client_id: "client_id",
                    audience: "audience",
                    scope: ["scope"],
                    organization_usage: "deny",
                    allow_any_organization: true,
                },
            ],
        };
        server
            .mockEndpoint()
            .get("/organizations/id/client-grants")
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const expected = {
            start: 1.1,
            limit: 1.1,
            total: 1.1,
            client_grants: [
                {
                    id: "id",
                    client_id: "client_id",
                    audience: "audience",
                    scope: ["scope"],
                    organization_usage: "deny",
                    allow_any_organization: true,
                },
            ],
        };
        const page = yield client.organizations.clientGrants.list("id");
        expect(expected.client_grants).toEqual(page.data);
        expect(page.hasNextPage()).toBe(true);
        const nextPage = yield page.getNextPage();
        expect(expected.client_grants).toEqual(nextPage.data);
    }));
    test("list (fcf9dbd1)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/organizations/id/client-grants")
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.organizations.clientGrants.list("id");
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("list (49d52691)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/organizations/id/client-grants")
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.organizations.clientGrants.list("id");
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("list (2428808d)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/organizations/id/client-grants")
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.organizations.clientGrants.list("id");
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("list (27b44cb5)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/organizations/id/client-grants")
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.organizations.clientGrants.list("id");
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("create (91a0f4d4)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { grant_id: "grant_id" };
        const rawResponseBody = {
            id: "id",
            client_id: "client_id",
            audience: "audience",
            scope: ["scope"],
            organization_usage: "deny",
            allow_any_organization: true,
        };
        server
            .mockEndpoint()
            .post("/organizations/id/client-grants")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.organizations.clientGrants.create("id", {
            grant_id: "grant_id",
        });
        expect(response).toEqual({
            id: "id",
            client_id: "client_id",
            audience: "audience",
            scope: ["scope"],
            organization_usage: "deny",
            allow_any_organization: true,
        });
    }));
    test("create (d9fea040)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { grant_id: "grant_id" };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/organizations/id/client-grants")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.organizations.clientGrants.create("id", {
                grant_id: "grant_id",
            });
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("create (a071fa8)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { grant_id: "grant_id" };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/organizations/id/client-grants")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.organizations.clientGrants.create("id", {
                grant_id: "grant_id",
            });
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("create (6dcd5eac)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { grant_id: "grant_id" };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/organizations/id/client-grants")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.organizations.clientGrants.create("id", {
                grant_id: "grant_id",
            });
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("create (c68e00f4)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { grant_id: "grant_id" };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/organizations/id/client-grants")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(404)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.organizations.clientGrants.create("id", {
                grant_id: "grant_id",
            });
        })).rejects.toThrow(new Management.NotFoundError({
            key: "value",
        }));
    }));
    test("create (7fdbab44)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { grant_id: "grant_id" };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/organizations/id/client-grants")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(409)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.organizations.clientGrants.create("id", {
                grant_id: "grant_id",
            });
        })).rejects.toThrow(new Management.ConflictError({
            key: "value",
        }));
    }));
    test("create (ebc7dff8)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { grant_id: "grant_id" };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/organizations/id/client-grants")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.organizations.clientGrants.create("id", {
                grant_id: "grant_id",
            });
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("delete (c7c225ed)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        server.mockEndpoint().delete("/organizations/id/client-grants/grant_id").respondWith().statusCode(200).build();
        const response = yield client.organizations.clientGrants.delete("id", "grant_id");
        expect(response).toEqual(undefined);
    }));
    test("delete (ad08b9ee)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/organizations/id/client-grants/grant_id")
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.organizations.clientGrants.delete("id", "grant_id");
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("delete (e5d01576)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/organizations/id/client-grants/grant_id")
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.organizations.clientGrants.delete("id", "grant_id");
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("delete (cdfc707a)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/organizations/id/client-grants/grant_id")
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.organizations.clientGrants.delete("id", "grant_id");
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("delete (1120f272)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/organizations/id/client-grants/grant_id")
            .respondWith()
            .statusCode(404)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.organizations.clientGrants.delete("id", "grant_id");
        })).rejects.toThrow(new Management.NotFoundError({
            key: "value",
        }));
    }));
    test("delete (73fbb716)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/organizations/id/client-grants/grant_id")
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.organizations.clientGrants.delete("id", "grant_id");
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
});
