// auth0/dist/cjs/management/tests/wire/organizations/enabledConnections.test.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const MockServerPool_js_1 = require("../../mock-server/MockServerPool.js");
const Client_js_1 = require("../../../Client.js");
const Management = __importStar(require("../../../api/index.js"));
describe("EnabledConnections", () => {
    test("list (9e9afec2)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = {
            start: 1.1,
            limit: 1.1,
            total: 1.1,
            enabled_connections: [
                {
                    connection_id: "connection_id",
                    assign_membership_on_login: true,
                    show_as_button: true,
                    is_signup_enabled: true,
                },
            ],
        };
        server
            .mockEndpoint()
            .get("/organizations/id/enabled_connections")
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const expected = {
            start: 1.1,
            limit: 1.1,
            total: 1.1,
            enabled_connections: [
                {
                    connection_id: "connection_id",
                    assign_membership_on_login: true,
                    show_as_button: true,
                    is_signup_enabled: true,
                },
            ],
        };
        const page = yield client.organizations.enabledConnections.list("id");
        expect(expected.enabled_connections).toEqual(page.data);
        expect(page.hasNextPage()).toBe(true);
        const nextPage = yield page.getNextPage();
        expect(expected.enabled_connections).toEqual(nextPage.data);
    }));
    test("list (fcf9dbd1)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/organizations/id/enabled_connections")
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.organizations.enabledConnections.list("id");
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("list (49d52691)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/organizations/id/enabled_connections")
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.organizations.enabledConnections.list("id");
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("list (2428808d)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/organizations/id/enabled_connections")
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.organizations.enabledConnections.list("id");
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("list (27b44cb5)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/organizations/id/enabled_connections")
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.organizations.enabledConnections.list("id");
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("add (382c2607)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { connection_id: "connection_id" };
        const rawResponseBody = {
            connection_id: "connection_id",
            assign_membership_on_login: true,
            show_as_button: true,
            is_signup_enabled: true,
            connection: { name: "name", strategy: "strategy" },
        };
        server
            .mockEndpoint()
            .post("/organizations/id/enabled_connections")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.organizations.enabledConnections.add("id", {
            connection_id: "connection_id",
        });
        expect(response).toEqual({
            connection_id: "connection_id",
            assign_membership_on_login: true,
            show_as_button: true,
            is_signup_enabled: true,
            connection: {
                name: "name",
                strategy: "strategy",
            },
        });
    }));
    test("add (668f4ffb)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            connection_id: "connection_id",
            assign_membership_on_login: undefined,
            is_signup_enabled: undefined,
            show_as_button: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/organizations/id/enabled_connections")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.organizations.enabledConnections.add("id", {
                connection_id: "connection_id",
                assign_membership_on_login: undefined,
                is_signup_enabled: undefined,
                show_as_button: undefined,
            });
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("add (8a865bab)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            connection_id: "connection_id",
            assign_membership_on_login: undefined,
            is_signup_enabled: undefined,
            show_as_button: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/organizations/id/enabled_connections")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.organizations.enabledConnections.add("id", {
                connection_id: "connection_id",
                assign_membership_on_login: undefined,
                is_signup_enabled: undefined,
                show_as_button: undefined,
            });
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("add (a52a5457)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            connection_id: "connection_id",
            assign_membership_on_login: undefined,
            is_signup_enabled: undefined,
            show_as_button: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/organizations/id/enabled_connections")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.organizations.enabledConnections.add("id", {
                connection_id: "connection_id",
                assign_membership_on_login: undefined,
                is_signup_enabled: undefined,
                show_as_button: undefined,
            });
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("add (c9afca7f)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            connection_id: "connection_id",
            assign_membership_on_login: undefined,
            is_signup_enabled: undefined,
            show_as_button: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/organizations/id/enabled_connections")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.organizations.enabledConnections.add("id", {
                connection_id: "connection_id",
                assign_membership_on_login: undefined,
                is_signup_enabled: undefined,
                show_as_button: undefined,
            });
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("get (4775f6)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = {
            connection_id: "connection_id",
            assign_membership_on_login: true,
            show_as_button: true,
            is_signup_enabled: true,
            connection: { name: "name", strategy: "strategy" },
        };
        server
            .mockEndpoint()
            .get("/organizations/id/enabled_connections/connectionId")
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.organizations.enabledConnections.get("id", "connectionId");
        expect(response).toEqual({
            connection_id: "connection_id",
            assign_membership_on_login: true,
            show_as_button: true,
            is_signup_enabled: true,
            connection: {
                name: "name",
                strategy: "strategy",
            },
        });
    }));
    test("get (2508cc35)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/organizations/id/enabled_connections/connectionId")
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.organizations.enabledConnections.get("id", "connectionId");
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("get (7ee17801)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/organizations/id/enabled_connections/connectionId")
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.organizations.enabledConnections.get("id", "connectionId");
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("get (45533d9)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/organizations/id/enabled_connections/connectionId")
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.organizations.enabledConnections.get("id", "connectionId");
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("delete (54e78333)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        server
            .mockEndpoint()
            .delete("/organizations/id/enabled_connections/connectionId")
            .respondWith()
            .statusCode(200)
            .build();
        const response = yield client.organizations.enabledConnections.delete("id", "connectionId");
        expect(response).toEqual(undefined);
    }));
    test("delete (18fb2e85)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/organizations/id/enabled_connections/connectionId")
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.organizations.enabledConnections.delete("id", "connectionId");
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("delete (2508cc35)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/organizations/id/enabled_connections/connectionId")
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.organizations.enabledConnections.delete("id", "connectionId");
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("delete (7ee17801)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/organizations/id/enabled_connections/connectionId")
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.organizations.enabledConnections.delete("id", "connectionId");
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("delete (45533d9)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/organizations/id/enabled_connections/connectionId")
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.organizations.enabledConnections.delete("id", "connectionId");
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("update (70b98bad)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {};
        const rawResponseBody = {
            connection_id: "connection_id",
            assign_membership_on_login: true,
            show_as_button: true,
            is_signup_enabled: true,
            connection: { name: "name", strategy: "strategy" },
        };
        server
            .mockEndpoint()
            .patch("/organizations/id/enabled_connections/connectionId")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.organizations.enabledConnections.update("id", "connectionId");
        expect(response).toEqual({
            connection_id: "connection_id",
            assign_membership_on_login: true,
            show_as_button: true,
            is_signup_enabled: true,
            connection: {
                name: "name",
                strategy: "strategy",
            },
        });
    }));
    test("update (84bdeb84)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            assign_membership_on_login: undefined,
            is_signup_enabled: undefined,
            show_as_button: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/organizations/id/enabled_connections/connectionId")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.organizations.enabledConnections.update("id", "connectionId", {
                assign_membership_on_login: undefined,
                is_signup_enabled: undefined,
                show_as_button: undefined,
            });
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("update (37f38eac)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            assign_membership_on_login: undefined,
            is_signup_enabled: undefined,
            show_as_button: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/organizations/id/enabled_connections/connectionId")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.organizations.enabledConnections.update("id", "connectionId", {
                assign_membership_on_login: undefined,
                is_signup_enabled: undefined,
                show_as_button: undefined,
            });
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("update (77018130)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            assign_membership_on_login: undefined,
            is_signup_enabled: undefined,
            show_as_button: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/organizations/id/enabled_connections/connectionId")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.organizations.enabledConnections.update("id", "connectionId", {
                assign_membership_on_login: undefined,
                is_signup_enabled: undefined,
                show_as_button: undefined,
            });
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("update (e63acfbc)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            assign_membership_on_login: undefined,
            is_signup_enabled: undefined,
            show_as_button: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/organizations/id/enabled_connections/connectionId")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.organizations.enabledConnections.update("id", "connectionId", {
                assign_membership_on_login: undefined,
                is_signup_enabled: undefined,
                show_as_button: undefined,
            });
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
});
