// auth0/dist/cjs/management/tests/wire/clientGrants.test.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const MockServerPool_js_1 = require("../mock-server/MockServerPool.js");
const Client_js_1 = require("../../Client.js");
const Management = __importStar(require("../../api/index.js"));
describe("ClientGrants", () => {
    test("list (71245d34)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = {
            next: "next",
            client_grants: [
                {
                    id: "id",
                    client_id: "client_id",
                    audience: "audience",
                    scope: ["scope"],
                    organization_usage: "deny",
                    allow_any_organization: true,
                    is_system: true,
                    subject_type: "client",
                    authorization_details_types: ["authorization_details_types"],
                },
            ],
        };
        server.mockEndpoint().get("/client-grants").respondWith().statusCode(200).jsonBody(rawResponseBody).build();
        const expected = {
            next: "next",
            client_grants: [
                {
                    id: "id",
                    client_id: "client_id",
                    audience: "audience",
                    scope: ["scope"],
                    organization_usage: "deny",
                    allow_any_organization: true,
                    is_system: true,
                    subject_type: "client",
                    authorization_details_types: ["authorization_details_types"],
                },
            ],
        };
        const page = yield client.clientGrants.list();
        expect(expected.client_grants).toEqual(page.data);
        expect(page.hasNextPage()).toBe(true);
        const nextPage = yield page.getNextPage();
        expect(expected.client_grants).toEqual(nextPage.data);
    }));
    test("list (1e230aeb)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/client-grants").respondWith().statusCode(401).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.clientGrants.list();
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("list (af841397)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/client-grants").respondWith().statusCode(403).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.clientGrants.list();
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("list (ee1e23bf)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/client-grants").respondWith().statusCode(429).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.clientGrants.list();
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("create (a5547e74)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { client_id: "client_id", audience: "audience", scope: ["scope"] };
        const rawResponseBody = {
            id: "id",
            client_id: "client_id",
            audience: "audience",
            scope: ["scope"],
            organization_usage: "deny",
            allow_any_organization: true,
            is_system: true,
            subject_type: "client",
            authorization_details_types: ["authorization_details_types"],
        };
        server
            .mockEndpoint()
            .post("/client-grants")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.clientGrants.create({
            client_id: "client_id",
            audience: "audience",
            scope: ["scope"],
        });
        expect(response).toEqual({
            id: "id",
            client_id: "client_id",
            audience: "audience",
            scope: ["scope"],
            organization_usage: "deny",
            allow_any_organization: true,
            is_system: true,
            subject_type: "client",
            authorization_details_types: ["authorization_details_types"],
        });
    }));
    test("create (7c8d492)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            client_id: "client_id",
            audience: "x",
            organization_usage: undefined,
            allow_any_organization: undefined,
            scope: ["scope", "scope"],
            subject_type: undefined,
            authorization_details_types: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/client-grants")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.clientGrants.create({
                client_id: "client_id",
                audience: "x",
                organization_usage: undefined,
                allow_any_organization: undefined,
                scope: ["scope", "scope"],
                subject_type: undefined,
                authorization_details_types: undefined,
            });
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("create (27cf866a)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            client_id: "client_id",
            audience: "x",
            organization_usage: undefined,
            allow_any_organization: undefined,
            scope: ["scope", "scope"],
            subject_type: undefined,
            authorization_details_types: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/client-grants")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.clientGrants.create({
                client_id: "client_id",
                audience: "x",
                organization_usage: undefined,
                allow_any_organization: undefined,
                scope: ["scope", "scope"],
                subject_type: undefined,
                authorization_details_types: undefined,
            });
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("create (773ce6ae)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            client_id: "client_id",
            audience: "x",
            organization_usage: undefined,
            allow_any_organization: undefined,
            scope: ["scope", "scope"],
            subject_type: undefined,
            authorization_details_types: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/client-grants")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.clientGrants.create({
                client_id: "client_id",
                audience: "x",
                organization_usage: undefined,
                allow_any_organization: undefined,
                scope: ["scope", "scope"],
                subject_type: undefined,
                authorization_details_types: undefined,
            });
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("create (97f841e6)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            client_id: "client_id",
            audience: "x",
            organization_usage: undefined,
            allow_any_organization: undefined,
            scope: ["scope", "scope"],
            subject_type: undefined,
            authorization_details_types: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/client-grants")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(404)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.clientGrants.create({
                client_id: "client_id",
                audience: "x",
                organization_usage: undefined,
                allow_any_organization: undefined,
                scope: ["scope", "scope"],
                subject_type: undefined,
                authorization_details_types: undefined,
            });
        })).rejects.toThrow(new Management.NotFoundError({
            key: "value",
        }));
    }));
    test("create (89c8f26)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            client_id: "client_id",
            audience: "x",
            organization_usage: undefined,
            allow_any_organization: undefined,
            scope: ["scope", "scope"],
            subject_type: undefined,
            authorization_details_types: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/client-grants")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(409)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.clientGrants.create({
                client_id: "client_id",
                audience: "x",
                organization_usage: undefined,
                allow_any_organization: undefined,
                scope: ["scope", "scope"],
                subject_type: undefined,
                authorization_details_types: undefined,
            });
        })).rejects.toThrow(new Management.ConflictError({
            key: "value",
        }));
    }));
    test("create (414d57ba)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            client_id: "client_id",
            audience: "x",
            organization_usage: undefined,
            allow_any_organization: undefined,
            scope: ["scope", "scope"],
            subject_type: undefined,
            authorization_details_types: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/client-grants")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.clientGrants.create({
                client_id: "client_id",
                audience: "x",
                organization_usage: undefined,
                allow_any_organization: undefined,
                scope: ["scope", "scope"],
                subject_type: undefined,
                authorization_details_types: undefined,
            });
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("delete (c7f0a6bf)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        server.mockEndpoint().delete("/client-grants/id").respondWith().statusCode(200).build();
        const response = yield client.clientGrants.delete("id");
        expect(response).toEqual(undefined);
    }));
    test("delete (fcf9dbd1)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/client-grants/id")
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.clientGrants.delete("id");
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("delete (49d52691)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/client-grants/id")
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.clientGrants.delete("id");
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("delete (2428808d)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/client-grants/id")
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.clientGrants.delete("id");
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("delete (27b44cb5)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/client-grants/id")
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.clientGrants.delete("id");
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("update (4944f99e)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {};
        const rawResponseBody = {
            id: "id",
            client_id: "client_id",
            audience: "audience",
            scope: ["scope"],
            organization_usage: "deny",
            allow_any_organization: true,
            is_system: true,
            subject_type: "client",
            authorization_details_types: ["authorization_details_types"],
        };
        server
            .mockEndpoint()
            .patch("/client-grants/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.clientGrants.update("id");
        expect(response).toEqual({
            id: "id",
            client_id: "client_id",
            audience: "audience",
            scope: ["scope"],
            organization_usage: "deny",
            allow_any_organization: true,
            is_system: true,
            subject_type: "client",
            authorization_details_types: ["authorization_details_types"],
        });
    }));
    test("update (c6bee645)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            scope: undefined,
            organization_usage: undefined,
            allow_any_organization: undefined,
            authorization_details_types: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/client-grants/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.clientGrants.update("id", {
                scope: undefined,
                organization_usage: undefined,
                allow_any_organization: undefined,
                authorization_details_types: undefined,
            });
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("update (6d8ad6f5)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            scope: undefined,
            organization_usage: undefined,
            allow_any_organization: undefined,
            authorization_details_types: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/client-grants/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.clientGrants.update("id", {
                scope: undefined,
                organization_usage: undefined,
                allow_any_organization: undefined,
                authorization_details_types: undefined,
            });
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("update (373138c1)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            scope: undefined,
            organization_usage: undefined,
            allow_any_organization: undefined,
            authorization_details_types: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/client-grants/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.clientGrants.update("id", {
                scope: undefined,
                organization_usage: undefined,
                allow_any_organization: undefined,
                authorization_details_types: undefined,
            });
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("update (d394df41)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            scope: undefined,
            organization_usage: undefined,
            allow_any_organization: undefined,
            authorization_details_types: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/client-grants/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(404)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.clientGrants.update("id", {
                scope: undefined,
                organization_usage: undefined,
                allow_any_organization: undefined,
                authorization_details_types: undefined,
            });
        })).rejects.toThrow(new Management.NotFoundError({
            key: "value",
        }));
    }));
    test("update (11c7a199)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            scope: undefined,
            organization_usage: undefined,
            allow_any_organization: undefined,
            authorization_details_types: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/client-grants/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.clientGrants.update("id", {
                scope: undefined,
                organization_usage: undefined,
                allow_any_organization: undefined,
                authorization_details_types: undefined,
            });
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
});
