// auth0/dist/cjs/management/tests/wire/customDomains.test.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const MockServerPool_js_1 = require("../mock-server/MockServerPool.js");
const Client_js_1 = require("../../Client.js");
const Management = __importStar(require("../../api/index.js"));
describe("CustomDomains", () => {
    test("list (5d649b9b)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = [
            {
                custom_domain_id: "custom_domain_id",
                domain: "domain",
                primary: true,
                status: "pending_verification",
                type: "auth0_managed_certs",
                origin_domain_name: "origin_domain_name",
                verification: { methods: [{ name: "cname", record: "record" }] },
                custom_client_ip_header: "custom_client_ip_header",
                tls_policy: "tls_policy",
            },
        ];
        server.mockEndpoint().get("/custom-domains").respondWith().statusCode(200).jsonBody(rawResponseBody).build();
        const response = yield client.customDomains.list();
        expect(response).toEqual([
            {
                custom_domain_id: "custom_domain_id",
                domain: "domain",
                primary: true,
                status: "pending_verification",
                type: "auth0_managed_certs",
                origin_domain_name: "origin_domain_name",
                verification: {
                    methods: [
                        {
                            name: "cname",
                            record: "record",
                        },
                    ],
                },
                custom_client_ip_header: "custom_client_ip_header",
                tls_policy: "tls_policy",
            },
        ]);
    }));
    test("list (1e230aeb)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/custom-domains").respondWith().statusCode(401).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.customDomains.list();
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("list (af841397)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/custom-domains").respondWith().statusCode(403).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.customDomains.list();
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("list (ee1e23bf)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/custom-domains").respondWith().statusCode(429).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.customDomains.list();
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("create (51076508)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { domain: "domain", type: "auth0_managed_certs" };
        const rawResponseBody = {
            custom_domain_id: "custom_domain_id",
            domain: "domain",
            primary: true,
            status: "pending_verification",
            type: "auth0_managed_certs",
            verification: { methods: [{ name: "cname", record: "record" }] },
            custom_client_ip_header: "custom_client_ip_header",
            tls_policy: "tls_policy",
        };
        server
            .mockEndpoint()
            .post("/custom-domains")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.customDomains.create({
            domain: "domain",
            type: "auth0_managed_certs",
        });
        expect(response).toEqual({
            custom_domain_id: "custom_domain_id",
            domain: "domain",
            primary: true,
            status: "pending_verification",
            type: "auth0_managed_certs",
            verification: {
                methods: [
                    {
                        name: "cname",
                        record: "record",
                    },
                ],
            },
            custom_client_ip_header: "custom_client_ip_header",
            tls_policy: "tls_policy",
        });
    }));
    test("create (fc5b69d7)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            domain: "domain",
            type: "auth0_managed_certs",
            verification_method: undefined,
            tls_policy: undefined,
            custom_client_ip_header: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/custom-domains")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.customDomains.create({
                domain: "domain",
                type: "auth0_managed_certs",
                verification_method: undefined,
                tls_policy: undefined,
                custom_client_ip_header: undefined,
            });
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("create (6ed4ada7)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            domain: "domain",
            type: "auth0_managed_certs",
            verification_method: undefined,
            tls_policy: undefined,
            custom_client_ip_header: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/custom-domains")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.customDomains.create({
                domain: "domain",
                type: "auth0_managed_certs",
                verification_method: undefined,
                tls_policy: undefined,
                custom_client_ip_header: undefined,
            });
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("create (49d6cdb3)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            domain: "domain",
            type: "auth0_managed_certs",
            verification_method: undefined,
            tls_policy: undefined,
            custom_client_ip_header: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/custom-domains")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.customDomains.create({
                domain: "domain",
                type: "auth0_managed_certs",
                verification_method: undefined,
                tls_policy: undefined,
                custom_client_ip_header: undefined,
            });
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("create (dbe89dcb)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            domain: "domain",
            type: "auth0_managed_certs",
            verification_method: undefined,
            tls_policy: undefined,
            custom_client_ip_header: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/custom-domains")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(409)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.customDomains.create({
                domain: "domain",
                type: "auth0_managed_certs",
                verification_method: undefined,
                tls_policy: undefined,
                custom_client_ip_header: undefined,
            });
        })).rejects.toThrow(new Management.ConflictError({
            key: "value",
        }));
    }));
    test("create (38d619ab)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            domain: "domain",
            type: "auth0_managed_certs",
            verification_method: undefined,
            tls_policy: undefined,
            custom_client_ip_header: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/custom-domains")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.customDomains.create({
                domain: "domain",
                type: "auth0_managed_certs",
                verification_method: undefined,
                tls_policy: undefined,
                custom_client_ip_header: undefined,
            });
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("get (50e36c51)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = {
            custom_domain_id: "custom_domain_id",
            domain: "domain",
            primary: true,
            status: "pending_verification",
            type: "auth0_managed_certs",
            origin_domain_name: "origin_domain_name",
            verification: { methods: [{ name: "cname", record: "record" }] },
            custom_client_ip_header: "custom_client_ip_header",
            tls_policy: "tls_policy",
        };
        server.mockEndpoint().get("/custom-domains/id").respondWith().statusCode(200).jsonBody(rawResponseBody).build();
        const response = yield client.customDomains.get("id");
        expect(response).toEqual({
            custom_domain_id: "custom_domain_id",
            domain: "domain",
            primary: true,
            status: "pending_verification",
            type: "auth0_managed_certs",
            origin_domain_name: "origin_domain_name",
            verification: {
                methods: [
                    {
                        name: "cname",
                        record: "record",
                    },
                ],
            },
            custom_client_ip_header: "custom_client_ip_header",
            tls_policy: "tls_policy",
        });
    }));
    test("get (fcf9dbd1)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/custom-domains/id").respondWith().statusCode(400).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.customDomains.get("id");
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("get (49d52691)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/custom-domains/id").respondWith().statusCode(401).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.customDomains.get("id");
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("get (2428808d)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/custom-domains/id").respondWith().statusCode(403).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.customDomains.get("id");
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("get (e55ce3fd)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/custom-domains/id").respondWith().statusCode(404).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.customDomains.get("id");
        })).rejects.toThrow(new Management.NotFoundError({
            key: "value",
        }));
    }));
    test("get (27b44cb5)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/custom-domains/id").respondWith().statusCode(429).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.customDomains.get("id");
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("delete (c7f0a6bf)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        server.mockEndpoint().delete("/custom-domains/id").respondWith().statusCode(200).build();
        const response = yield client.customDomains.delete("id");
        expect(response).toEqual(undefined);
    }));
    test("delete (fcf9dbd1)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/custom-domains/id")
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.customDomains.delete("id");
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("delete (49d52691)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/custom-domains/id")
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.customDomains.delete("id");
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("delete (2428808d)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/custom-domains/id")
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.customDomains.delete("id");
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("delete (27b44cb5)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/custom-domains/id")
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.customDomains.delete("id");
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("update (89c9b9e)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {};
        const rawResponseBody = {
            custom_domain_id: "custom_domain_id",
            domain: "domain",
            primary: true,
            status: "pending_verification",
            type: "auth0_managed_certs",
            verification: { methods: [{ name: "cname", record: "record" }] },
            custom_client_ip_header: "custom_client_ip_header",
            tls_policy: "tls_policy",
        };
        server
            .mockEndpoint()
            .patch("/custom-domains/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.customDomains.update("id");
        expect(response).toEqual({
            custom_domain_id: "custom_domain_id",
            domain: "domain",
            primary: true,
            status: "pending_verification",
            type: "auth0_managed_certs",
            verification: {
                methods: [
                    {
                        name: "cname",
                        record: "record",
                    },
                ],
            },
            custom_client_ip_header: "custom_client_ip_header",
            tls_policy: "tls_policy",
        });
    }));
    test("update (1ad4fbe6)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { tls_policy: undefined, custom_client_ip_header: undefined };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/custom-domains/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.customDomains.update("id", {
                tls_policy: undefined,
                custom_client_ip_header: undefined,
            });
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("update (da08224e)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { tls_policy: undefined, custom_client_ip_header: undefined };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/custom-domains/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.customDomains.update("id", {
                tls_policy: undefined,
                custom_client_ip_header: undefined,
            });
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("update (a0747af2)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { tls_policy: undefined, custom_client_ip_header: undefined };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/custom-domains/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.customDomains.update("id", {
                tls_policy: undefined,
                custom_client_ip_header: undefined,
            });
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("update (a4a6fee)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { tls_policy: undefined, custom_client_ip_header: undefined };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/custom-domains/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.customDomains.update("id", {
                tls_policy: undefined,
                custom_client_ip_header: undefined,
            });
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("test (d52f7dcb)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { success: true, message: "message" };
        server
            .mockEndpoint()
            .post("/custom-domains/id/test")
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.customDomains.test("id");
        expect(response).toEqual({
            success: true,
            message: "message",
        });
    }));
    test("test (fcf9dbd1)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/custom-domains/id/test")
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.customDomains.test("id");
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("test (49d52691)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/custom-domains/id/test")
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.customDomains.test("id");
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("test (2428808d)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/custom-domains/id/test")
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.customDomains.test("id");
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("test (e55ce3fd)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/custom-domains/id/test")
            .respondWith()
            .statusCode(404)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.customDomains.test("id");
        })).rejects.toThrow(new Management.NotFoundError({
            key: "value",
        }));
    }));
    test("test (3a5dfad5)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/custom-domains/id/test")
            .respondWith()
            .statusCode(409)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.customDomains.test("id");
        })).rejects.toThrow(new Management.ConflictError({
            key: "value",
        }));
    }));
    test("verify (4adc706b)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = {
            custom_domain_id: "custom_domain_id",
            domain: "domain",
            primary: true,
            status: "pending_verification",
            type: "auth0_managed_certs",
            cname_api_key: "cname_api_key",
            origin_domain_name: "origin_domain_name",
            verification: { methods: [{ name: "cname", record: "record" }] },
            custom_client_ip_header: "custom_client_ip_header",
            tls_policy: "tls_policy",
        };
        server
            .mockEndpoint()
            .post("/custom-domains/id/verify")
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.customDomains.verify("id");
        expect(response).toEqual({
            custom_domain_id: "custom_domain_id",
            domain: "domain",
            primary: true,
            status: "pending_verification",
            type: "auth0_managed_certs",
            cname_api_key: "cname_api_key",
            origin_domain_name: "origin_domain_name",
            verification: {
                methods: [
                    {
                        name: "cname",
                        record: "record",
                    },
                ],
            },
            custom_client_ip_header: "custom_client_ip_header",
            tls_policy: "tls_policy",
        });
    }));
    test("verify (fcf9dbd1)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/custom-domains/id/verify")
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.customDomains.verify("id");
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("verify (49d52691)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/custom-domains/id/verify")
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.customDomains.verify("id");
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("verify (2428808d)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/custom-domains/id/verify")
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.customDomains.verify("id");
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("verify (e55ce3fd)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/custom-domains/id/verify")
            .respondWith()
            .statusCode(404)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.customDomains.verify("id");
        })).rejects.toThrow(new Management.NotFoundError({
            key: "value",
        }));
    }));
    test("verify (27b44cb5)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/custom-domains/id/verify")
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.customDomains.verify("id");
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
});
