// auth0/dist/cjs/management/tests/wire/deviceCredentials.test.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const MockServerPool_js_1 = require("../mock-server/MockServerPool.js");
const Client_js_1 = require("../../Client.js");
const Management = __importStar(require("../../api/index.js"));
describe("DeviceCredentials", () => {
    test("list (f7f51b7)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = {
            start: 1.1,
            limit: 1.1,
            total: 1.1,
            device_credentials: [
                {
                    id: "id",
                    device_name: "device_name",
                    device_id: "device_id",
                    type: "public_key",
                    user_id: "user_id",
                    client_id: "client_id",
                },
            ],
        };
        server
            .mockEndpoint()
            .get("/device-credentials")
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const expected = {
            start: 1.1,
            limit: 1.1,
            total: 1.1,
            device_credentials: [
                {
                    id: "id",
                    device_name: "device_name",
                    device_id: "device_id",
                    type: "public_key",
                    user_id: "user_id",
                    client_id: "client_id",
                },
            ],
        };
        const page = yield client.deviceCredentials.list();
        expect(expected.device_credentials).toEqual(page.data);
        expect(page.hasNextPage()).toBe(true);
        const nextPage = yield page.getNextPage();
        expect(expected.device_credentials).toEqual(nextPage.data);
    }));
    test("list (c60dd33b)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/device-credentials")
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.deviceCredentials.list();
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("list (1e230aeb)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/device-credentials")
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.deviceCredentials.list();
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("list (af841397)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/device-credentials")
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.deviceCredentials.list();
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("list (ee1e23bf)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/device-credentials")
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.deviceCredentials.list();
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("createPublicKey (6bf0f16)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            device_name: "device_name",
            type: "public_key",
            value: "value",
            device_id: "device_id",
        };
        const rawResponseBody = { id: "id" };
        server
            .mockEndpoint()
            .post("/device-credentials")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.deviceCredentials.createPublicKey({
            device_name: "device_name",
            value: "value",
            device_id: "device_id",
        });
        expect(response).toEqual({
            id: "id",
        });
    }));
    test("createPublicKey (4bc65d1b)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            device_name: "x",
            type: "public_key",
            value: "x",
            device_id: "device_id",
            client_id: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/device-credentials")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.deviceCredentials.createPublicKey({
                device_name: "x",
                value: "x",
                device_id: "device_id",
                client_id: undefined,
            });
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("createPublicKey (73f319cb)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            device_name: "x",
            type: "public_key",
            value: "x",
            device_id: "device_id",
            client_id: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/device-credentials")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.deviceCredentials.createPublicKey({
                device_name: "x",
                value: "x",
                device_id: "device_id",
                client_id: undefined,
            });
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("createPublicKey (43a7f577)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            device_name: "x",
            type: "public_key",
            value: "x",
            device_id: "device_id",
            client_id: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/device-credentials")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.deviceCredentials.createPublicKey({
                device_name: "x",
                value: "x",
                device_id: "device_id",
                client_id: undefined,
            });
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("createPublicKey (5a04990f)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            device_name: "x",
            type: "public_key",
            value: "x",
            device_id: "device_id",
            client_id: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/device-credentials")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(409)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.deviceCredentials.createPublicKey({
                device_name: "x",
                value: "x",
                device_id: "device_id",
                client_id: undefined,
            });
        })).rejects.toThrow(new Management.ConflictError({
            key: "value",
        }));
    }));
    test("createPublicKey (891e351f)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            device_name: "x",
            type: "public_key",
            value: "x",
            device_id: "device_id",
            client_id: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/device-credentials")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.deviceCredentials.createPublicKey({
                device_name: "x",
                value: "x",
                device_id: "device_id",
                client_id: undefined,
            });
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("delete (c7f0a6bf)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        server.mockEndpoint().delete("/device-credentials/id").respondWith().statusCode(200).build();
        const response = yield client.deviceCredentials.delete("id");
        expect(response).toEqual(undefined);
    }));
    test("delete (fcf9dbd1)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/device-credentials/id")
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.deviceCredentials.delete("id");
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("delete (49d52691)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/device-credentials/id")
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.deviceCredentials.delete("id");
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("delete (2428808d)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/device-credentials/id")
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.deviceCredentials.delete("id");
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("delete (27b44cb5)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/device-credentials/id")
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.deviceCredentials.delete("id");
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
});
