// auth0/dist/cjs/management/tests/wire/emailTemplates.test.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const MockServerPool_js_1 = require("../mock-server/MockServerPool.js");
const Client_js_1 = require("../../Client.js");
const Management = __importStar(require("../../api/index.js"));
describe("EmailTemplates", () => {
    test("create (57a851c3)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { template: "verify_email" };
        const rawResponseBody = {
            template: "verify_email",
            body: "body",
            from: "from",
            resultUrl: "resultUrl",
            subject: "subject",
            syntax: "syntax",
            urlLifetimeInSeconds: 1.1,
            includeEmailInRedirect: true,
            enabled: true,
        };
        server
            .mockEndpoint()
            .post("/email-templates")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.emailTemplates.create({
            template: "verify_email",
        });
        expect(response).toEqual({
            template: "verify_email",
            body: "body",
            from: "from",
            resultUrl: "resultUrl",
            subject: "subject",
            syntax: "syntax",
            urlLifetimeInSeconds: 1.1,
            includeEmailInRedirect: true,
            enabled: true,
        });
    }));
    test("create (2950673f)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            template: "verify_email",
            body: undefined,
            from: undefined,
            resultUrl: undefined,
            subject: undefined,
            syntax: undefined,
            urlLifetimeInSeconds: undefined,
            includeEmailInRedirect: undefined,
            enabled: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/email-templates")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.emailTemplates.create({
                template: "verify_email",
                body: undefined,
                from: undefined,
                resultUrl: undefined,
                subject: undefined,
                syntax: undefined,
                urlLifetimeInSeconds: undefined,
                includeEmailInRedirect: undefined,
                enabled: undefined,
            });
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("create (d5a196cf)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            template: "verify_email",
            body: undefined,
            from: undefined,
            resultUrl: undefined,
            subject: undefined,
            syntax: undefined,
            urlLifetimeInSeconds: undefined,
            includeEmailInRedirect: undefined,
            enabled: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/email-templates")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.emailTemplates.create({
                template: "verify_email",
                body: undefined,
                from: undefined,
                resultUrl: undefined,
                subject: undefined,
                syntax: undefined,
                urlLifetimeInSeconds: undefined,
                includeEmailInRedirect: undefined,
                enabled: undefined,
            });
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("create (2fe2da1b)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            template: "verify_email",
            body: undefined,
            from: undefined,
            resultUrl: undefined,
            subject: undefined,
            syntax: undefined,
            urlLifetimeInSeconds: undefined,
            includeEmailInRedirect: undefined,
            enabled: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/email-templates")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.emailTemplates.create({
                template: "verify_email",
                body: undefined,
                from: undefined,
                resultUrl: undefined,
                subject: undefined,
                syntax: undefined,
                urlLifetimeInSeconds: undefined,
                includeEmailInRedirect: undefined,
                enabled: undefined,
            });
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("create (87f15e93)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            template: "verify_email",
            body: undefined,
            from: undefined,
            resultUrl: undefined,
            subject: undefined,
            syntax: undefined,
            urlLifetimeInSeconds: undefined,
            includeEmailInRedirect: undefined,
            enabled: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/email-templates")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(409)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.emailTemplates.create({
                template: "verify_email",
                body: undefined,
                from: undefined,
                resultUrl: undefined,
                subject: undefined,
                syntax: undefined,
                urlLifetimeInSeconds: undefined,
                includeEmailInRedirect: undefined,
                enabled: undefined,
            });
        })).rejects.toThrow(new Management.ConflictError({
            key: "value",
        }));
    }));
    test("create (785c8ed3)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            template: "verify_email",
            body: undefined,
            from: undefined,
            resultUrl: undefined,
            subject: undefined,
            syntax: undefined,
            urlLifetimeInSeconds: undefined,
            includeEmailInRedirect: undefined,
            enabled: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/email-templates")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.emailTemplates.create({
                template: "verify_email",
                body: undefined,
                from: undefined,
                resultUrl: undefined,
                subject: undefined,
                syntax: undefined,
                urlLifetimeInSeconds: undefined,
                includeEmailInRedirect: undefined,
                enabled: undefined,
            });
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("get (f7f2f263)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = {
            template: "verify_email",
            body: "body",
            from: "from",
            resultUrl: "resultUrl",
            subject: "subject",
            syntax: "syntax",
            urlLifetimeInSeconds: 1.1,
            includeEmailInRedirect: true,
            enabled: true,
        };
        server
            .mockEndpoint()
            .get("/email-templates/verify_email")
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.emailTemplates.get("verify_email");
        expect(response).toEqual({
            template: "verify_email",
            body: "body",
            from: "from",
            resultUrl: "resultUrl",
            subject: "subject",
            syntax: "syntax",
            urlLifetimeInSeconds: 1.1,
            includeEmailInRedirect: true,
            enabled: true,
        });
    }));
    test("get (7b91a5cf)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/email-templates/verify_email")
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.emailTemplates.get("verify_email");
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("get (fbc4e15f)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/email-templates/verify_email")
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.emailTemplates.get("verify_email");
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("get (5f38062b)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/email-templates/verify_email")
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.emailTemplates.get("verify_email");
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("get (d9541e23)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/email-templates/verify_email")
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.emailTemplates.get("verify_email");
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("set (514c2a72)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { template: "verify_email" };
        const rawResponseBody = {
            template: "verify_email",
            body: "body",
            from: "from",
            resultUrl: "resultUrl",
            subject: "subject",
            syntax: "syntax",
            urlLifetimeInSeconds: 1.1,
            includeEmailInRedirect: true,
            enabled: true,
        };
        server
            .mockEndpoint()
            .put("/email-templates/verify_email")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.emailTemplates.set("verify_email", {
            template: "verify_email",
        });
        expect(response).toEqual({
            template: "verify_email",
            body: "body",
            from: "from",
            resultUrl: "resultUrl",
            subject: "subject",
            syntax: "syntax",
            urlLifetimeInSeconds: 1.1,
            includeEmailInRedirect: true,
            enabled: true,
        });
    }));
    test("set (5ae28ccb)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            template: "verify_email",
            body: undefined,
            from: undefined,
            resultUrl: undefined,
            subject: undefined,
            syntax: undefined,
            urlLifetimeInSeconds: undefined,
            includeEmailInRedirect: undefined,
            enabled: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .put("/email-templates/verify_email")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.emailTemplates.set("verify_email", {
                template: "verify_email",
                body: undefined,
                from: undefined,
                resultUrl: undefined,
                subject: undefined,
                syntax: undefined,
                urlLifetimeInSeconds: undefined,
                includeEmailInRedirect: undefined,
                enabled: undefined,
            });
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("set (2e3a953b)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            template: "verify_email",
            body: undefined,
            from: undefined,
            resultUrl: undefined,
            subject: undefined,
            syntax: undefined,
            urlLifetimeInSeconds: undefined,
            includeEmailInRedirect: undefined,
            enabled: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .put("/email-templates/verify_email")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.emailTemplates.set("verify_email", {
                template: "verify_email",
                body: undefined,
                from: undefined,
                resultUrl: undefined,
                subject: undefined,
                syntax: undefined,
                urlLifetimeInSeconds: undefined,
                includeEmailInRedirect: undefined,
                enabled: undefined,
            });
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("set (b42c4f27)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            template: "verify_email",
            body: undefined,
            from: undefined,
            resultUrl: undefined,
            subject: undefined,
            syntax: undefined,
            urlLifetimeInSeconds: undefined,
            includeEmailInRedirect: undefined,
            enabled: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .put("/email-templates/verify_email")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.emailTemplates.set("verify_email", {
                template: "verify_email",
                body: undefined,
                from: undefined,
                resultUrl: undefined,
                subject: undefined,
                syntax: undefined,
                urlLifetimeInSeconds: undefined,
                includeEmailInRedirect: undefined,
                enabled: undefined,
            });
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("set (ac443757)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            template: "verify_email",
            body: undefined,
            from: undefined,
            resultUrl: undefined,
            subject: undefined,
            syntax: undefined,
            urlLifetimeInSeconds: undefined,
            includeEmailInRedirect: undefined,
            enabled: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .put("/email-templates/verify_email")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(404)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.emailTemplates.set("verify_email", {
                template: "verify_email",
                body: undefined,
                from: undefined,
                resultUrl: undefined,
                subject: undefined,
                syntax: undefined,
                urlLifetimeInSeconds: undefined,
                includeEmailInRedirect: undefined,
                enabled: undefined,
            });
        })).rejects.toThrow(new Management.NotFoundError({
            key: "value",
        }));
    }));
    test("set (3465cd0f)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            template: "verify_email",
            body: undefined,
            from: undefined,
            resultUrl: undefined,
            subject: undefined,
            syntax: undefined,
            urlLifetimeInSeconds: undefined,
            includeEmailInRedirect: undefined,
            enabled: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .put("/email-templates/verify_email")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.emailTemplates.set("verify_email", {
                template: "verify_email",
                body: undefined,
                from: undefined,
                resultUrl: undefined,
                subject: undefined,
                syntax: undefined,
                urlLifetimeInSeconds: undefined,
                includeEmailInRedirect: undefined,
                enabled: undefined,
            });
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("update (eaf199e)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {};
        const rawResponseBody = {
            template: "verify_email",
            body: "body",
            from: "from",
            resultUrl: "resultUrl",
            subject: "subject",
            syntax: "syntax",
            urlLifetimeInSeconds: 1.1,
            includeEmailInRedirect: true,
            enabled: true,
        };
        server
            .mockEndpoint()
            .patch("/email-templates/verify_email")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.emailTemplates.update("verify_email");
        expect(response).toEqual({
            template: "verify_email",
            body: "body",
            from: "from",
            resultUrl: "resultUrl",
            subject: "subject",
            syntax: "syntax",
            urlLifetimeInSeconds: 1.1,
            includeEmailInRedirect: true,
            enabled: true,
        });
    }));
    test("update (2fc11c8a)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            template: undefined,
            body: undefined,
            from: undefined,
            resultUrl: undefined,
            subject: undefined,
            syntax: undefined,
            urlLifetimeInSeconds: undefined,
            includeEmailInRedirect: undefined,
            enabled: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/email-templates/verify_email")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.emailTemplates.update("verify_email", {
                template: undefined,
                body: undefined,
                from: undefined,
                resultUrl: undefined,
                subject: undefined,
                syntax: undefined,
                urlLifetimeInSeconds: undefined,
                includeEmailInRedirect: undefined,
                enabled: undefined,
            });
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("update (81cf60e2)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            template: undefined,
            body: undefined,
            from: undefined,
            resultUrl: undefined,
            subject: undefined,
            syntax: undefined,
            urlLifetimeInSeconds: undefined,
            includeEmailInRedirect: undefined,
            enabled: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/email-templates/verify_email")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.emailTemplates.update("verify_email", {
                template: undefined,
                body: undefined,
                from: undefined,
                resultUrl: undefined,
                subject: undefined,
                syntax: undefined,
                urlLifetimeInSeconds: undefined,
                includeEmailInRedirect: undefined,
                enabled: undefined,
            });
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("update (2c22746)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            template: undefined,
            body: undefined,
            from: undefined,
            resultUrl: undefined,
            subject: undefined,
            syntax: undefined,
            urlLifetimeInSeconds: undefined,
            includeEmailInRedirect: undefined,
            enabled: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/email-templates/verify_email")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.emailTemplates.update("verify_email", {
                template: undefined,
                body: undefined,
                from: undefined,
                resultUrl: undefined,
                subject: undefined,
                syntax: undefined,
                urlLifetimeInSeconds: undefined,
                includeEmailInRedirect: undefined,
                enabled: undefined,
            });
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("update (f30efdfe)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            template: undefined,
            body: undefined,
            from: undefined,
            resultUrl: undefined,
            subject: undefined,
            syntax: undefined,
            urlLifetimeInSeconds: undefined,
            includeEmailInRedirect: undefined,
            enabled: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/email-templates/verify_email")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(404)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.emailTemplates.update("verify_email", {
                template: undefined,
                body: undefined,
                from: undefined,
                resultUrl: undefined,
                subject: undefined,
                syntax: undefined,
                urlLifetimeInSeconds: undefined,
                includeEmailInRedirect: undefined,
                enabled: undefined,
            });
        })).rejects.toThrow(new Management.NotFoundError({
            key: "value",
        }));
    }));
    test("update (f25f3052)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            template: undefined,
            body: undefined,
            from: undefined,
            resultUrl: undefined,
            subject: undefined,
            syntax: undefined,
            urlLifetimeInSeconds: undefined,
            includeEmailInRedirect: undefined,
            enabled: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/email-templates/verify_email")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.emailTemplates.update("verify_email", {
                template: undefined,
                body: undefined,
                from: undefined,
                resultUrl: undefined,
                subject: undefined,
                syntax: undefined,
                urlLifetimeInSeconds: undefined,
                includeEmailInRedirect: undefined,
                enabled: undefined,
            });
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
});
