// auth0/dist/cjs/management/tests/wire/forms.test.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const MockServerPool_js_1 = require("../mock-server/MockServerPool.js");
const Client_js_1 = require("../../Client.js");
const Management = __importStar(require("../../api/index.js"));
describe("Forms", () => {
    test("list (ac62b2b)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = {
            start: 1.1,
            limit: 1.1,
            total: 1.1,
            forms: [
                {
                    id: "id",
                    name: "name",
                    created_at: "2024-01-15T09:30:00Z",
                    updated_at: "2024-01-15T09:30:00Z",
                    embedded_at: "embedded_at",
                    submitted_at: "submitted_at",
                },
            ],
        };
        server.mockEndpoint().get("/forms").respondWith().statusCode(200).jsonBody(rawResponseBody).build();
        const expected = {
            start: 1.1,
            limit: 1.1,
            total: 1.1,
            forms: [
                {
                    id: "id",
                    name: "name",
                    created_at: "2024-01-15T09:30:00Z",
                    updated_at: "2024-01-15T09:30:00Z",
                    embedded_at: "embedded_at",
                    submitted_at: "submitted_at",
                },
            ],
        };
        const page = yield client.forms.list();
        expect(expected.forms).toEqual(page.data);
        expect(page.hasNextPage()).toBe(true);
        const nextPage = yield page.getNextPage();
        expect(expected.forms).toEqual(nextPage.data);
    }));
    test("list (c60dd33b)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/forms").respondWith().statusCode(400).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.forms.list();
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("list (1e230aeb)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/forms").respondWith().statusCode(401).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.forms.list();
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("list (af841397)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/forms").respondWith().statusCode(403).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.forms.list();
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("list (ee1e23bf)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/forms").respondWith().statusCode(429).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.forms.list();
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("create (be2e52c4)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { name: "name" };
        const rawResponseBody = {
            id: "id",
            name: "name",
            messages: { errors: { key: "value" }, custom: { key: "value" } },
            languages: { primary: "primary", default: "default" },
            translations: { key: { key: "value" } },
            nodes: [
                { id: "id", type: "FLOW", coordinates: { x: 1, y: 1 }, alias: "alias", config: { flow_id: "flow_id" } },
            ],
            start: { hidden_fields: [{ key: "key" }], next_node: "next_node", coordinates: { x: 1, y: 1 } },
            ending: {
                redirection: { delay: 1, target: "target" },
                after_submit: { flow_id: "flow_id" },
                coordinates: { x: 1, y: 1 },
                resume_flow: true,
            },
            style: { css: "css" },
            created_at: "2024-01-15T09:30:00Z",
            updated_at: "2024-01-15T09:30:00Z",
            embedded_at: "embedded_at",
            submitted_at: "submitted_at",
        };
        server
            .mockEndpoint()
            .post("/forms")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.forms.create({
            name: "name",
        });
        expect(response).toEqual({
            id: "id",
            name: "name",
            messages: {
                errors: {
                    key: "value",
                },
                custom: {
                    key: "value",
                },
            },
            languages: {
                primary: "primary",
                default: "default",
            },
            translations: {
                key: {
                    key: "value",
                },
            },
            nodes: [
                {
                    id: "id",
                    type: "FLOW",
                    coordinates: {
                        x: 1,
                        y: 1,
                    },
                    alias: "alias",
                    config: {
                        flow_id: "flow_id",
                    },
                },
            ],
            start: {
                hidden_fields: [
                    {
                        key: "key",
                    },
                ],
                next_node: "next_node",
                coordinates: {
                    x: 1,
                    y: 1,
                },
            },
            ending: {
                redirection: {
                    delay: 1,
                    target: "target",
                },
                after_submit: {
                    flow_id: "flow_id",
                },
                coordinates: {
                    x: 1,
                    y: 1,
                },
                resume_flow: true,
            },
            style: {
                css: "css",
            },
            created_at: "2024-01-15T09:30:00Z",
            updated_at: "2024-01-15T09:30:00Z",
            embedded_at: "embedded_at",
            submitted_at: "submitted_at",
        });
    }));
    test("create (23664f69)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            name: "x",
            messages: undefined,
            languages: undefined,
            translations: undefined,
            nodes: undefined,
            start: undefined,
            ending: undefined,
            style: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/forms")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.forms.create({
                name: "x",
                messages: undefined,
                languages: undefined,
                translations: undefined,
                nodes: undefined,
                start: undefined,
                ending: undefined,
                style: undefined,
            });
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("create (e22e6749)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            name: "x",
            messages: undefined,
            languages: undefined,
            translations: undefined,
            nodes: undefined,
            start: undefined,
            ending: undefined,
            style: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/forms")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.forms.create({
                name: "x",
                messages: undefined,
                languages: undefined,
                translations: undefined,
                nodes: undefined,
                start: undefined,
                ending: undefined,
                style: undefined,
            });
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("create (f0ea1aa5)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            name: "x",
            messages: undefined,
            languages: undefined,
            translations: undefined,
            nodes: undefined,
            start: undefined,
            ending: undefined,
            style: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/forms")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.forms.create({
                name: "x",
                messages: undefined,
                languages: undefined,
                translations: undefined,
                nodes: undefined,
                start: undefined,
                ending: undefined,
                style: undefined,
            });
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("create (e9650aed)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            name: "x",
            messages: undefined,
            languages: undefined,
            translations: undefined,
            nodes: undefined,
            start: undefined,
            ending: undefined,
            style: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/forms")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.forms.create({
                name: "x",
                messages: undefined,
                languages: undefined,
                translations: undefined,
                nodes: undefined,
                start: undefined,
                ending: undefined,
                style: undefined,
            });
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("get (f0b309fd)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = {
            id: "id",
            name: "name",
            messages: { errors: { key: "value" }, custom: { key: "value" } },
            languages: { primary: "primary", default: "default" },
            translations: { key: { key: "value" } },
            nodes: [
                { id: "id", type: "FLOW", coordinates: { x: 1, y: 1 }, alias: "alias", config: { flow_id: "flow_id" } },
            ],
            start: { hidden_fields: [{ key: "key" }], next_node: "next_node", coordinates: { x: 1, y: 1 } },
            ending: {
                redirection: { delay: 1, target: "target" },
                after_submit: { flow_id: "flow_id" },
                coordinates: { x: 1, y: 1 },
                resume_flow: true,
            },
            style: { css: "css" },
            created_at: "2024-01-15T09:30:00Z",
            updated_at: "2024-01-15T09:30:00Z",
            embedded_at: "embedded_at",
            submitted_at: "submitted_at",
        };
        server.mockEndpoint().get("/forms/id").respondWith().statusCode(200).jsonBody(rawResponseBody).build();
        const response = yield client.forms.get("id");
        expect(response).toEqual({
            id: "id",
            name: "name",
            messages: {
                errors: {
                    key: "value",
                },
                custom: {
                    key: "value",
                },
            },
            languages: {
                primary: "primary",
                default: "default",
            },
            translations: {
                key: {
                    key: "value",
                },
            },
            nodes: [
                {
                    id: "id",
                    type: "FLOW",
                    coordinates: {
                        x: 1,
                        y: 1,
                    },
                    alias: "alias",
                    config: {
                        flow_id: "flow_id",
                    },
                },
            ],
            start: {
                hidden_fields: [
                    {
                        key: "key",
                    },
                ],
                next_node: "next_node",
                coordinates: {
                    x: 1,
                    y: 1,
                },
            },
            ending: {
                redirection: {
                    delay: 1,
                    target: "target",
                },
                after_submit: {
                    flow_id: "flow_id",
                },
                coordinates: {
                    x: 1,
                    y: 1,
                },
                resume_flow: true,
            },
            style: {
                css: "css",
            },
            created_at: "2024-01-15T09:30:00Z",
            updated_at: "2024-01-15T09:30:00Z",
            embedded_at: "embedded_at",
            submitted_at: "submitted_at",
        });
    }));
    test("get (fcf9dbd1)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/forms/id").respondWith().statusCode(400).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.forms.get("id");
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("get (49d52691)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/forms/id").respondWith().statusCode(401).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.forms.get("id");
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("get (2428808d)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/forms/id").respondWith().statusCode(403).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.forms.get("id");
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("get (e55ce3fd)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/forms/id").respondWith().statusCode(404).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.forms.get("id");
        })).rejects.toThrow(new Management.NotFoundError({
            key: "value",
        }));
    }));
    test("get (27b44cb5)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/forms/id").respondWith().statusCode(429).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.forms.get("id");
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("delete (c7f0a6bf)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        server.mockEndpoint().delete("/forms/id").respondWith().statusCode(200).build();
        const response = yield client.forms.delete("id");
        expect(response).toEqual(undefined);
    }));
    test("delete (fcf9dbd1)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().delete("/forms/id").respondWith().statusCode(400).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.forms.delete("id");
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("delete (49d52691)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().delete("/forms/id").respondWith().statusCode(401).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.forms.delete("id");
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("delete (2428808d)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().delete("/forms/id").respondWith().statusCode(403).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.forms.delete("id");
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("delete (27b44cb5)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().delete("/forms/id").respondWith().statusCode(429).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.forms.delete("id");
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("update (e44eb82)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {};
        const rawResponseBody = {
            id: "id",
            name: "name",
            messages: { errors: { key: "value" }, custom: { key: "value" } },
            languages: { primary: "primary", default: "default" },
            translations: { key: { key: "value" } },
            nodes: [
                { id: "id", type: "FLOW", coordinates: { x: 1, y: 1 }, alias: "alias", config: { flow_id: "flow_id" } },
            ],
            start: { hidden_fields: [{ key: "key" }], next_node: "next_node", coordinates: { x: 1, y: 1 } },
            ending: {
                redirection: { delay: 1, target: "target" },
                after_submit: { flow_id: "flow_id" },
                coordinates: { x: 1, y: 1 },
                resume_flow: true,
            },
            style: { css: "css" },
            created_at: "2024-01-15T09:30:00Z",
            updated_at: "2024-01-15T09:30:00Z",
            embedded_at: "embedded_at",
            submitted_at: "submitted_at",
        };
        server
            .mockEndpoint()
            .patch("/forms/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.forms.update("id");
        expect(response).toEqual({
            id: "id",
            name: "name",
            messages: {
                errors: {
                    key: "value",
                },
                custom: {
                    key: "value",
                },
            },
            languages: {
                primary: "primary",
                default: "default",
            },
            translations: {
                key: {
                    key: "value",
                },
            },
            nodes: [
                {
                    id: "id",
                    type: "FLOW",
                    coordinates: {
                        x: 1,
                        y: 1,
                    },
                    alias: "alias",
                    config: {
                        flow_id: "flow_id",
                    },
                },
            ],
            start: {
                hidden_fields: [
                    {
                        key: "key",
                    },
                ],
                next_node: "next_node",
                coordinates: {
                    x: 1,
                    y: 1,
                },
            },
            ending: {
                redirection: {
                    delay: 1,
                    target: "target",
                },
                after_submit: {
                    flow_id: "flow_id",
                },
                coordinates: {
                    x: 1,
                    y: 1,
                },
                resume_flow: true,
            },
            style: {
                css: "css",
            },
            created_at: "2024-01-15T09:30:00Z",
            updated_at: "2024-01-15T09:30:00Z",
            embedded_at: "embedded_at",
            submitted_at: "submitted_at",
        });
    }));
    test("update (74a1ba4a)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            name: undefined,
            messages: undefined,
            languages: undefined,
            translations: undefined,
            nodes: undefined,
            start: undefined,
            ending: undefined,
            style: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/forms/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.forms.update("id", {
                name: undefined,
                messages: undefined,
                languages: undefined,
                translations: undefined,
                nodes: undefined,
                start: undefined,
                ending: undefined,
                style: undefined,
            });
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("update (ceee290e)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            name: undefined,
            messages: undefined,
            languages: undefined,
            translations: undefined,
            nodes: undefined,
            start: undefined,
            ending: undefined,
            style: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/forms/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.forms.update("id", {
                name: undefined,
                messages: undefined,
                languages: undefined,
                translations: undefined,
                nodes: undefined,
                start: undefined,
                ending: undefined,
                style: undefined,
            });
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("update (f75bec9a)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            name: undefined,
            messages: undefined,
            languages: undefined,
            translations: undefined,
            nodes: undefined,
            start: undefined,
            ending: undefined,
            style: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/forms/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.forms.update("id", {
                name: undefined,
                messages: undefined,
                languages: undefined,
                translations: undefined,
                nodes: undefined,
                start: undefined,
                ending: undefined,
                style: undefined,
            });
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
});
