// auth0/dist/cjs/management/tests/wire/hooks.test.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const MockServerPool_js_1 = require("../mock-server/MockServerPool.js");
const Client_js_1 = require("../../Client.js");
const Management = __importStar(require("../../api/index.js"));
describe("Hooks", () => {
    test("list (70af6fe5)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = {
            start: 1.1,
            limit: 1.1,
            total: 1.1,
            hooks: [
                {
                    triggerId: "triggerId",
                    id: "id",
                    name: "name",
                    enabled: true,
                    script: "script",
                    dependencies: { key: "value" },
                },
            ],
        };
        server.mockEndpoint().get("/hooks").respondWith().statusCode(200).jsonBody(rawResponseBody).build();
        const expected = {
            start: 1.1,
            limit: 1.1,
            total: 1.1,
            hooks: [
                {
                    triggerId: "triggerId",
                    id: "id",
                    name: "name",
                    enabled: true,
                    script: "script",
                    dependencies: {
                        key: "value",
                    },
                },
            ],
        };
        const page = yield client.hooks.list();
        expect(expected.hooks).toEqual(page.data);
        expect(page.hasNextPage()).toBe(true);
        const nextPage = yield page.getNextPage();
        expect(expected.hooks).toEqual(nextPage.data);
    }));
    test("list (c60dd33b)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/hooks").respondWith().statusCode(400).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.hooks.list();
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("list (1e230aeb)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/hooks").respondWith().statusCode(401).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.hooks.list();
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("list (af841397)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/hooks").respondWith().statusCode(403).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.hooks.list();
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("list (c29c5807)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/hooks").respondWith().statusCode(404).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.hooks.list();
        })).rejects.toThrow(new Management.NotFoundError({
            key: "value",
        }));
    }));
    test("list (ee1e23bf)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/hooks").respondWith().statusCode(429).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.hooks.list();
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("create (97c3e762)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { name: "name", script: "script", triggerId: "credentials-exchange" };
        const rawResponseBody = {
            triggerId: "triggerId",
            id: "id",
            name: "name",
            enabled: true,
            script: "script",
            dependencies: { key: "value" },
        };
        server
            .mockEndpoint()
            .post("/hooks")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.hooks.create({
            name: "name",
            script: "script",
            triggerId: "credentials-exchange",
        });
        expect(response).toEqual({
            triggerId: "triggerId",
            id: "id",
            name: "name",
            enabled: true,
            script: "script",
            dependencies: {
                key: "value",
            },
        });
    }));
    test("create (5ac04fde)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            name: "my-hook",
            script: "module.exports = function(client, scope, audience, context, cb) cb(null, access_token); };",
            enabled: undefined,
            dependencies: undefined,
            triggerId: "credentials-exchange",
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/hooks")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.hooks.create({
                name: "my-hook",
                script: "module.exports = function(client, scope, audience, context, cb) cb(null, access_token); };",
                enabled: undefined,
                dependencies: undefined,
                triggerId: "credentials-exchange",
            });
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("create (7de53de6)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            name: "my-hook",
            script: "module.exports = function(client, scope, audience, context, cb) cb(null, access_token); };",
            enabled: undefined,
            dependencies: undefined,
            triggerId: "credentials-exchange",
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/hooks")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.hooks.create({
                name: "my-hook",
                script: "module.exports = function(client, scope, audience, context, cb) cb(null, access_token); };",
                enabled: undefined,
                dependencies: undefined,
                triggerId: "credentials-exchange",
            });
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("create (40a063aa)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            name: "my-hook",
            script: "module.exports = function(client, scope, audience, context, cb) cb(null, access_token); };",
            enabled: undefined,
            dependencies: undefined,
            triggerId: "credentials-exchange",
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/hooks")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.hooks.create({
                name: "my-hook",
                script: "module.exports = function(client, scope, audience, context, cb) cb(null, access_token); };",
                enabled: undefined,
                dependencies: undefined,
                triggerId: "credentials-exchange",
            });
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("create (17150fc2)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            name: "my-hook",
            script: "module.exports = function(client, scope, audience, context, cb) cb(null, access_token); };",
            enabled: undefined,
            dependencies: undefined,
            triggerId: "credentials-exchange",
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/hooks")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(409)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.hooks.create({
                name: "my-hook",
                script: "module.exports = function(client, scope, audience, context, cb) cb(null, access_token); };",
                enabled: undefined,
                dependencies: undefined,
                triggerId: "credentials-exchange",
            });
        })).rejects.toThrow(new Management.ConflictError({
            key: "value",
        }));
    }));
    test("create (5ccab7c6)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            name: "my-hook",
            script: "module.exports = function(client, scope, audience, context, cb) cb(null, access_token); };",
            enabled: undefined,
            dependencies: undefined,
            triggerId: "credentials-exchange",
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/hooks")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.hooks.create({
                name: "my-hook",
                script: "module.exports = function(client, scope, audience, context, cb) cb(null, access_token); };",
                enabled: undefined,
                dependencies: undefined,
                triggerId: "credentials-exchange",
            });
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("get (ee316992)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = {
            triggerId: "triggerId",
            id: "id",
            name: "name",
            enabled: true,
            script: "script",
            dependencies: { key: "value" },
        };
        server.mockEndpoint().get("/hooks/id").respondWith().statusCode(200).jsonBody(rawResponseBody).build();
        const response = yield client.hooks.get("id");
        expect(response).toEqual({
            triggerId: "triggerId",
            id: "id",
            name: "name",
            enabled: true,
            script: "script",
            dependencies: {
                key: "value",
            },
        });
    }));
    test("get (fcf9dbd1)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/hooks/id").respondWith().statusCode(400).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.hooks.get("id");
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("get (49d52691)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/hooks/id").respondWith().statusCode(401).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.hooks.get("id");
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("get (2428808d)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/hooks/id").respondWith().statusCode(403).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.hooks.get("id");
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("get (e55ce3fd)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/hooks/id").respondWith().statusCode(404).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.hooks.get("id");
        })).rejects.toThrow(new Management.NotFoundError({
            key: "value",
        }));
    }));
    test("get (27b44cb5)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/hooks/id").respondWith().statusCode(429).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.hooks.get("id");
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("delete (c7f0a6bf)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        server.mockEndpoint().delete("/hooks/id").respondWith().statusCode(200).build();
        const response = yield client.hooks.delete("id");
        expect(response).toEqual(undefined);
    }));
    test("delete (fcf9dbd1)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().delete("/hooks/id").respondWith().statusCode(400).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.hooks.delete("id");
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("delete (49d52691)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().delete("/hooks/id").respondWith().statusCode(401).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.hooks.delete("id");
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("delete (2428808d)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().delete("/hooks/id").respondWith().statusCode(403).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.hooks.delete("id");
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("delete (27b44cb5)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().delete("/hooks/id").respondWith().statusCode(429).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.hooks.delete("id");
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("update (cd2be091)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {};
        const rawResponseBody = {
            triggerId: "triggerId",
            id: "id",
            name: "name",
            enabled: true,
            script: "script",
            dependencies: { key: "value" },
        };
        server
            .mockEndpoint()
            .patch("/hooks/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.hooks.update("id");
        expect(response).toEqual({
            triggerId: "triggerId",
            id: "id",
            name: "name",
            enabled: true,
            script: "script",
            dependencies: {
                key: "value",
            },
        });
    }));
    test("update (fdebfe7f)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { name: undefined, script: undefined, enabled: undefined, dependencies: undefined };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/hooks/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.hooks.update("id", {
                name: undefined,
                script: undefined,
                enabled: undefined,
                dependencies: undefined,
            });
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("update (359a0e0f)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { name: undefined, script: undefined, enabled: undefined, dependencies: undefined };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/hooks/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.hooks.update("id", {
                name: undefined,
                script: undefined,
                enabled: undefined,
                dependencies: undefined,
            });
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("update (c653e65b)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { name: undefined, script: undefined, enabled: undefined, dependencies: undefined };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/hooks/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.hooks.update("id", {
                name: undefined,
                script: undefined,
                enabled: undefined,
                dependencies: undefined,
            });
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("update (51ae21fb)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { name: undefined, script: undefined, enabled: undefined, dependencies: undefined };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/hooks/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(404)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.hooks.update("id", {
                name: undefined,
                script: undefined,
                enabled: undefined,
                dependencies: undefined,
            });
        })).rejects.toThrow(new Management.NotFoundError({
            key: "value",
        }));
    }));
    test("update (56a732d3)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { name: undefined, script: undefined, enabled: undefined, dependencies: undefined };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/hooks/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(409)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.hooks.update("id", {
                name: undefined,
                script: undefined,
                enabled: undefined,
                dependencies: undefined,
            });
        })).rejects.toThrow(new Management.ConflictError({
            key: "value",
        }));
    }));
    test("update (b6d13213)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { name: undefined, script: undefined, enabled: undefined, dependencies: undefined };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/hooks/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.hooks.update("id", {
                name: undefined,
                script: undefined,
                enabled: undefined,
                dependencies: undefined,
            });
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
});
