// auth0/dist/cjs/management/tests/wire/logStreams.test.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const MockServerPool_js_1 = require("../mock-server/MockServerPool.js");
const Client_js_1 = require("../../Client.js");
const Management = __importStar(require("../../api/index.js"));
describe("LogStreams", () => {
    test("list (45fcfa8b)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = [
            {
                id: "id",
                name: "name",
                status: "active",
                type: "http",
                isPriority: true,
                filters: [{}],
                pii_config: { log_fields: ["first_name"], method: "mask", algorithm: "xxhash" },
                sink: {
                    httpAuthorization: "httpAuthorization",
                    httpContentFormat: "JSONARRAY",
                    httpContentType: "httpContentType",
                    httpEndpoint: "httpEndpoint",
                    httpCustomHeaders: [{}],
                },
            },
        ];
        server.mockEndpoint().get("/log-streams").respondWith().statusCode(200).jsonBody(rawResponseBody).build();
        const response = yield client.logStreams.list();
        expect(response).toEqual([
            {
                id: "id",
                name: "name",
                status: "active",
                type: "http",
                isPriority: true,
                filters: [{}],
                pii_config: {
                    log_fields: ["first_name"],
                    method: "mask",
                    algorithm: "xxhash",
                },
                sink: {
                    httpAuthorization: "httpAuthorization",
                    httpContentFormat: "JSONARRAY",
                    httpContentType: "httpContentType",
                    httpEndpoint: "httpEndpoint",
                    httpCustomHeaders: [{}],
                },
            },
        ]);
    }));
    test("list (1e230aeb)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/log-streams").respondWith().statusCode(401).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.logStreams.list();
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("list (af841397)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/log-streams").respondWith().statusCode(403).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.logStreams.list();
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("list (c29c5807)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/log-streams").respondWith().statusCode(404).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.logStreams.list();
        })).rejects.toThrow(new Management.NotFoundError({
            key: "value",
        }));
    }));
    test("list (ee1e23bf)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/log-streams").respondWith().statusCode(429).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.logStreams.list();
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("create (bbc0ceb5)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { type: "http", sink: { httpEndpoint: "httpEndpoint" } };
        const rawResponseBody = {
            id: "id",
            name: "name",
            status: "active",
            type: "http",
            isPriority: true,
            filters: [{ type: "category", name: "auth.login.fail" }],
            pii_config: { log_fields: ["first_name"], method: "mask", algorithm: "xxhash" },
            sink: {
                httpAuthorization: "httpAuthorization",
                httpContentFormat: "JSONARRAY",
                httpContentType: "httpContentType",
                httpEndpoint: "httpEndpoint",
                httpCustomHeaders: [{}],
            },
        };
        server
            .mockEndpoint()
            .post("/log-streams")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.logStreams.create({
            type: "http",
            sink: {
                httpEndpoint: "httpEndpoint",
            },
        });
        expect(response).toEqual({
            id: "id",
            name: "name",
            status: "active",
            type: "http",
            isPriority: true,
            filters: [
                {
                    type: "category",
                    name: "auth.login.fail",
                },
            ],
            pii_config: {
                log_fields: ["first_name"],
                method: "mask",
                algorithm: "xxhash",
            },
            sink: {
                httpAuthorization: "httpAuthorization",
                httpContentFormat: "JSONARRAY",
                httpContentType: "httpContentType",
                httpEndpoint: "httpEndpoint",
                httpCustomHeaders: [{}],
            },
        });
    }));
    test("create (dac8f575)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            name: undefined,
            type: "http",
            isPriority: undefined,
            filters: undefined,
            pii_config: undefined,
            sink: {
                httpAuthorization: undefined,
                httpContentFormat: undefined,
                httpContentType: undefined,
                httpEndpoint: "httpEndpoint",
                httpCustomHeaders: undefined,
            },
            startFrom: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/log-streams")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.logStreams.create({
                name: undefined,
                type: "http",
                isPriority: undefined,
                filters: undefined,
                pii_config: undefined,
                sink: {
                    httpAuthorization: undefined,
                    httpContentFormat: undefined,
                    httpContentType: undefined,
                    httpEndpoint: "httpEndpoint",
                    httpCustomHeaders: undefined,
                },
                startFrom: undefined,
            });
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("create (14865)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            name: undefined,
            type: "http",
            isPriority: undefined,
            filters: undefined,
            pii_config: undefined,
            sink: {
                httpAuthorization: undefined,
                httpContentFormat: undefined,
                httpContentType: undefined,
                httpEndpoint: "httpEndpoint",
                httpCustomHeaders: undefined,
            },
            startFrom: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/log-streams")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.logStreams.create({
                name: undefined,
                type: "http",
                isPriority: undefined,
                filters: undefined,
                pii_config: undefined,
                sink: {
                    httpAuthorization: undefined,
                    httpContentFormat: undefined,
                    httpContentType: undefined,
                    httpEndpoint: "httpEndpoint",
                    httpCustomHeaders: undefined,
                },
                startFrom: undefined,
            });
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("create (608d6031)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            name: undefined,
            type: "http",
            isPriority: undefined,
            filters: undefined,
            pii_config: undefined,
            sink: {
                httpAuthorization: undefined,
                httpContentFormat: undefined,
                httpContentType: undefined,
                httpEndpoint: "httpEndpoint",
                httpCustomHeaders: undefined,
            },
            startFrom: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/log-streams")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.logStreams.create({
                name: undefined,
                type: "http",
                isPriority: undefined,
                filters: undefined,
                pii_config: undefined,
                sink: {
                    httpAuthorization: undefined,
                    httpContentFormat: undefined,
                    httpContentType: undefined,
                    httpEndpoint: "httpEndpoint",
                    httpCustomHeaders: undefined,
                },
                startFrom: undefined,
            });
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("create (5750f049)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            name: undefined,
            type: "http",
            isPriority: undefined,
            filters: undefined,
            pii_config: undefined,
            sink: {
                httpAuthorization: undefined,
                httpContentFormat: undefined,
                httpContentType: undefined,
                httpEndpoint: "httpEndpoint",
                httpCustomHeaders: undefined,
            },
            startFrom: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/log-streams")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(409)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.logStreams.create({
                name: undefined,
                type: "http",
                isPriority: undefined,
                filters: undefined,
                pii_config: undefined,
                sink: {
                    httpAuthorization: undefined,
                    httpContentFormat: undefined,
                    httpContentType: undefined,
                    httpEndpoint: "httpEndpoint",
                    httpCustomHeaders: undefined,
                },
                startFrom: undefined,
            });
        })).rejects.toThrow(new Management.ConflictError({
            key: "value",
        }));
    }));
    test("create (6f8e3989)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            name: undefined,
            type: "http",
            isPriority: undefined,
            filters: undefined,
            pii_config: undefined,
            sink: {
                httpAuthorization: undefined,
                httpContentFormat: undefined,
                httpContentType: undefined,
                httpEndpoint: "httpEndpoint",
                httpCustomHeaders: undefined,
            },
            startFrom: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/log-streams")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.logStreams.create({
                name: undefined,
                type: "http",
                isPriority: undefined,
                filters: undefined,
                pii_config: undefined,
                sink: {
                    httpAuthorization: undefined,
                    httpContentFormat: undefined,
                    httpContentType: undefined,
                    httpEndpoint: "httpEndpoint",
                    httpCustomHeaders: undefined,
                },
                startFrom: undefined,
            });
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("get (e9ced4ed)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = {
            id: "id",
            name: "name",
            status: "active",
            type: "http",
            isPriority: true,
            filters: [{ type: "category", name: "auth.login.fail" }],
            pii_config: { log_fields: ["first_name"], method: "mask", algorithm: "xxhash" },
            sink: {
                httpAuthorization: "httpAuthorization",
                httpContentFormat: "JSONARRAY",
                httpContentType: "httpContentType",
                httpEndpoint: "httpEndpoint",
                httpCustomHeaders: [{}],
            },
        };
        server.mockEndpoint().get("/log-streams/id").respondWith().statusCode(200).jsonBody(rawResponseBody).build();
        const response = yield client.logStreams.get("id");
        expect(response).toEqual({
            id: "id",
            name: "name",
            status: "active",
            type: "http",
            isPriority: true,
            filters: [
                {
                    type: "category",
                    name: "auth.login.fail",
                },
            ],
            pii_config: {
                log_fields: ["first_name"],
                method: "mask",
                algorithm: "xxhash",
            },
            sink: {
                httpAuthorization: "httpAuthorization",
                httpContentFormat: "JSONARRAY",
                httpContentType: "httpContentType",
                httpEndpoint: "httpEndpoint",
                httpCustomHeaders: [{}],
            },
        });
    }));
    test("get (49d52691)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/log-streams/id").respondWith().statusCode(401).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.logStreams.get("id");
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("get (2428808d)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/log-streams/id").respondWith().statusCode(403).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.logStreams.get("id");
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("get (e55ce3fd)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/log-streams/id").respondWith().statusCode(404).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.logStreams.get("id");
        })).rejects.toThrow(new Management.NotFoundError({
            key: "value",
        }));
    }));
    test("get (27b44cb5)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/log-streams/id").respondWith().statusCode(429).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.logStreams.get("id");
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("delete (c7f0a6bf)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        server.mockEndpoint().delete("/log-streams/id").respondWith().statusCode(200).build();
        const response = yield client.logStreams.delete("id");
        expect(response).toEqual(undefined);
    }));
    test("delete (fcf9dbd1)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().delete("/log-streams/id").respondWith().statusCode(400).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.logStreams.delete("id");
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("delete (49d52691)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().delete("/log-streams/id").respondWith().statusCode(401).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.logStreams.delete("id");
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("delete (2428808d)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().delete("/log-streams/id").respondWith().statusCode(403).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.logStreams.delete("id");
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("delete (e55ce3fd)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().delete("/log-streams/id").respondWith().statusCode(404).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.logStreams.delete("id");
        })).rejects.toThrow(new Management.NotFoundError({
            key: "value",
        }));
    }));
    test("delete (27b44cb5)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().delete("/log-streams/id").respondWith().statusCode(429).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.logStreams.delete("id");
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("update (3653938)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {};
        const rawResponseBody = {
            id: "id",
            name: "name",
            status: "active",
            type: "http",
            isPriority: true,
            filters: [{ type: "category", name: "auth.login.fail" }],
            pii_config: { log_fields: ["first_name"], method: "mask", algorithm: "xxhash" },
            sink: {
                httpAuthorization: "httpAuthorization",
                httpContentFormat: "JSONARRAY",
                httpContentType: "httpContentType",
                httpEndpoint: "httpEndpoint",
                httpCustomHeaders: [{}],
            },
        };
        server
            .mockEndpoint()
            .patch("/log-streams/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.logStreams.update("id");
        expect(response).toEqual({
            id: "id",
            name: "name",
            status: "active",
            type: "http",
            isPriority: true,
            filters: [
                {
                    type: "category",
                    name: "auth.login.fail",
                },
            ],
            pii_config: {
                log_fields: ["first_name"],
                method: "mask",
                algorithm: "xxhash",
            },
            sink: {
                httpAuthorization: "httpAuthorization",
                httpContentFormat: "JSONARRAY",
                httpContentType: "httpContentType",
                httpEndpoint: "httpEndpoint",
                httpCustomHeaders: [{}],
            },
        });
    }));
    test("update (36a78731)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            name: undefined,
            status: undefined,
            isPriority: undefined,
            filters: undefined,
            pii_config: undefined,
            sink: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/log-streams/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.logStreams.update("id", {
                name: undefined,
                status: undefined,
                isPriority: undefined,
                filters: undefined,
                pii_config: undefined,
                sink: undefined,
            });
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("update (20103571)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            name: undefined,
            status: undefined,
            isPriority: undefined,
            filters: undefined,
            pii_config: undefined,
            sink: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/log-streams/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.logStreams.update("id", {
                name: undefined,
                status: undefined,
                isPriority: undefined,
                filters: undefined,
                pii_config: undefined,
                sink: undefined,
            });
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("update (bc05b36d)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            name: undefined,
            status: undefined,
            isPriority: undefined,
            filters: undefined,
            pii_config: undefined,
            sink: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/log-streams/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.logStreams.update("id", {
                name: undefined,
                status: undefined,
                isPriority: undefined,
                filters: undefined,
                pii_config: undefined,
                sink: undefined,
            });
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("update (78a35815)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            name: undefined,
            status: undefined,
            isPriority: undefined,
            filters: undefined,
            pii_config: undefined,
            sink: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/log-streams/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.logStreams.update("id", {
                name: undefined,
                status: undefined,
                isPriority: undefined,
                filters: undefined,
                pii_config: undefined,
                sink: undefined,
            });
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
});
