// auth0/dist/cjs/management/tests/wire/networkAcls.test.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const MockServerPool_js_1 = require("../mock-server/MockServerPool.js");
const Client_js_1 = require("../../Client.js");
const Management = __importStar(require("../../api/index.js"));
describe("NetworkAcls", () => {
    test("list (f9c0d692)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = {
            network_acls: [
                {
                    id: "id",
                    description: "description",
                    active: true,
                    priority: 1.1,
                    rule: { action: {}, scope: "management" },
                    created_at: "created_at",
                    updated_at: "updated_at",
                },
            ],
            start: 1.1,
            limit: 1.1,
            total: 1.1,
        };
        server.mockEndpoint().get("/network-acls").respondWith().statusCode(200).jsonBody(rawResponseBody).build();
        const expected = {
            network_acls: [
                {
                    id: "id",
                    description: "description",
                    active: true,
                    priority: 1.1,
                    rule: {
                        action: {},
                        scope: "management",
                    },
                    created_at: "created_at",
                    updated_at: "updated_at",
                },
            ],
            start: 1.1,
            limit: 1.1,
            total: 1.1,
        };
        const page = yield client.networkAcls.list();
        expect(expected.network_acls).toEqual(page.data);
        expect(page.hasNextPage()).toBe(true);
        const nextPage = yield page.getNextPage();
        expect(expected.network_acls).toEqual(nextPage.data);
    }));
    test("list (1e230aeb)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/network-acls").respondWith().statusCode(401).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.networkAcls.list();
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("list (af841397)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/network-acls").respondWith().statusCode(403).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.networkAcls.list();
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("list (c29c5807)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/network-acls").respondWith().statusCode(404).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.networkAcls.list();
        })).rejects.toThrow(new Management.NotFoundError({
            key: "value",
        }));
    }));
    test("list (ee1e23bf)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/network-acls").respondWith().statusCode(429).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.networkAcls.list();
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("create (ea903a5f)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            description: "description",
            active: true,
            priority: 1.1,
            rule: { action: {}, scope: "management" },
        };
        server.mockEndpoint().post("/network-acls").jsonBody(rawRequestBody).respondWith().statusCode(200).build();
        const response = yield client.networkAcls.create({
            description: "description",
            active: true,
            priority: 1.1,
            rule: {
                action: {},
                scope: "management",
            },
        });
        expect(response).toEqual(undefined);
    }));
    test("create (c2e283a4)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            description: "description",
            active: true,
            priority: 1.1,
            rule: {
                action: {
                    block: undefined,
                    allow: undefined,
                    log: undefined,
                    redirect: undefined,
                    redirect_uri: undefined,
                },
                match: undefined,
                not_match: undefined,
                scope: "management",
            },
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/network-acls")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.networkAcls.create({
                description: "description",
                active: true,
                priority: 1.1,
                rule: {
                    action: {
                        block: undefined,
                        allow: undefined,
                        log: undefined,
                        redirect: undefined,
                        redirect_uri: undefined,
                    },
                    match: undefined,
                    not_match: undefined,
                    scope: "management",
                },
            });
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("create (ee0ff0cc)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            description: "description",
            active: true,
            priority: 1.1,
            rule: {
                action: {
                    block: undefined,
                    allow: undefined,
                    log: undefined,
                    redirect: undefined,
                    redirect_uri: undefined,
                },
                match: undefined,
                not_match: undefined,
                scope: "management",
            },
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/network-acls")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.networkAcls.create({
                description: "description",
                active: true,
                priority: 1.1,
                rule: {
                    action: {
                        block: undefined,
                        allow: undefined,
                        log: undefined,
                        redirect: undefined,
                        redirect_uri: undefined,
                    },
                    match: undefined,
                    not_match: undefined,
                    scope: "management",
                },
            });
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("create (1c23e3d0)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            description: "description",
            active: true,
            priority: 1.1,
            rule: {
                action: {
                    block: undefined,
                    allow: undefined,
                    log: undefined,
                    redirect: undefined,
                    redirect_uri: undefined,
                },
                match: undefined,
                not_match: undefined,
                scope: "management",
            },
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/network-acls")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.networkAcls.create({
                description: "description",
                active: true,
                priority: 1.1,
                rule: {
                    action: {
                        block: undefined,
                        allow: undefined,
                        log: undefined,
                        redirect: undefined,
                        redirect_uri: undefined,
                    },
                    match: undefined,
                    not_match: undefined,
                    scope: "management",
                },
            });
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("create (e59936e8)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            description: "description",
            active: true,
            priority: 1.1,
            rule: {
                action: {
                    block: undefined,
                    allow: undefined,
                    log: undefined,
                    redirect: undefined,
                    redirect_uri: undefined,
                },
                match: undefined,
                not_match: undefined,
                scope: "management",
            },
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/network-acls")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(409)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.networkAcls.create({
                description: "description",
                active: true,
                priority: 1.1,
                rule: {
                    action: {
                        block: undefined,
                        allow: undefined,
                        log: undefined,
                        redirect: undefined,
                        redirect_uri: undefined,
                    },
                    match: undefined,
                    not_match: undefined,
                    scope: "management",
                },
            });
        })).rejects.toThrow(new Management.ConflictError({
            key: "value",
        }));
    }));
    test("create (5d4241dc)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            description: "description",
            active: true,
            priority: 1.1,
            rule: {
                action: {
                    block: undefined,
                    allow: undefined,
                    log: undefined,
                    redirect: undefined,
                    redirect_uri: undefined,
                },
                match: undefined,
                not_match: undefined,
                scope: "management",
            },
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/network-acls")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.networkAcls.create({
                description: "description",
                active: true,
                priority: 1.1,
                rule: {
                    action: {
                        block: undefined,
                        allow: undefined,
                        log: undefined,
                        redirect: undefined,
                        redirect_uri: undefined,
                    },
                    match: undefined,
                    not_match: undefined,
                    scope: "management",
                },
            });
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("create (27b83a80)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            description: "description",
            active: true,
            priority: 1.1,
            rule: {
                action: {
                    block: undefined,
                    allow: undefined,
                    log: undefined,
                    redirect: undefined,
                    redirect_uri: undefined,
                },
                match: undefined,
                not_match: undefined,
                scope: "management",
            },
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/network-acls")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(500)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.networkAcls.create({
                description: "description",
                active: true,
                priority: 1.1,
                rule: {
                    action: {
                        block: undefined,
                        allow: undefined,
                        log: undefined,
                        redirect: undefined,
                        redirect_uri: undefined,
                    },
                    match: undefined,
                    not_match: undefined,
                    scope: "management",
                },
            });
        })).rejects.toThrow(new Management.InternalServerError({
            key: "value",
        }));
    }));
    test("get (f3f30785)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = {
            id: "id",
            description: "description",
            active: true,
            priority: 1.1,
            rule: {
                action: { block: true, allow: true, log: true, redirect: true, redirect_uri: "redirect_uri" },
                match: {
                    asns: [1],
                    geo_country_codes: ["geo_country_codes"],
                    geo_subdivision_codes: ["geo_subdivision_codes"],
                    ipv4_cidrs: ["ipv4_cidrs"],
                    ipv6_cidrs: ["ipv6_cidrs"],
                    ja3_fingerprints: ["ja3_fingerprints"],
                    ja4_fingerprints: ["ja4_fingerprints"],
                    user_agents: ["user_agents"],
                },
                not_match: {
                    asns: [1],
                    geo_country_codes: ["geo_country_codes"],
                    geo_subdivision_codes: ["geo_subdivision_codes"],
                    ipv4_cidrs: ["ipv4_cidrs"],
                    ipv6_cidrs: ["ipv6_cidrs"],
                    ja3_fingerprints: ["ja3_fingerprints"],
                    ja4_fingerprints: ["ja4_fingerprints"],
                    user_agents: ["user_agents"],
                },
                scope: "management",
            },
            created_at: "created_at",
            updated_at: "updated_at",
        };
        server.mockEndpoint().get("/network-acls/id").respondWith().statusCode(200).jsonBody(rawResponseBody).build();
        const response = yield client.networkAcls.get("id");
        expect(response).toEqual({
            id: "id",
            description: "description",
            active: true,
            priority: 1.1,
            rule: {
                action: {
                    block: true,
                    allow: true,
                    log: true,
                    redirect: true,
                    redirect_uri: "redirect_uri",
                },
                match: {
                    asns: [1],
                    geo_country_codes: ["geo_country_codes"],
                    geo_subdivision_codes: ["geo_subdivision_codes"],
                    ipv4_cidrs: ["ipv4_cidrs"],
                    ipv6_cidrs: ["ipv6_cidrs"],
                    ja3_fingerprints: ["ja3_fingerprints"],
                    ja4_fingerprints: ["ja4_fingerprints"],
                    user_agents: ["user_agents"],
                },
                not_match: {
                    asns: [1],
                    geo_country_codes: ["geo_country_codes"],
                    geo_subdivision_codes: ["geo_subdivision_codes"],
                    ipv4_cidrs: ["ipv4_cidrs"],
                    ipv6_cidrs: ["ipv6_cidrs"],
                    ja3_fingerprints: ["ja3_fingerprints"],
                    ja4_fingerprints: ["ja4_fingerprints"],
                    user_agents: ["user_agents"],
                },
                scope: "management",
            },
            created_at: "created_at",
            updated_at: "updated_at",
        });
    }));
    test("get (49d52691)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/network-acls/id").respondWith().statusCode(401).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.networkAcls.get("id");
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("get (2428808d)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/network-acls/id").respondWith().statusCode(403).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.networkAcls.get("id");
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("get (e55ce3fd)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/network-acls/id").respondWith().statusCode(404).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.networkAcls.get("id");
        })).rejects.toThrow(new Management.NotFoundError({
            key: "value",
        }));
    }));
    test("get (27b44cb5)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server.mockEndpoint().get("/network-acls/id").respondWith().statusCode(429).jsonBody(rawResponseBody).build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.networkAcls.get("id");
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("set (1acfd4fb)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            description: "description",
            active: true,
            priority: 1.1,
            rule: { action: {}, scope: "management" },
        };
        const rawResponseBody = {
            id: "id",
            description: "description",
            active: true,
            priority: 1.1,
            rule: {
                action: { block: true, allow: true, log: true, redirect: true, redirect_uri: "redirect_uri" },
                match: {
                    asns: [1],
                    geo_country_codes: ["geo_country_codes"],
                    geo_subdivision_codes: ["geo_subdivision_codes"],
                    ipv4_cidrs: ["ipv4_cidrs"],
                    ipv6_cidrs: ["ipv6_cidrs"],
                    ja3_fingerprints: ["ja3_fingerprints"],
                    ja4_fingerprints: ["ja4_fingerprints"],
                    user_agents: ["user_agents"],
                },
                not_match: {
                    asns: [1],
                    geo_country_codes: ["geo_country_codes"],
                    geo_subdivision_codes: ["geo_subdivision_codes"],
                    ipv4_cidrs: ["ipv4_cidrs"],
                    ipv6_cidrs: ["ipv6_cidrs"],
                    ja3_fingerprints: ["ja3_fingerprints"],
                    ja4_fingerprints: ["ja4_fingerprints"],
                    user_agents: ["user_agents"],
                },
                scope: "management",
            },
            created_at: "created_at",
            updated_at: "updated_at",
        };
        server
            .mockEndpoint()
            .put("/network-acls/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.networkAcls.set("id", {
            description: "description",
            active: true,
            priority: 1.1,
            rule: {
                action: {},
                scope: "management",
            },
        });
        expect(response).toEqual({
            id: "id",
            description: "description",
            active: true,
            priority: 1.1,
            rule: {
                action: {
                    block: true,
                    allow: true,
                    log: true,
                    redirect: true,
                    redirect_uri: "redirect_uri",
                },
                match: {
                    asns: [1],
                    geo_country_codes: ["geo_country_codes"],
                    geo_subdivision_codes: ["geo_subdivision_codes"],
                    ipv4_cidrs: ["ipv4_cidrs"],
                    ipv6_cidrs: ["ipv6_cidrs"],
                    ja3_fingerprints: ["ja3_fingerprints"],
                    ja4_fingerprints: ["ja4_fingerprints"],
                    user_agents: ["user_agents"],
                },
                not_match: {
                    asns: [1],
                    geo_country_codes: ["geo_country_codes"],
                    geo_subdivision_codes: ["geo_subdivision_codes"],
                    ipv4_cidrs: ["ipv4_cidrs"],
                    ipv6_cidrs: ["ipv6_cidrs"],
                    ja3_fingerprints: ["ja3_fingerprints"],
                    ja4_fingerprints: ["ja4_fingerprints"],
                    user_agents: ["user_agents"],
                },
                scope: "management",
            },
            created_at: "created_at",
            updated_at: "updated_at",
        });
    }));
    test("set (17f2ed2a)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            description: "description",
            active: true,
            priority: 1.1,
            rule: {
                action: {
                    block: undefined,
                    allow: undefined,
                    log: undefined,
                    redirect: undefined,
                    redirect_uri: undefined,
                },
                match: undefined,
                not_match: undefined,
                scope: "management",
            },
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .put("/network-acls/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.networkAcls.set("id", {
                description: "description",
                active: true,
                priority: 1.1,
                rule: {
                    action: {
                        block: undefined,
                        allow: undefined,
                        log: undefined,
                        redirect: undefined,
                        redirect_uri: undefined,
                    },
                    match: undefined,
                    not_match: undefined,
                    scope: "management",
                },
            });
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("set (299f5a82)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            description: "description",
            active: true,
            priority: 1.1,
            rule: {
                action: {
                    block: undefined,
                    allow: undefined,
                    log: undefined,
                    redirect: undefined,
                    redirect_uri: undefined,
                },
                match: undefined,
                not_match: undefined,
                scope: "management",
            },
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .put("/network-acls/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.networkAcls.set("id", {
                description: "description",
                active: true,
                priority: 1.1,
                rule: {
                    action: {
                        block: undefined,
                        allow: undefined,
                        log: undefined,
                        redirect: undefined,
                        redirect_uri: undefined,
                    },
                    match: undefined,
                    not_match: undefined,
                    scope: "management",
                },
            });
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("set (57a63466)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            description: "description",
            active: true,
            priority: 1.1,
            rule: {
                action: {
                    block: undefined,
                    allow: undefined,
                    log: undefined,
                    redirect: undefined,
                    redirect_uri: undefined,
                },
                match: undefined,
                not_match: undefined,
                scope: "management",
            },
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .put("/network-acls/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.networkAcls.set("id", {
                description: "description",
                active: true,
                priority: 1.1,
                rule: {
                    action: {
                        block: undefined,
                        allow: undefined,
                        log: undefined,
                        redirect: undefined,
                        redirect_uri: undefined,
                    },
                    match: undefined,
                    not_match: undefined,
                    scope: "management",
                },
            });
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("set (313cc19e)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            description: "description",
            active: true,
            priority: 1.1,
            rule: {
                action: {
                    block: undefined,
                    allow: undefined,
                    log: undefined,
                    redirect: undefined,
                    redirect_uri: undefined,
                },
                match: undefined,
                not_match: undefined,
                scope: "management",
            },
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .put("/network-acls/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(404)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.networkAcls.set("id", {
                description: "description",
                active: true,
                priority: 1.1,
                rule: {
                    action: {
                        block: undefined,
                        allow: undefined,
                        log: undefined,
                        redirect: undefined,
                        redirect_uri: undefined,
                    },
                    match: undefined,
                    not_match: undefined,
                    scope: "management",
                },
            });
        })).rejects.toThrow(new Management.NotFoundError({
            key: "value",
        }));
    }));
    test("set (1f543e72)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            description: "description",
            active: true,
            priority: 1.1,
            rule: {
                action: {
                    block: undefined,
                    allow: undefined,
                    log: undefined,
                    redirect: undefined,
                    redirect_uri: undefined,
                },
                match: undefined,
                not_match: undefined,
                scope: "management",
            },
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .put("/network-acls/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.networkAcls.set("id", {
                description: "description",
                active: true,
                priority: 1.1,
                rule: {
                    action: {
                        block: undefined,
                        allow: undefined,
                        log: undefined,
                        redirect: undefined,
                        redirect_uri: undefined,
                    },
                    match: undefined,
                    not_match: undefined,
                    scope: "management",
                },
            });
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("delete (c7f0a6bf)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        server.mockEndpoint().delete("/network-acls/id").respondWith().statusCode(200).build();
        const response = yield client.networkAcls.delete("id");
        expect(response).toEqual(undefined);
    }));
    test("delete (fcf9dbd1)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/network-acls/id")
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.networkAcls.delete("id");
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("delete (49d52691)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/network-acls/id")
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.networkAcls.delete("id");
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("delete (2428808d)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/network-acls/id")
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.networkAcls.delete("id");
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("delete (e55ce3fd)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/network-acls/id")
            .respondWith()
            .statusCode(404)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.networkAcls.delete("id");
        })).rejects.toThrow(new Management.NotFoundError({
            key: "value",
        }));
    }));
    test("delete (27b44cb5)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/network-acls/id")
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.networkAcls.delete("id");
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("update (bb550618)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {};
        const rawResponseBody = {
            id: "id",
            description: "description",
            active: true,
            priority: 1.1,
            rule: {
                action: { block: true, allow: true, log: true, redirect: true, redirect_uri: "redirect_uri" },
                match: {
                    asns: [1],
                    geo_country_codes: ["geo_country_codes"],
                    geo_subdivision_codes: ["geo_subdivision_codes"],
                    ipv4_cidrs: ["ipv4_cidrs"],
                    ipv6_cidrs: ["ipv6_cidrs"],
                    ja3_fingerprints: ["ja3_fingerprints"],
                    ja4_fingerprints: ["ja4_fingerprints"],
                    user_agents: ["user_agents"],
                },
                not_match: {
                    asns: [1],
                    geo_country_codes: ["geo_country_codes"],
                    geo_subdivision_codes: ["geo_subdivision_codes"],
                    ipv4_cidrs: ["ipv4_cidrs"],
                    ipv6_cidrs: ["ipv6_cidrs"],
                    ja3_fingerprints: ["ja3_fingerprints"],
                    ja4_fingerprints: ["ja4_fingerprints"],
                    user_agents: ["user_agents"],
                },
                scope: "management",
            },
            created_at: "created_at",
            updated_at: "updated_at",
        };
        server
            .mockEndpoint()
            .patch("/network-acls/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.networkAcls.update("id");
        expect(response).toEqual({
            id: "id",
            description: "description",
            active: true,
            priority: 1.1,
            rule: {
                action: {
                    block: true,
                    allow: true,
                    log: true,
                    redirect: true,
                    redirect_uri: "redirect_uri",
                },
                match: {
                    asns: [1],
                    geo_country_codes: ["geo_country_codes"],
                    geo_subdivision_codes: ["geo_subdivision_codes"],
                    ipv4_cidrs: ["ipv4_cidrs"],
                    ipv6_cidrs: ["ipv6_cidrs"],
                    ja3_fingerprints: ["ja3_fingerprints"],
                    ja4_fingerprints: ["ja4_fingerprints"],
                    user_agents: ["user_agents"],
                },
                not_match: {
                    asns: [1],
                    geo_country_codes: ["geo_country_codes"],
                    geo_subdivision_codes: ["geo_subdivision_codes"],
                    ipv4_cidrs: ["ipv4_cidrs"],
                    ipv6_cidrs: ["ipv6_cidrs"],
                    ja3_fingerprints: ["ja3_fingerprints"],
                    ja4_fingerprints: ["ja4_fingerprints"],
                    user_agents: ["user_agents"],
                },
                scope: "management",
            },
            created_at: "created_at",
            updated_at: "updated_at",
        });
    }));
    test("update (1f4f3271)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { description: undefined, active: undefined, priority: undefined, rule: undefined };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/network-acls/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.networkAcls.update("id", {
                description: undefined,
                active: undefined,
                priority: undefined,
                rule: undefined,
            });
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("update (4db2a2b1)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { description: undefined, active: undefined, priority: undefined, rule: undefined };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/network-acls/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.networkAcls.update("id", {
                description: undefined,
                active: undefined,
                priority: undefined,
                rule: undefined,
            });
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("update (fa32ecad)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { description: undefined, active: undefined, priority: undefined, rule: undefined };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/network-acls/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.networkAcls.update("id", {
                description: undefined,
                active: undefined,
                priority: undefined,
                rule: undefined,
            });
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("update (fa01531d)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { description: undefined, active: undefined, priority: undefined, rule: undefined };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/network-acls/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(404)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.networkAcls.update("id", {
                description: undefined,
                active: undefined,
                priority: undefined,
                rule: undefined,
            });
        })).rejects.toThrow(new Management.NotFoundError({
            key: "value",
        }));
    }));
    test("update (18eac855)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { description: undefined, active: undefined, priority: undefined, rule: undefined };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/network-acls/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.networkAcls.update("id", {
                description: undefined,
                active: undefined,
                priority: undefined,
                rule: undefined,
            });
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
});
