// auth0/dist/cjs/management/tests/mock-server/MockServerPool.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.mockServerPool = void 0;
const node_1 = require("msw/node");
const json_js_1 = require("../../core/json.js");
const MockServer_js_1 = require("./MockServer.js");
const randomBaseUrl_js_1 = require("./randomBaseUrl.js");
const mswServer = (0, node_1.setupServer)();
function formatHttpRequest(request, id) {
    return __awaiter(this, void 0, void 0, function* () {
        try {
            const clone = request.clone();
            const headers = [...clone.headers.entries()].map(([k, v]) => `${k}: ${v}`).join("\n");
            let body = "";
            try {
                const contentType = clone.headers.get("content-type");
                if (contentType === null || contentType === void 0 ? void 0 : contentType.includes("application/json")) {
                    body = (0, json_js_1.toJson)((0, json_js_1.fromJson)(yield clone.text()), undefined, 2);
                }
                else if (clone.body) {
                    body = yield clone.text();
                }
            }
            catch (e) {
                body = "(unable to parse body)";
            }
            const title = id ? `### Request ${id} ###\n` : "";
            const firstLine = `${title}${request.method} ${request.url.toString()} HTTP/1.1`;
            return `\n${firstLine}\n${headers}\n\n${body || "(no body)"}\n`;
        }
        catch (e) {
            return `Error formatting request: ${e}`;
        }
    });
}
function formatHttpResponse(response, id) {
    return __awaiter(this, void 0, void 0, function* () {
        try {
            const clone = response.clone();
            const headers = [...clone.headers.entries()].map(([k, v]) => `${k}: ${v}`).join("\n");
            let body = "";
            try {
                const contentType = clone.headers.get("content-type");
                if (contentType === null || contentType === void 0 ? void 0 : contentType.includes("application/json")) {
                    body = (0, json_js_1.toJson)((0, json_js_1.fromJson)(yield clone.text()), undefined, 2);
                }
                else if (clone.body) {
                    body = yield clone.text();
                }
            }
            catch (e) {
                body = "(unable to parse body)";
            }
            const title = id ? `### Response for ${id} ###\n` : "";
            const firstLine = `${title}HTTP/1.1 ${response.status} ${response.statusText}`;
            return `\n${firstLine}\n${headers}\n\n${body || "(no body)"}\n`;
        }
        catch (e) {
            return `Error formatting response: ${e}`;
        }
    });
}
class MockServerPool {
    constructor() {
        this.servers = [];
    }
    createServer(options) {
        const baseUrl = (options === null || options === void 0 ? void 0 : options.baseUrl) || (0, randomBaseUrl_js_1.randomBaseUrl)();
        const server = new MockServer_js_1.MockServer({ baseUrl, server: mswServer });
        this.servers.push(server);
        return server;
    }
    getServers() {
        return [...this.servers];
    }
    listen() {
        const onUnhandledRequest = process.env.LOG_LEVEL === "debug" ? "warn" : "bypass";
        mswServer.listen({ onUnhandledRequest });
        if (process.env.LOG_LEVEL === "debug") {
            mswServer.events.on("request:start", (_a) => __awaiter(this, [_a], void 0, function* ({ request, requestId }) {
                const formattedRequest = yield formatHttpRequest(request, requestId);
                console.debug("request:start\n" + formattedRequest);
            }));
            mswServer.events.on("request:unhandled", (_a) => __awaiter(this, [_a], void 0, function* ({ request, requestId }) {
                const formattedRequest = yield formatHttpRequest(request, requestId);
                console.debug("request:unhandled\n" + formattedRequest);
            }));
            mswServer.events.on("response:mocked", (_a) => __awaiter(this, [_a], void 0, function* ({ request, response, requestId }) {
                const formattedResponse = yield formatHttpResponse(response, requestId);
                console.debug("response:mocked\n" + formattedResponse);
            }));
        }
    }
    close() {
        this.servers = [];
        mswServer.close();
    }
}
exports.mockServerPool = new MockServerPool();
