// auth0/dist/cjs/management/tests/unit/fetcher/Fetcher.test.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const fs_1 = __importDefault(require("fs"));
const stream_1 = __importDefault(require("stream"));
const path_1 = require("path");
const Fetcher_js_1 = require("../../../../../src/management/core/fetcher/Fetcher.js");
describe("Test fetcherImpl", () => {
    it("should handle successful request", () => __awaiter(void 0, void 0, void 0, function* () {
        const mockArgs = {
            url: "https://httpbin.org/post",
            method: "POST",
            headers: { "X-Test": "x-test-header" },
            body: { data: "test" },
            contentType: "application/json",
            requestType: "json",
            responseType: "json",
        };
        global.fetch = jest.fn().mockResolvedValue(new Response(JSON.stringify({ data: "test" }), {
            status: 200,
            statusText: "OK",
        }));
        const result = yield (0, Fetcher_js_1.fetcherImpl)(mockArgs);
        expect(result.ok).toBe(true);
        if (result.ok) {
            expect(result.body).toEqual({ data: "test" });
        }
        expect(global.fetch).toHaveBeenCalledWith("https://httpbin.org/post", expect.objectContaining({
            method: "POST",
            headers: expect.objectContaining({ "X-Test": "x-test-header" }),
            body: JSON.stringify({ data: "test" }),
        }));
    }));
    it("should send octet stream", () => __awaiter(void 0, void 0, void 0, function* () {
        const url = "https://httpbin.org/post/file";
        const mockArgs = {
            url,
            method: "POST",
            headers: { "X-Test": "x-test-header" },
            contentType: "application/octet-stream",
            requestType: "bytes",
            responseType: "json",
            body: fs_1.default.createReadStream((0, path_1.join)(__dirname, "test-file.txt")),
        };
        global.fetch = jest.fn().mockResolvedValue(new Response(JSON.stringify({ data: "test" }), {
            status: 200,
            statusText: "OK",
        }));
        const result = yield (0, Fetcher_js_1.fetcherImpl)(mockArgs);
        expect(global.fetch).toHaveBeenCalledWith(url, expect.objectContaining({
            method: "POST",
            headers: expect.objectContaining({ "X-Test": "x-test-header" }),
            body: expect.any(fs_1.default.ReadStream),
        }));
        expect(result.ok).toBe(true);
        if (result.ok) {
            expect(result.body).toEqual({ data: "test" });
        }
    }));
    it("should receive file as stream", () => __awaiter(void 0, void 0, void 0, function* () {
        const url = "https://httpbin.org/post/file";
        const mockArgs = {
            url,
            method: "GET",
            headers: { "X-Test": "x-test-header" },
            responseType: "binary-response",
        };
        global.fetch = jest.fn().mockResolvedValue(new Response(stream_1.default.Readable.toWeb(fs_1.default.createReadStream((0, path_1.join)(__dirname, "test-file.txt"))), {
            status: 200,
            statusText: "OK",
        }));
        const result = yield (0, Fetcher_js_1.fetcherImpl)(mockArgs);
        expect(global.fetch).toHaveBeenCalledWith(url, expect.objectContaining({
            method: "GET",
            headers: expect.objectContaining({ "X-Test": "x-test-header" }),
        }));
        expect(result.ok).toBe(true);
        if (result.ok) {
            const body = result.body;
            expect(body).toBeDefined();
            expect(body.bodyUsed).toBe(false);
            expect(typeof body.stream).toBe("function");
            const stream = body.stream();
            expect(stream).toBeInstanceOf(ReadableStream);
            const reader = stream.getReader();
            const { value } = yield reader.read();
            const decoder = new TextDecoder();
            const streamContent = decoder.decode(value);
            expect(streamContent).toBe("This is a test file!\n");
            expect(body.bodyUsed).toBe(true);
        }
    }));
    it("should receive file as blob", () => __awaiter(void 0, void 0, void 0, function* () {
        const url = "https://httpbin.org/post/file";
        const mockArgs = {
            url,
            method: "GET",
            headers: { "X-Test": "x-test-header" },
            responseType: "binary-response",
        };
        global.fetch = jest.fn().mockResolvedValue(new Response(stream_1.default.Readable.toWeb(fs_1.default.createReadStream((0, path_1.join)(__dirname, "test-file.txt"))), {
            status: 200,
            statusText: "OK",
        }));
        const result = yield (0, Fetcher_js_1.fetcherImpl)(mockArgs);
        expect(global.fetch).toHaveBeenCalledWith(url, expect.objectContaining({
            method: "GET",
            headers: expect.objectContaining({ "X-Test": "x-test-header" }),
        }));
        expect(result.ok).toBe(true);
        if (result.ok) {
            const body = result.body;
            expect(body).toBeDefined();
            expect(body.bodyUsed).toBe(false);
            expect(typeof body.blob).toBe("function");
            const blob = yield body.blob();
            expect(blob).toBeInstanceOf(Blob);
            const reader = blob.stream().getReader();
            const { value } = yield reader.read();
            const decoder = new TextDecoder();
            const streamContent = decoder.decode(value);
            expect(streamContent).toBe("This is a test file!\n");
            expect(body.bodyUsed).toBe(true);
        }
    }));
    it("should receive file as arraybuffer", () => __awaiter(void 0, void 0, void 0, function* () {
        const url = "https://httpbin.org/post/file";
        const mockArgs = {
            url,
            method: "GET",
            headers: { "X-Test": "x-test-header" },
            responseType: "binary-response",
        };
        global.fetch = jest.fn().mockResolvedValue(new Response(stream_1.default.Readable.toWeb(fs_1.default.createReadStream((0, path_1.join)(__dirname, "test-file.txt"))), {
            status: 200,
            statusText: "OK",
        }));
        const result = yield (0, Fetcher_js_1.fetcherImpl)(mockArgs);
        expect(global.fetch).toHaveBeenCalledWith(url, expect.objectContaining({
            method: "GET",
            headers: expect.objectContaining({ "X-Test": "x-test-header" }),
        }));
        expect(result.ok).toBe(true);
        if (result.ok) {
            const body = result.body;
            expect(body).toBeDefined();
            expect(body.bodyUsed).toBe(false);
            expect(typeof body.arrayBuffer).toBe("function");
            const arrayBuffer = yield body.arrayBuffer();
            expect(arrayBuffer).toBeInstanceOf(ArrayBuffer);
            const decoder = new TextDecoder();
            const streamContent = decoder.decode(new Uint8Array(arrayBuffer));
            expect(streamContent).toBe("This is a test file!\n");
            expect(body.bodyUsed).toBe(true);
        }
    }));
    it("should receive file as bytes", () => __awaiter(void 0, void 0, void 0, function* () {
        const url = "https://httpbin.org/post/file";
        const mockArgs = {
            url,
            method: "GET",
            headers: { "X-Test": "x-test-header" },
            responseType: "binary-response",
        };
        global.fetch = jest.fn().mockResolvedValue(new Response(stream_1.default.Readable.toWeb(fs_1.default.createReadStream((0, path_1.join)(__dirname, "test-file.txt"))), {
            status: 200,
            statusText: "OK",
        }));
        const result = yield (0, Fetcher_js_1.fetcherImpl)(mockArgs);
        expect(global.fetch).toHaveBeenCalledWith(url, expect.objectContaining({
            method: "GET",
            headers: expect.objectContaining({ "X-Test": "x-test-header" }),
        }));
        expect(result.ok).toBe(true);
        if (result.ok) {
            const body = result.body;
            expect(body).toBeDefined();
            expect(body.bodyUsed).toBe(false);
            expect(typeof body.bytes).toBe("function");
            if (!body.bytes) {
                return;
            }
            const bytes = yield body.bytes();
            expect(bytes).toBeInstanceOf(Uint8Array);
            const decoder = new TextDecoder();
            const streamContent = decoder.decode(bytes);
            expect(streamContent).toBe("This is a test file!\n");
            expect(body.bodyUsed).toBe(true);
        }
    }));
});
