// auth0/dist/cjs/management/tests/unit/fetcher/getResponseBody.test.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const index_js_1 = require("../../../../../src/management/core/runtime/index.js");
const getResponseBody_js_1 = require("../../../../../src/management/core/fetcher/getResponseBody.js");
describe("Test getResponseBody", () => {
    it("should handle blob response type", () => __awaiter(void 0, void 0, void 0, function* () {
        const mockBlob = new Blob(["test"], { type: "text/plain" });
        const mockResponse = new Response(mockBlob);
        const result = yield (0, getResponseBody_js_1.getResponseBody)(mockResponse, "blob");
        // @ts-expect-error
        expect(result.constructor.name).toBe("Blob");
    }));
    it("should handle sse response type", () => __awaiter(void 0, void 0, void 0, function* () {
        if (index_js_1.RUNTIME.type === "node") {
            const mockStream = new ReadableStream();
            const mockResponse = new Response(mockStream);
            const result = yield (0, getResponseBody_js_1.getResponseBody)(mockResponse, "sse");
            expect(result).toBe(mockStream);
        }
    }));
    it("should handle streaming response type", () => __awaiter(void 0, void 0, void 0, function* () {
        // Create a ReadableStream with some test data
        const encoder = new TextEncoder();
        const testData = "test stream data";
        const mockStream = new ReadableStream({
            start(controller) {
                controller.enqueue(encoder.encode(testData));
                controller.close();
            },
        });
        const mockResponse = new Response(mockStream);
        const result = (yield (0, getResponseBody_js_1.getResponseBody)(mockResponse, "streaming"));
        expect(result).toBeInstanceOf(ReadableStream);
        // Read and verify the stream content
        const reader = result.getReader();
        const decoder = new TextDecoder();
        const { value } = yield reader.read();
        const streamContent = decoder.decode(value);
        expect(streamContent).toBe(testData);
    }));
    it("should handle text response type", () => __awaiter(void 0, void 0, void 0, function* () {
        const mockResponse = new Response("test text");
        const result = yield (0, getResponseBody_js_1.getResponseBody)(mockResponse, "text");
        expect(result).toBe("test text");
    }));
    it("should handle JSON response", () => __awaiter(void 0, void 0, void 0, function* () {
        const mockJson = { key: "value" };
        const mockResponse = new Response(JSON.stringify(mockJson));
        const result = yield (0, getResponseBody_js_1.getResponseBody)(mockResponse);
        expect(result).toEqual(mockJson);
    }));
    it("should handle empty response", () => __awaiter(void 0, void 0, void 0, function* () {
        const mockResponse = new Response("");
        const result = yield (0, getResponseBody_js_1.getResponseBody)(mockResponse);
        expect(result).toBeUndefined();
    }));
    it("should handle non-JSON response", () => __awaiter(void 0, void 0, void 0, function* () {
        const mockResponse = new Response("invalid json");
        const result = yield (0, getResponseBody_js_1.getResponseBody)(mockResponse);
        expect(result).toEqual({
            ok: false,
            error: {
                reason: "non-json",
                statusCode: 200,
                rawBody: "invalid json",
            },
        });
    }));
});
