// auth0/dist/cjs/management/tests/unit/fetcher/HttpResponsePromise.test.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const globals_1 = require("@jest/globals");
const HttpResponsePromise_js_1 = require("../../../../../src/management/core/fetcher/HttpResponsePromise.js");
(0, globals_1.describe)("HttpResponsePromise", () => {
    const mockRawResponse = {
        headers: new Headers(),
        redirected: false,
        status: 200,
        statusText: "OK",
        type: "basic",
        url: "https://example.com",
    };
    const mockData = { id: "123", name: "test" };
    const mockWithRawResponse = {
        data: mockData,
        rawResponse: mockRawResponse,
    };
    (0, globals_1.describe)("fromFunction", () => {
        (0, globals_1.it)("should create an HttpResponsePromise from a function", () => __awaiter(void 0, void 0, void 0, function* () {
            const mockFn = globals_1.jest
                .fn()
                .mockResolvedValue(mockWithRawResponse);
            const responsePromise = HttpResponsePromise_js_1.HttpResponsePromise.fromFunction(mockFn, "arg1", "arg2");
            const result = yield responsePromise;
            (0, globals_1.expect)(result).toEqual(mockData);
            (0, globals_1.expect)(mockFn).toHaveBeenCalledWith("arg1", "arg2");
            const resultWithRawResponse = yield responsePromise.withRawResponse();
            (0, globals_1.expect)(resultWithRawResponse).toEqual({
                data: mockData,
                rawResponse: mockRawResponse,
            });
        }));
    });
    (0, globals_1.describe)("fromPromise", () => {
        (0, globals_1.it)("should create an HttpResponsePromise from a promise", () => __awaiter(void 0, void 0, void 0, function* () {
            const promise = Promise.resolve(mockWithRawResponse);
            const responsePromise = HttpResponsePromise_js_1.HttpResponsePromise.fromPromise(promise);
            const result = yield responsePromise;
            (0, globals_1.expect)(result).toEqual(mockData);
            const resultWithRawResponse = yield responsePromise.withRawResponse();
            (0, globals_1.expect)(resultWithRawResponse).toEqual({
                data: mockData,
                rawResponse: mockRawResponse,
            });
        }));
    });
    (0, globals_1.describe)("fromExecutor", () => {
        (0, globals_1.it)("should create an HttpResponsePromise from an executor function", () => __awaiter(void 0, void 0, void 0, function* () {
            const responsePromise = HttpResponsePromise_js_1.HttpResponsePromise.fromExecutor((resolve) => {
                resolve(mockWithRawResponse);
            });
            const result = yield responsePromise;
            (0, globals_1.expect)(result).toEqual(mockData);
            const resultWithRawResponse = yield responsePromise.withRawResponse();
            (0, globals_1.expect)(resultWithRawResponse).toEqual({
                data: mockData,
                rawResponse: mockRawResponse,
            });
        }));
    });
    (0, globals_1.describe)("fromResult", () => {
        (0, globals_1.it)("should create an HttpResponsePromise from a result", () => __awaiter(void 0, void 0, void 0, function* () {
            const responsePromise = HttpResponsePromise_js_1.HttpResponsePromise.fromResult(mockWithRawResponse);
            const result = yield responsePromise;
            (0, globals_1.expect)(result).toEqual(mockData);
            const resultWithRawResponse = yield responsePromise.withRawResponse();
            (0, globals_1.expect)(resultWithRawResponse).toEqual({
                data: mockData,
                rawResponse: mockRawResponse,
            });
        }));
    });
    (0, globals_1.describe)("Promise methods", () => {
        let responsePromise;
        (0, globals_1.beforeEach)(() => {
            responsePromise = HttpResponsePromise_js_1.HttpResponsePromise.fromResult(mockWithRawResponse);
        });
        (0, globals_1.it)("should support then() method", () => __awaiter(void 0, void 0, void 0, function* () {
            const result = yield responsePromise.then((data) => (Object.assign(Object.assign({}, data), { modified: true })));
            (0, globals_1.expect)(result).toEqual(Object.assign(Object.assign({}, mockData), { modified: true }));
        }));
        (0, globals_1.it)("should support catch() method", () => __awaiter(void 0, void 0, void 0, function* () {
            var _a;
            const errorResponsePromise = HttpResponsePromise_js_1.HttpResponsePromise.fromExecutor((_, reject) => {
                reject(new Error("Test error"));
            });
            const catchSpy = globals_1.jest.fn();
            yield errorResponsePromise.catch(catchSpy);
            (0, globals_1.expect)(catchSpy).toHaveBeenCalled();
            const error = (_a = catchSpy.mock.calls[0]) === null || _a === void 0 ? void 0 : _a[0];
            (0, globals_1.expect)(error).toBeInstanceOf(Error);
            (0, globals_1.expect)(error.message).toBe("Test error");
        }));
        (0, globals_1.it)("should support finally() method", () => __awaiter(void 0, void 0, void 0, function* () {
            const finallySpy = globals_1.jest.fn();
            yield responsePromise.finally(finallySpy);
            (0, globals_1.expect)(finallySpy).toHaveBeenCalled();
        }));
    });
    (0, globals_1.describe)("withRawResponse", () => {
        (0, globals_1.it)("should return both data and raw response", () => __awaiter(void 0, void 0, void 0, function* () {
            const responsePromise = HttpResponsePromise_js_1.HttpResponsePromise.fromResult(mockWithRawResponse);
            const result = yield responsePromise.withRawResponse();
            (0, globals_1.expect)(result).toEqual({
                data: mockData,
                rawResponse: mockRawResponse,
            });
        }));
    });
});
