// auth0/dist/cjs/management/tests/unit/fetcher/RawResponse.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const globals_1 = require("@jest/globals");
const RawResponse_js_1 = require("../../../../../src/management/core/fetcher/RawResponse.js");
(0, globals_1.describe)("RawResponse", () => {
    (0, globals_1.describe)("toRawResponse", () => {
        (0, globals_1.it)("should convert Response to RawResponse by removing body, bodyUsed, and ok properties", () => {
            const mockHeaders = new Headers({ "content-type": "application/json" });
            const mockResponse = {
                body: "test body",
                bodyUsed: false,
                ok: true,
                headers: mockHeaders,
                redirected: false,
                status: 200,
                statusText: "OK",
                type: "basic",
                url: "https://example.com",
            };
            const result = (0, RawResponse_js_1.toRawResponse)(mockResponse);
            (0, globals_1.expect)("body" in result).toBe(false);
            (0, globals_1.expect)("bodyUsed" in result).toBe(false);
            (0, globals_1.expect)("ok" in result).toBe(false);
            (0, globals_1.expect)(result.headers).toBe(mockHeaders);
            (0, globals_1.expect)(result.redirected).toBe(false);
            (0, globals_1.expect)(result.status).toBe(200);
            (0, globals_1.expect)(result.statusText).toBe("OK");
            (0, globals_1.expect)(result.type).toBe("basic");
            (0, globals_1.expect)(result.url).toBe("https://example.com");
        });
    });
});
