// auth0/dist/cjs/management/tests/wire/verifiableCredentials/verification/templates.test.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const MockServerPool_js_1 = require("../../../mock-server/MockServerPool.js");
const Client_js_1 = require("../../../../Client.js");
const Management = __importStar(require("../../../../api/index.js"));
describe("Templates", () => {
    test("list (7452f08f)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = {
            next: "next",
            templates: [
                {
                    id: "id",
                    name: "name",
                    type: "type",
                    dialect: "dialect",
                    presentation: { "org.iso.18013.5.1.mDL": { "org.iso.18013.5.1": {} } },
                    custom_certificate_authority: "custom_certificate_authority",
                    well_known_trusted_issuers: "well_known_trusted_issuers",
                    created_at: "2024-01-15T09:30:00Z",
                    updated_at: "2024-01-15T09:30:00Z",
                },
            ],
        };
        server
            .mockEndpoint()
            .get("/verifiable-credentials/verification/templates")
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const expected = {
            next: "next",
            templates: [
                {
                    id: "id",
                    name: "name",
                    type: "type",
                    dialect: "dialect",
                    presentation: {
                        "org.iso.18013.5.1.mDL": {
                            "org.iso.18013.5.1": {},
                        },
                    },
                    custom_certificate_authority: "custom_certificate_authority",
                    well_known_trusted_issuers: "well_known_trusted_issuers",
                    created_at: "2024-01-15T09:30:00Z",
                    updated_at: "2024-01-15T09:30:00Z",
                },
            ],
        };
        const page = yield client.verifiableCredentials.verification.templates.list();
        expect(expected.templates).toEqual(page.data);
        expect(page.hasNextPage()).toBe(true);
        const nextPage = yield page.getNextPage();
        expect(expected.templates).toEqual(nextPage.data);
    }));
    test("list (c60dd33b)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/verifiable-credentials/verification/templates")
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.verifiableCredentials.verification.templates.list();
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("list (1e230aeb)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/verifiable-credentials/verification/templates")
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.verifiableCredentials.verification.templates.list();
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("list (af841397)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/verifiable-credentials/verification/templates")
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.verifiableCredentials.verification.templates.list();
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("list (ee1e23bf)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/verifiable-credentials/verification/templates")
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.verifiableCredentials.verification.templates.list();
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("create (4259b620)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            name: "name",
            type: "type",
            dialect: "dialect",
            presentation: { "org.iso.18013.5.1.mDL": { "org.iso.18013.5.1": {} } },
            well_known_trusted_issuers: "well_known_trusted_issuers",
        };
        const rawResponseBody = {
            id: "id",
            name: "name",
            type: "type",
            dialect: "dialect",
            presentation: { "org.iso.18013.5.1.mDL": { "org.iso.18013.5.1": {} } },
            custom_certificate_authority: "custom_certificate_authority",
            well_known_trusted_issuers: "well_known_trusted_issuers",
            created_at: "2024-01-15T09:30:00Z",
            updated_at: "2024-01-15T09:30:00Z",
        };
        server
            .mockEndpoint()
            .post("/verifiable-credentials/verification/templates")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.verifiableCredentials.verification.templates.create({
            name: "name",
            type: "type",
            dialect: "dialect",
            presentation: {
                "org.iso.18013.5.1.mDL": {
                    "org.iso.18013.5.1": {},
                },
            },
            well_known_trusted_issuers: "well_known_trusted_issuers",
        });
        expect(response).toEqual({
            id: "id",
            name: "name",
            type: "type",
            dialect: "dialect",
            presentation: {
                "org.iso.18013.5.1.mDL": {
                    "org.iso.18013.5.1": {},
                },
            },
            custom_certificate_authority: "custom_certificate_authority",
            well_known_trusted_issuers: "well_known_trusted_issuers",
            created_at: "2024-01-15T09:30:00Z",
            updated_at: "2024-01-15T09:30:00Z",
        });
    }));
    test("create (e6d1c4a4)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            name: "x",
            type: "x",
            dialect: "dialect",
            presentation: {
                "org.iso.18013.5.1.mDL": {
                    "org.iso.18013.5.1": {
                        family_name: undefined,
                        given_name: undefined,
                        birth_date: undefined,
                        issue_date: undefined,
                        expiry_date: undefined,
                        issuing_country: undefined,
                        issuing_authority: undefined,
                        portrait: undefined,
                        driving_privileges: undefined,
                        resident_address: undefined,
                        portrait_capture_date: undefined,
                        age_in_years: undefined,
                        age_birth_year: undefined,
                        issuing_jurisdiction: undefined,
                        nationality: undefined,
                        resident_city: undefined,
                        resident_state: undefined,
                        resident_postal_code: undefined,
                        resident_country: undefined,
                        family_name_national_character: undefined,
                        given_name_national_character: undefined,
                    },
                },
            },
            custom_certificate_authority: undefined,
            well_known_trusted_issuers: "x",
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/verifiable-credentials/verification/templates")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.verifiableCredentials.verification.templates.create({
                name: "x",
                type: "x",
                dialect: "dialect",
                presentation: {
                    "org.iso.18013.5.1.mDL": {
                        "org.iso.18013.5.1": {
                            family_name: undefined,
                            given_name: undefined,
                            birth_date: undefined,
                            issue_date: undefined,
                            expiry_date: undefined,
                            issuing_country: undefined,
                            issuing_authority: undefined,
                            portrait: undefined,
                            driving_privileges: undefined,
                            resident_address: undefined,
                            portrait_capture_date: undefined,
                            age_in_years: undefined,
                            age_birth_year: undefined,
                            issuing_jurisdiction: undefined,
                            nationality: undefined,
                            resident_city: undefined,
                            resident_state: undefined,
                            resident_postal_code: undefined,
                            resident_country: undefined,
                            family_name_national_character: undefined,
                            given_name_national_character: undefined,
                        },
                    },
                },
                custom_certificate_authority: undefined,
                well_known_trusted_issuers: "x",
            });
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("create (a7d171cc)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            name: "x",
            type: "x",
            dialect: "dialect",
            presentation: {
                "org.iso.18013.5.1.mDL": {
                    "org.iso.18013.5.1": {
                        family_name: undefined,
                        given_name: undefined,
                        birth_date: undefined,
                        issue_date: undefined,
                        expiry_date: undefined,
                        issuing_country: undefined,
                        issuing_authority: undefined,
                        portrait: undefined,
                        driving_privileges: undefined,
                        resident_address: undefined,
                        portrait_capture_date: undefined,
                        age_in_years: undefined,
                        age_birth_year: undefined,
                        issuing_jurisdiction: undefined,
                        nationality: undefined,
                        resident_city: undefined,
                        resident_state: undefined,
                        resident_postal_code: undefined,
                        resident_country: undefined,
                        family_name_national_character: undefined,
                        given_name_national_character: undefined,
                    },
                },
            },
            custom_certificate_authority: undefined,
            well_known_trusted_issuers: "x",
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/verifiable-credentials/verification/templates")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.verifiableCredentials.verification.templates.create({
                name: "x",
                type: "x",
                dialect: "dialect",
                presentation: {
                    "org.iso.18013.5.1.mDL": {
                        "org.iso.18013.5.1": {
                            family_name: undefined,
                            given_name: undefined,
                            birth_date: undefined,
                            issue_date: undefined,
                            expiry_date: undefined,
                            issuing_country: undefined,
                            issuing_authority: undefined,
                            portrait: undefined,
                            driving_privileges: undefined,
                            resident_address: undefined,
                            portrait_capture_date: undefined,
                            age_in_years: undefined,
                            age_birth_year: undefined,
                            issuing_jurisdiction: undefined,
                            nationality: undefined,
                            resident_city: undefined,
                            resident_state: undefined,
                            resident_postal_code: undefined,
                            resident_country: undefined,
                            family_name_national_character: undefined,
                            given_name_national_character: undefined,
                        },
                    },
                },
                custom_certificate_authority: undefined,
                well_known_trusted_issuers: "x",
            });
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("create (fe134cd0)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            name: "x",
            type: "x",
            dialect: "dialect",
            presentation: {
                "org.iso.18013.5.1.mDL": {
                    "org.iso.18013.5.1": {
                        family_name: undefined,
                        given_name: undefined,
                        birth_date: undefined,
                        issue_date: undefined,
                        expiry_date: undefined,
                        issuing_country: undefined,
                        issuing_authority: undefined,
                        portrait: undefined,
                        driving_privileges: undefined,
                        resident_address: undefined,
                        portrait_capture_date: undefined,
                        age_in_years: undefined,
                        age_birth_year: undefined,
                        issuing_jurisdiction: undefined,
                        nationality: undefined,
                        resident_city: undefined,
                        resident_state: undefined,
                        resident_postal_code: undefined,
                        resident_country: undefined,
                        family_name_national_character: undefined,
                        given_name_national_character: undefined,
                    },
                },
            },
            custom_certificate_authority: undefined,
            well_known_trusted_issuers: "x",
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/verifiable-credentials/verification/templates")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.verifiableCredentials.verification.templates.create({
                name: "x",
                type: "x",
                dialect: "dialect",
                presentation: {
                    "org.iso.18013.5.1.mDL": {
                        "org.iso.18013.5.1": {
                            family_name: undefined,
                            given_name: undefined,
                            birth_date: undefined,
                            issue_date: undefined,
                            expiry_date: undefined,
                            issuing_country: undefined,
                            issuing_authority: undefined,
                            portrait: undefined,
                            driving_privileges: undefined,
                            resident_address: undefined,
                            portrait_capture_date: undefined,
                            age_in_years: undefined,
                            age_birth_year: undefined,
                            issuing_jurisdiction: undefined,
                            nationality: undefined,
                            resident_city: undefined,
                            resident_state: undefined,
                            resident_postal_code: undefined,
                            resident_country: undefined,
                            family_name_national_character: undefined,
                            given_name_national_character: undefined,
                        },
                    },
                },
                custom_certificate_authority: undefined,
                well_known_trusted_issuers: "x",
            });
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("create (570b17e8)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            name: "x",
            type: "x",
            dialect: "dialect",
            presentation: {
                "org.iso.18013.5.1.mDL": {
                    "org.iso.18013.5.1": {
                        family_name: undefined,
                        given_name: undefined,
                        birth_date: undefined,
                        issue_date: undefined,
                        expiry_date: undefined,
                        issuing_country: undefined,
                        issuing_authority: undefined,
                        portrait: undefined,
                        driving_privileges: undefined,
                        resident_address: undefined,
                        portrait_capture_date: undefined,
                        age_in_years: undefined,
                        age_birth_year: undefined,
                        issuing_jurisdiction: undefined,
                        nationality: undefined,
                        resident_city: undefined,
                        resident_state: undefined,
                        resident_postal_code: undefined,
                        resident_country: undefined,
                        family_name_national_character: undefined,
                        given_name_national_character: undefined,
                    },
                },
            },
            custom_certificate_authority: undefined,
            well_known_trusted_issuers: "x",
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/verifiable-credentials/verification/templates")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(409)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.verifiableCredentials.verification.templates.create({
                name: "x",
                type: "x",
                dialect: "dialect",
                presentation: {
                    "org.iso.18013.5.1.mDL": {
                        "org.iso.18013.5.1": {
                            family_name: undefined,
                            given_name: undefined,
                            birth_date: undefined,
                            issue_date: undefined,
                            expiry_date: undefined,
                            issuing_country: undefined,
                            issuing_authority: undefined,
                            portrait: undefined,
                            driving_privileges: undefined,
                            resident_address: undefined,
                            portrait_capture_date: undefined,
                            age_in_years: undefined,
                            age_birth_year: undefined,
                            issuing_jurisdiction: undefined,
                            nationality: undefined,
                            resident_city: undefined,
                            resident_state: undefined,
                            resident_postal_code: undefined,
                            resident_country: undefined,
                            family_name_national_character: undefined,
                            given_name_national_character: undefined,
                        },
                    },
                },
                custom_certificate_authority: undefined,
                well_known_trusted_issuers: "x",
            });
        })).rejects.toThrow(new Management.ConflictError({
            key: "value",
        }));
    }));
    test("create (93203adc)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            name: "x",
            type: "x",
            dialect: "dialect",
            presentation: {
                "org.iso.18013.5.1.mDL": {
                    "org.iso.18013.5.1": {
                        family_name: undefined,
                        given_name: undefined,
                        birth_date: undefined,
                        issue_date: undefined,
                        expiry_date: undefined,
                        issuing_country: undefined,
                        issuing_authority: undefined,
                        portrait: undefined,
                        driving_privileges: undefined,
                        resident_address: undefined,
                        portrait_capture_date: undefined,
                        age_in_years: undefined,
                        age_birth_year: undefined,
                        issuing_jurisdiction: undefined,
                        nationality: undefined,
                        resident_city: undefined,
                        resident_state: undefined,
                        resident_postal_code: undefined,
                        resident_country: undefined,
                        family_name_national_character: undefined,
                        given_name_national_character: undefined,
                    },
                },
            },
            custom_certificate_authority: undefined,
            well_known_trusted_issuers: "x",
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/verifiable-credentials/verification/templates")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.verifiableCredentials.verification.templates.create({
                name: "x",
                type: "x",
                dialect: "dialect",
                presentation: {
                    "org.iso.18013.5.1.mDL": {
                        "org.iso.18013.5.1": {
                            family_name: undefined,
                            given_name: undefined,
                            birth_date: undefined,
                            issue_date: undefined,
                            expiry_date: undefined,
                            issuing_country: undefined,
                            issuing_authority: undefined,
                            portrait: undefined,
                            driving_privileges: undefined,
                            resident_address: undefined,
                            portrait_capture_date: undefined,
                            age_in_years: undefined,
                            age_birth_year: undefined,
                            issuing_jurisdiction: undefined,
                            nationality: undefined,
                            resident_city: undefined,
                            resident_state: undefined,
                            resident_postal_code: undefined,
                            resident_country: undefined,
                            family_name_national_character: undefined,
                            given_name_national_character: undefined,
                        },
                    },
                },
                custom_certificate_authority: undefined,
                well_known_trusted_issuers: "x",
            });
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("get (ca2a3e16)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = {
            id: "id",
            name: "name",
            type: "type",
            dialect: "dialect",
            presentation: { "org.iso.18013.5.1.mDL": { "org.iso.18013.5.1": {} } },
            custom_certificate_authority: "custom_certificate_authority",
            well_known_trusted_issuers: "well_known_trusted_issuers",
            created_at: "2024-01-15T09:30:00Z",
            updated_at: "2024-01-15T09:30:00Z",
        };
        server
            .mockEndpoint()
            .get("/verifiable-credentials/verification/templates/id")
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.verifiableCredentials.verification.templates.get("id");
        expect(response).toEqual({
            id: "id",
            name: "name",
            type: "type",
            dialect: "dialect",
            presentation: {
                "org.iso.18013.5.1.mDL": {
                    "org.iso.18013.5.1": {},
                },
            },
            custom_certificate_authority: "custom_certificate_authority",
            well_known_trusted_issuers: "well_known_trusted_issuers",
            created_at: "2024-01-15T09:30:00Z",
            updated_at: "2024-01-15T09:30:00Z",
        });
    }));
    test("get (fcf9dbd1)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/verifiable-credentials/verification/templates/id")
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.verifiableCredentials.verification.templates.get("id");
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("get (49d52691)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/verifiable-credentials/verification/templates/id")
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.verifiableCredentials.verification.templates.get("id");
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("get (2428808d)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/verifiable-credentials/verification/templates/id")
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.verifiableCredentials.verification.templates.get("id");
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("get (e55ce3fd)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/verifiable-credentials/verification/templates/id")
            .respondWith()
            .statusCode(404)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.verifiableCredentials.verification.templates.get("id");
        })).rejects.toThrow(new Management.NotFoundError({
            key: "value",
        }));
    }));
    test("get (27b44cb5)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/verifiable-credentials/verification/templates/id")
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.verifiableCredentials.verification.templates.get("id");
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("delete (c7f0a6bf)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        server
            .mockEndpoint()
            .delete("/verifiable-credentials/verification/templates/id")
            .respondWith()
            .statusCode(200)
            .build();
        const response = yield client.verifiableCredentials.verification.templates.delete("id");
        expect(response).toEqual(undefined);
    }));
    test("delete (fcf9dbd1)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/verifiable-credentials/verification/templates/id")
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.verifiableCredentials.verification.templates.delete("id");
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("delete (49d52691)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/verifiable-credentials/verification/templates/id")
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.verifiableCredentials.verification.templates.delete("id");
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("delete (2428808d)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/verifiable-credentials/verification/templates/id")
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.verifiableCredentials.verification.templates.delete("id");
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("delete (27b44cb5)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .delete("/verifiable-credentials/verification/templates/id")
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.verifiableCredentials.verification.templates.delete("id");
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("update (5bc46ab5)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {};
        const rawResponseBody = {
            id: "id",
            name: "name",
            type: "type",
            dialect: "dialect",
            presentation: { "org.iso.18013.5.1.mDL": { "org.iso.18013.5.1": {} } },
            custom_certificate_authority: "custom_certificate_authority",
            well_known_trusted_issuers: "well_known_trusted_issuers",
            created_at: "2024-01-15T09:30:00Z",
            updated_at: "2024-01-15T09:30:00Z",
        };
        server
            .mockEndpoint()
            .patch("/verifiable-credentials/verification/templates/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.verifiableCredentials.verification.templates.update("id");
        expect(response).toEqual({
            id: "id",
            name: "name",
            type: "type",
            dialect: "dialect",
            presentation: {
                "org.iso.18013.5.1.mDL": {
                    "org.iso.18013.5.1": {},
                },
            },
            custom_certificate_authority: "custom_certificate_authority",
            well_known_trusted_issuers: "well_known_trusted_issuers",
            created_at: "2024-01-15T09:30:00Z",
            updated_at: "2024-01-15T09:30:00Z",
        });
    }));
    test("update (bdb2fc50)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            name: undefined,
            type: undefined,
            dialect: undefined,
            presentation: undefined,
            well_known_trusted_issuers: undefined,
            version: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/verifiable-credentials/verification/templates/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.verifiableCredentials.verification.templates.update("id", {
                name: undefined,
                type: undefined,
                dialect: undefined,
                presentation: undefined,
                well_known_trusted_issuers: undefined,
                version: undefined,
            });
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("update (f15da378)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            name: undefined,
            type: undefined,
            dialect: undefined,
            presentation: undefined,
            well_known_trusted_issuers: undefined,
            version: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/verifiable-credentials/verification/templates/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.verifiableCredentials.verification.templates.update("id", {
                name: undefined,
                type: undefined,
                dialect: undefined,
                presentation: undefined,
                well_known_trusted_issuers: undefined,
                version: undefined,
            });
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("update (d3351a7c)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            name: undefined,
            type: undefined,
            dialect: undefined,
            presentation: undefined,
            well_known_trusted_issuers: undefined,
            version: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/verifiable-credentials/verification/templates/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.verifiableCredentials.verification.templates.update("id", {
                name: undefined,
                type: undefined,
                dialect: undefined,
                presentation: undefined,
                well_known_trusted_issuers: undefined,
                version: undefined,
            });
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("update (ef4f43d4)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            name: undefined,
            type: undefined,
            dialect: undefined,
            presentation: undefined,
            well_known_trusted_issuers: undefined,
            version: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/verifiable-credentials/verification/templates/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(409)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.verifiableCredentials.verification.templates.update("id", {
                name: undefined,
                type: undefined,
                dialect: undefined,
                presentation: undefined,
                well_known_trusted_issuers: undefined,
                version: undefined,
            });
        })).rejects.toThrow(new Management.ConflictError({
            key: "value",
        }));
    }));
    test("update (b488d448)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {
            name: undefined,
            type: undefined,
            dialect: undefined,
            presentation: undefined,
            well_known_trusted_issuers: undefined,
            version: undefined,
        };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .patch("/verifiable-credentials/verification/templates/id")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.verifiableCredentials.verification.templates.update("id", {
                name: undefined,
                type: undefined,
                dialect: undefined,
                presentation: undefined,
                well_known_trusted_issuers: undefined,
                version: undefined,
            });
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
});
