// auth0/dist/esm/management/tests/wire/jobs/usersExports.test.mjs
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import { mockServerPool } from "../../mock-server/MockServerPool.mjs";
import { ManagementClient } from "../../../Client.mjs";
import * as Management from "../../../api/index.mjs";
describe("UsersExports", () => {
    test("create (9bd53ff1)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = mockServerPool.createServer();
        const client = new ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = {};
        const rawResponseBody = {
            status: "status",
            type: "type",
            created_at: "created_at",
            id: "id",
            connection_id: "connection_id",
            format: "json",
            limit: 1,
            fields: [{ name: "name", export_as: "export_as" }],
        };
        server
            .mockEndpoint()
            .post("/jobs/users-exports")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.jobs.usersExports.create();
        expect(response).toEqual({
            status: "status",
            type: "type",
            created_at: "created_at",
            id: "id",
            connection_id: "connection_id",
            format: "json",
            limit: 1,
            fields: [
                {
                    name: "name",
                    export_as: "export_as",
                },
            ],
        });
    }));
    test("create (f557ff6e)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = mockServerPool.createServer();
        const client = new ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { connection_id: undefined, format: undefined, limit: undefined, fields: undefined };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/jobs/users-exports")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.jobs.usersExports.create({
                connection_id: undefined,
                format: undefined,
                limit: undefined,
                fields: undefined,
            });
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("create (37bbbaf6)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = mockServerPool.createServer();
        const client = new ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { connection_id: undefined, format: undefined, limit: undefined, fields: undefined };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/jobs/users-exports")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.jobs.usersExports.create({
                connection_id: undefined,
                format: undefined,
                limit: undefined,
                fields: undefined,
            });
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("create (4b5391fa)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = mockServerPool.createServer();
        const client = new ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { connection_id: undefined, format: undefined, limit: undefined, fields: undefined };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/jobs/users-exports")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.jobs.usersExports.create({
                connection_id: undefined,
                format: undefined,
                limit: undefined,
                fields: undefined,
            });
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("create (2f18f096)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = mockServerPool.createServer();
        const client = new ManagementClient({ token: "test", environment: server.baseUrl });
        const rawRequestBody = { connection_id: undefined, format: undefined, limit: undefined, fields: undefined };
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/jobs/users-exports")
            .jsonBody(rawRequestBody)
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.jobs.usersExports.create({
                connection_id: undefined,
                format: undefined,
                limit: undefined,
                fields: undefined,
            });
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
});
