// auth0/dist/cjs/management/tests/wire/actions/versions.test.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const MockServerPool_js_1 = require("../../mock-server/MockServerPool.js");
const Client_js_1 = require("../../../Client.js");
const Management = __importStar(require("../../../api/index.js"));
describe("Versions", () => {
    test("list (9b6673ae)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = {
            total: 1.1,
            page: 1.1,
            per_page: 1.1,
            versions: [
                {
                    id: "id",
                    action_id: "action_id",
                    code: "code",
                    dependencies: [{}],
                    deployed: true,
                    runtime: "runtime",
                    secrets: [{}],
                    status: "pending",
                    number: 1.1,
                    errors: [{}],
                    built_at: "2024-01-15T09:30:00Z",
                    created_at: "2024-01-15T09:30:00Z",
                    updated_at: "2024-01-15T09:30:00Z",
                    supported_triggers: [{ id: "id" }],
                },
            ],
        };
        server
            .mockEndpoint()
            .get("/actions/actions/actionId/versions")
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const expected = {
            total: 1.1,
            page: 1.1,
            per_page: 1.1,
            versions: [
                {
                    id: "id",
                    action_id: "action_id",
                    code: "code",
                    dependencies: [{}],
                    deployed: true,
                    runtime: "runtime",
                    secrets: [{}],
                    status: "pending",
                    number: 1.1,
                    errors: [{}],
                    built_at: "2024-01-15T09:30:00Z",
                    created_at: "2024-01-15T09:30:00Z",
                    updated_at: "2024-01-15T09:30:00Z",
                    supported_triggers: [
                        {
                            id: "id",
                        },
                    ],
                },
            ],
        };
        const page = yield client.actions.versions.list("actionId");
        expect(expected.versions).toEqual(page.data);
        expect(page.hasNextPage()).toBe(true);
        const nextPage = yield page.getNextPage();
        expect(expected.versions).toEqual(nextPage.data);
    }));
    test("list (4f66e55f)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/actions/actions/actionId/versions")
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.actions.versions.list("actionId");
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("list (a73309ef)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/actions/actions/actionId/versions")
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.actions.versions.list("actionId");
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("list (352abebb)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/actions/actions/actionId/versions")
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.actions.versions.list("actionId");
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("list (5f69aff3)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/actions/actions/actionId/versions")
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.actions.versions.list("actionId");
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("get (fe1f28e3)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = {
            id: "id",
            action_id: "action_id",
            code: "code",
            dependencies: [{ name: "name", version: "version", registry_url: "registry_url" }],
            deployed: true,
            runtime: "runtime",
            secrets: [{ name: "name", updated_at: "2024-01-15T09:30:00Z" }],
            status: "pending",
            number: 1.1,
            errors: [{ id: "id", msg: "msg", url: "url" }],
            action: {
                id: "id",
                name: "name",
                supported_triggers: [{ id: "id" }],
                all_changes_deployed: true,
                created_at: "2024-01-15T09:30:00Z",
                updated_at: "2024-01-15T09:30:00Z",
            },
            built_at: "2024-01-15T09:30:00Z",
            created_at: "2024-01-15T09:30:00Z",
            updated_at: "2024-01-15T09:30:00Z",
            supported_triggers: [
                {
                    id: "id",
                    version: "version",
                    status: "status",
                    runtimes: ["runtimes"],
                    default_runtime: "default_runtime",
                    compatible_triggers: [{ id: "id", version: "version" }],
                    binding_policy: "trigger-bound",
                },
            ],
        };
        server
            .mockEndpoint()
            .get("/actions/actions/actionId/versions/id")
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.actions.versions.get("actionId", "id");
        expect(response).toEqual({
            id: "id",
            action_id: "action_id",
            code: "code",
            dependencies: [
                {
                    name: "name",
                    version: "version",
                    registry_url: "registry_url",
                },
            ],
            deployed: true,
            runtime: "runtime",
            secrets: [
                {
                    name: "name",
                    updated_at: "2024-01-15T09:30:00Z",
                },
            ],
            status: "pending",
            number: 1.1,
            errors: [
                {
                    id: "id",
                    msg: "msg",
                    url: "url",
                },
            ],
            action: {
                id: "id",
                name: "name",
                supported_triggers: [
                    {
                        id: "id",
                    },
                ],
                all_changes_deployed: true,
                created_at: "2024-01-15T09:30:00Z",
                updated_at: "2024-01-15T09:30:00Z",
            },
            built_at: "2024-01-15T09:30:00Z",
            created_at: "2024-01-15T09:30:00Z",
            updated_at: "2024-01-15T09:30:00Z",
            supported_triggers: [
                {
                    id: "id",
                    version: "version",
                    status: "status",
                    runtimes: ["runtimes"],
                    default_runtime: "default_runtime",
                    compatible_triggers: [
                        {
                            id: "id",
                            version: "version",
                        },
                    ],
                    binding_policy: "trigger-bound",
                },
            ],
        });
    }));
    test("get (2f818d9f)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/actions/actions/actionId/versions/id")
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.actions.versions.get("actionId", "id");
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("get (e858d62f)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/actions/actions/actionId/versions/id")
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.actions.versions.get("actionId", "id");
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("get (c30357fb)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/actions/actions/actionId/versions/id")
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.actions.versions.get("actionId", "id");
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("get (9605349b)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/actions/actions/actionId/versions/id")
            .respondWith()
            .statusCode(404)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.actions.versions.get("actionId", "id");
        })).rejects.toThrow(new Management.NotFoundError({
            key: "value",
        }));
    }));
    test("get (429ebe33)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .get("/actions/actions/actionId/versions/id")
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.actions.versions.get("actionId", "id");
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
    test("deploy (5f2fffbe)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = {
            id: "12a3b9e6-06e6-4a29-96bf-90c82fe79a0d",
            action_id: "910b1053-577f-4d81-a8c8-020e7319a38a",
            code: "module.exports = () => {}",
            dependencies: [
                { name: "name", version: "version", registry_url: "registry_url" },
                { name: "name", version: "version", registry_url: "registry_url" },
            ],
            deployed: true,
            runtime: "node22",
            secrets: [
                { name: "mySecret", updated_at: "2024-01-15T09:30:00Z" },
                { name: "mySecret", updated_at: "2024-01-15T09:30:00Z" },
            ],
            status: "pending",
            number: 1.1,
            errors: [
                { id: "id", msg: "msg", url: "url" },
                { id: "id", msg: "msg", url: "url" },
            ],
            action: {
                id: "910b1053-577f-4d81-a8c8-020e7319a38a",
                name: "my-action",
                supported_triggers: [
                    {
                        id: "id",
                        version: "version",
                        status: "status",
                        runtimes: ["runtimes", "runtimes"],
                        default_runtime: "default_runtime",
                        compatible_triggers: [
                            { id: "id", version: "version" },
                            { id: "id", version: "version" },
                        ],
                        binding_policy: "trigger-bound",
                    },
                    {
                        id: "id",
                        version: "version",
                        status: "status",
                        runtimes: ["runtimes", "runtimes"],
                        default_runtime: "default_runtime",
                        compatible_triggers: [
                            { id: "id", version: "version" },
                            { id: "id", version: "version" },
                        ],
                        binding_policy: "trigger-bound",
                    },
                ],
                all_changes_deployed: true,
                created_at: "2024-01-15T09:30:00Z",
                updated_at: "2024-01-15T09:30:00Z",
            },
            built_at: "2024-01-15T09:30:00Z",
            created_at: "2024-01-15T09:30:00Z",
            updated_at: "2024-01-15T09:30:00Z",
            supported_triggers: [
                {
                    id: "id",
                    version: "version",
                    status: "status",
                    runtimes: ["runtimes", "runtimes"],
                    default_runtime: "default_runtime",
                    compatible_triggers: [
                        { id: "id", version: "version" },
                        { id: "id", version: "version" },
                    ],
                    binding_policy: "trigger-bound",
                },
                {
                    id: "id",
                    version: "version",
                    status: "status",
                    runtimes: ["runtimes", "runtimes"],
                    default_runtime: "default_runtime",
                    compatible_triggers: [
                        { id: "id", version: "version" },
                        { id: "id", version: "version" },
                    ],
                    binding_policy: "trigger-bound",
                },
            ],
        };
        server
            .mockEndpoint()
            .post("/actions/actions/actionId/versions/id/deploy")
            .respondWith()
            .statusCode(200)
            .jsonBody(rawResponseBody)
            .build();
        const response = yield client.actions.versions.deploy("id", "actionId", undefined);
        expect(response).toEqual({
            id: "12a3b9e6-06e6-4a29-96bf-90c82fe79a0d",
            action_id: "910b1053-577f-4d81-a8c8-020e7319a38a",
            code: "module.exports = () => {}",
            dependencies: [
                {
                    name: "name",
                    version: "version",
                    registry_url: "registry_url",
                },
                {
                    name: "name",
                    version: "version",
                    registry_url: "registry_url",
                },
            ],
            deployed: true,
            runtime: "node22",
            secrets: [
                {
                    name: "mySecret",
                    updated_at: "2024-01-15T09:30:00Z",
                },
                {
                    name: "mySecret",
                    updated_at: "2024-01-15T09:30:00Z",
                },
            ],
            status: "pending",
            number: 1.1,
            errors: [
                {
                    id: "id",
                    msg: "msg",
                    url: "url",
                },
                {
                    id: "id",
                    msg: "msg",
                    url: "url",
                },
            ],
            action: {
                id: "910b1053-577f-4d81-a8c8-020e7319a38a",
                name: "my-action",
                supported_triggers: [
                    {
                        id: "id",
                        version: "version",
                        status: "status",
                        runtimes: ["runtimes", "runtimes"],
                        default_runtime: "default_runtime",
                        compatible_triggers: [
                            {
                                id: "id",
                                version: "version",
                            },
                            {
                                id: "id",
                                version: "version",
                            },
                        ],
                        binding_policy: "trigger-bound",
                    },
                    {
                        id: "id",
                        version: "version",
                        status: "status",
                        runtimes: ["runtimes", "runtimes"],
                        default_runtime: "default_runtime",
                        compatible_triggers: [
                            {
                                id: "id",
                                version: "version",
                            },
                            {
                                id: "id",
                                version: "version",
                            },
                        ],
                        binding_policy: "trigger-bound",
                    },
                ],
                all_changes_deployed: true,
                created_at: "2024-01-15T09:30:00Z",
                updated_at: "2024-01-15T09:30:00Z",
            },
            built_at: "2024-01-15T09:30:00Z",
            created_at: "2024-01-15T09:30:00Z",
            updated_at: "2024-01-15T09:30:00Z",
            supported_triggers: [
                {
                    id: "id",
                    version: "version",
                    status: "status",
                    runtimes: ["runtimes", "runtimes"],
                    default_runtime: "default_runtime",
                    compatible_triggers: [
                        {
                            id: "id",
                            version: "version",
                        },
                        {
                            id: "id",
                            version: "version",
                        },
                    ],
                    binding_policy: "trigger-bound",
                },
                {
                    id: "id",
                    version: "version",
                    status: "status",
                    runtimes: ["runtimes", "runtimes"],
                    default_runtime: "default_runtime",
                    compatible_triggers: [
                        {
                            id: "id",
                            version: "version",
                        },
                        {
                            id: "id",
                            version: "version",
                        },
                    ],
                    binding_policy: "trigger-bound",
                },
            ],
        });
    }));
    test("deploy (b1a577f0)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/actions/actions/actionId/versions/id/deploy")
            .respondWith()
            .statusCode(400)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.actions.versions.deploy("id", "actionId", undefined);
        })).rejects.toThrow(new Management.BadRequestError({
            key: "value",
        }));
    }));
    test("deploy (8c564298)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/actions/actions/actionId/versions/id/deploy")
            .respondWith()
            .statusCode(401)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.actions.versions.deploy("id", "actionId", undefined);
        })).rejects.toThrow(new Management.UnauthorizedError({
            key: "value",
        }));
    }));
    test("deploy (b251bf9c)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/actions/actions/actionId/versions/id/deploy")
            .respondWith()
            .statusCode(403)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.actions.versions.deploy("id", "actionId", undefined);
        })).rejects.toThrow(new Management.ForbiddenError({
            key: "value",
        }));
    }));
    test("deploy (9f001568)", () => __awaiter(void 0, void 0, void 0, function* () {
        const server = MockServerPool_js_1.mockServerPool.createServer();
        const client = new Client_js_1.ManagementClient({ token: "test", environment: server.baseUrl });
        const rawResponseBody = { key: "value" };
        server
            .mockEndpoint()
            .post("/actions/actions/actionId/versions/id/deploy")
            .respondWith()
            .statusCode(429)
            .jsonBody(rawResponseBody)
            .build();
        yield expect(() => __awaiter(void 0, void 0, void 0, function* () {
            return yield client.actions.versions.deploy("id", "actionId", undefined);
        })).rejects.toThrow(new Management.TooManyRequestsError({
            key: "value",
        }));
    }));
});
