// auth0/dist/cjs/management/api/resources/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.verifiableCredentials = exports.tenants = exports.riskAssessments = exports.keys = exports.guardian = exports.groups = exports.emails = exports.attackProtection = exports.anomaly = exports.users = exports.userBlocks = exports.tokenExchangeProfiles = exports.tickets = exports.supplementalSignals = exports.stats = exports.sessions = exports.selfServiceProfiles = exports.rulesConfigs = exports.rules = exports.roles = exports.resourceServers = exports.refreshTokens = exports.prompts = exports.organizations = exports.networkAcls = exports.logs = exports.logStreams = exports.hooks = exports.userGrants = exports.forms = exports.flows = exports.emailTemplates = exports.deviceCredentials = exports.customDomains = exports.connections = exports.clients = exports.clientGrants = exports.branding = exports.actions = exports.jobs = exports.eventStreams = void 0;
exports.eventStreams = __importStar(require("./eventStreams/index.js"));
__exportStar(require("./eventStreams/types/index.js"), exports);
exports.jobs = __importStar(require("./jobs/index.js"));
exports.actions = __importStar(require("./actions/index.js"));
exports.branding = __importStar(require("./branding/index.js"));
exports.clientGrants = __importStar(require("./clientGrants/index.js"));
exports.clients = __importStar(require("./clients/index.js"));
exports.connections = __importStar(require("./connections/index.js"));
exports.customDomains = __importStar(require("./customDomains/index.js"));
exports.deviceCredentials = __importStar(require("./deviceCredentials/index.js"));
exports.emailTemplates = __importStar(require("./emailTemplates/index.js"));
exports.flows = __importStar(require("./flows/index.js"));
exports.forms = __importStar(require("./forms/index.js"));
exports.userGrants = __importStar(require("./userGrants/index.js"));
exports.hooks = __importStar(require("./hooks/index.js"));
exports.logStreams = __importStar(require("./logStreams/index.js"));
exports.logs = __importStar(require("./logs/index.js"));
exports.networkAcls = __importStar(require("./networkAcls/index.js"));
exports.organizations = __importStar(require("./organizations/index.js"));
exports.prompts = __importStar(require("./prompts/index.js"));
exports.refreshTokens = __importStar(require("./refreshTokens/index.js"));
exports.resourceServers = __importStar(require("./resourceServers/index.js"));
exports.roles = __importStar(require("./roles/index.js"));
exports.rules = __importStar(require("./rules/index.js"));
exports.rulesConfigs = __importStar(require("./rulesConfigs/index.js"));
exports.selfServiceProfiles = __importStar(require("./selfServiceProfiles/index.js"));
exports.sessions = __importStar(require("./sessions/index.js"));
exports.stats = __importStar(require("./stats/index.js"));
exports.supplementalSignals = __importStar(require("./supplementalSignals/index.js"));
exports.tickets = __importStar(require("./tickets/index.js"));
exports.tokenExchangeProfiles = __importStar(require("./tokenExchangeProfiles/index.js"));
exports.userBlocks = __importStar(require("./userBlocks/index.js"));
exports.users = __importStar(require("./users/index.js"));
exports.anomaly = __importStar(require("./anomaly/index.js"));
exports.attackProtection = __importStar(require("./attackProtection/index.js"));
exports.emails = __importStar(require("./emails/index.js"));
exports.groups = __importStar(require("./groups/index.js"));
exports.guardian = __importStar(require("./guardian/index.js"));
exports.keys = __importStar(require("./keys/index.js"));
exports.riskAssessments = __importStar(require("./riskAssessments/index.js"));
exports.tenants = __importStar(require("./tenants/index.js"));
exports.verifiableCredentials = __importStar(require("./verifiableCredentials/index.js"));
__exportStar(require("./actions/client/requests/index.js"), exports);
__exportStar(require("./branding/client/requests/index.js"), exports);
__exportStar(require("./clientGrants/client/requests/index.js"), exports);
__exportStar(require("./clients/client/requests/index.js"), exports);
__exportStar(require("./connections/client/requests/index.js"), exports);
__exportStar(require("./customDomains/client/requests/index.js"), exports);
__exportStar(require("./deviceCredentials/client/requests/index.js"), exports);
__exportStar(require("./emailTemplates/client/requests/index.js"), exports);
__exportStar(require("./eventStreams/client/requests/index.js"), exports);
__exportStar(require("./flows/client/requests/index.js"), exports);
__exportStar(require("./forms/client/requests/index.js"), exports);
__exportStar(require("./userGrants/client/requests/index.js"), exports);
__exportStar(require("./hooks/client/requests/index.js"), exports);
__exportStar(require("./logStreams/client/requests/index.js"), exports);
__exportStar(require("./logs/client/requests/index.js"), exports);
__exportStar(require("./networkAcls/client/requests/index.js"), exports);
__exportStar(require("./organizations/client/requests/index.js"), exports);
__exportStar(require("./prompts/client/requests/index.js"), exports);
__exportStar(require("./resourceServers/client/requests/index.js"), exports);
__exportStar(require("./roles/client/requests/index.js"), exports);
__exportStar(require("./rules/client/requests/index.js"), exports);
__exportStar(require("./rulesConfigs/client/requests/index.js"), exports);
__exportStar(require("./selfServiceProfiles/client/requests/index.js"), exports);
__exportStar(require("./stats/client/requests/index.js"), exports);
__exportStar(require("./supplementalSignals/client/requests/index.js"), exports);
__exportStar(require("./tickets/client/requests/index.js"), exports);
__exportStar(require("./tokenExchangeProfiles/client/requests/index.js"), exports);
__exportStar(require("./userBlocks/client/requests/index.js"), exports);
__exportStar(require("./users/client/requests/index.js"), exports);
__exportStar(require("./actions/resources/versions/client/requests/index.js"), exports);
__exportStar(require("./actions/resources/triggers/resources/bindings/client/requests/index.js"), exports);
__exportStar(require("./attackProtection/resources/breachedPasswordDetection/client/requests/index.js"), exports);
__exportStar(require("./attackProtection/resources/bruteForceProtection/client/requests/index.js"), exports);
__exportStar(require("./attackProtection/resources/suspiciousIpThrottling/client/requests/index.js"), exports);
__exportStar(require("./branding/resources/themes/client/requests/index.js"), exports);
__exportStar(require("./branding/resources/phone/resources/providers/client/requests/index.js"), exports);
__exportStar(require("./branding/resources/phone/resources/templates/client/requests/index.js"), exports);
__exportStar(require("./clientGrants/resources/organizations/client/requests/index.js"), exports);
__exportStar(require("./clients/resources/credentials/client/requests/index.js"), exports);
__exportStar(require("./clients/resources/connections/client/requests/index.js"), exports);
__exportStar(require("./connections/resources/clients/client/requests/index.js"), exports);
__exportStar(require("./connections/resources/scimConfiguration/client/requests/index.js"), exports);
__exportStar(require("./connections/resources/users/client/requests/index.js"), exports);
__exportStar(require("./connections/resources/scimConfiguration/resources/tokens/client/requests/index.js"), exports);
__exportStar(require("./emails/resources/provider/client/requests/index.js"), exports);
__exportStar(require("./eventStreams/resources/deliveries/client/requests/index.js"), exports);
__exportStar(require("./eventStreams/resources/redeliveries/client/requests/index.js"), exports);
__exportStar(require("./flows/resources/executions/client/requests/index.js"), exports);
__exportStar(require("./flows/resources/vault/resources/connections/client/requests/index.js"), exports);
__exportStar(require("./groups/resources/members/client/requests/index.js"), exports);
__exportStar(require("./guardian/resources/enrollments/client/requests/index.js"), exports);
__exportStar(require("./guardian/resources/factors/client/requests/index.js"), exports);
__exportStar(require("./guardian/resources/factors/resources/phone/client/requests/index.js"), exports);
__exportStar(require("./guardian/resources/factors/resources/pushNotification/client/requests/index.js"), exports);
__exportStar(require("./guardian/resources/factors/resources/sms/client/requests/index.js"), exports);
__exportStar(require("./guardian/resources/factors/resources/duo/resources/settings/client/requests/index.js"), exports);
__exportStar(require("./jobs/resources/usersExports/client/requests/index.js"), exports);
__exportStar(require("./jobs/resources/usersImports/client/requests/index.js"), exports);
__exportStar(require("./jobs/resources/verificationEmail/client/requests/index.js"), exports);
__exportStar(require("./keys/resources/customSigning/client/requests/index.js"), exports);
__exportStar(require("./keys/resources/encryption/client/requests/index.js"), exports);
__exportStar(require("./organizations/resources/clientGrants/client/requests/index.js"), exports);
__exportStar(require("./organizations/resources/enabledConnections/client/requests/index.js"), exports);
__exportStar(require("./organizations/resources/invitations/client/requests/index.js"), exports);
__exportStar(require("./organizations/resources/members/client/requests/index.js"), exports);
__exportStar(require("./organizations/resources/members/resources/roles/client/requests/index.js"), exports);
__exportStar(require("./prompts/resources/rendering/client/requests/index.js"), exports);
__exportStar(require("./riskAssessments/resources/settings/client/requests/index.js"), exports);
__exportStar(require("./riskAssessments/resources/settings/resources/newDevice/client/requests/index.js"), exports);
__exportStar(require("./roles/resources/permissions/client/requests/index.js"), exports);
__exportStar(require("./roles/resources/users/client/requests/index.js"), exports);
__exportStar(require("./selfServiceProfiles/resources/ssoTicket/client/requests/index.js"), exports);
__exportStar(require("./tenants/resources/settings/client/requests/index.js"), exports);
__exportStar(require("./users/resources/authenticationMethods/client/requests/index.js"), exports);
__exportStar(require("./users/resources/groups/client/requests/index.js"), exports);
__exportStar(require("./users/resources/identities/client/requests/index.js"), exports);
__exportStar(require("./users/resources/logs/client/requests/index.js"), exports);
__exportStar(require("./users/resources/organizations/client/requests/index.js"), exports);
__exportStar(require("./users/resources/permissions/client/requests/index.js"), exports);
__exportStar(require("./users/resources/riskAssessments/client/requests/index.js"), exports);
__exportStar(require("./users/resources/roles/client/requests/index.js"), exports);
__exportStar(require("./users/resources/refreshToken/client/requests/index.js"), exports);
__exportStar(require("./users/resources/sessions/client/requests/index.js"), exports);
__exportStar(require("./verifiableCredentials/resources/verification/resources/templates/client/requests/index.js"), exports);
