// auth0/dist/cjs/management/api/types/LogStreamFilterGroupNameEnum.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LogStreamFilterGroupNameEnum = void 0;
exports.LogStreamFilterGroupNameEnum = {
    AuthLoginFail: "auth.login.fail",
    AuthLoginNotification: "auth.login.notification",
    AuthLoginSuccess: "auth.login.success",
    AuthLogoutFail: "auth.logout.fail",
    AuthLogoutSuccess: "auth.logout.success",
    AuthSignupFail: "auth.signup.fail",
    AuthSignupSuccess: "auth.signup.success",
    AuthSilentAuthFail: "auth.silent_auth.fail",
    AuthSilentAuthSuccess: "auth.silent_auth.success",
    AuthTokenExchangeFail: "auth.token_exchange.fail",
    AuthTokenExchangeSuccess: "auth.token_exchange.success",
    ManagementFail: "management.fail",
    ManagementSuccess: "management.success",
    ScimEvent: "scim.event",
    SystemNotification: "system.notification",
    UserFail: "user.fail",
    UserNotification: "user.notification",
    UserSuccess: "user.success",
    Actions: "actions",
    Other: "other",
};
