// auth0/dist/esm/management/api/types/OauthScope.mjs
/**
 * This file was auto-generated by Fern from our API Definition.
 */
export const OauthScope = {
    ReadActions: "read:actions",
    CreateActions: "create:actions",
    DeleteActions: "delete:actions",
    UpdateActions: "update:actions",
    ReadAnomalyBlocks: "read:anomaly_blocks",
    DeleteAnomalyBlocks: "delete:anomaly_blocks",
    ReadShields: "read:shields",
    CreateShields: "create:shields",
    UpdateShields: "update:shields",
    ReadAttackProtection: "read:attack_protection",
    UpdateAttackProtection: "update:attack_protection",
    ReadBranding: "read:branding",
    UpdateBranding: "update:branding",
    ReadPhoneProviders: "read:phone_providers",
    CreatePhoneProviders: "create:phone_providers",
    UpdatePhoneProviders: "update:phone_providers",
    DeletePhoneProviders: "delete:phone_providers",
    ReadPhoneTemplates: "read:phone_templates",
    CreatePhoneTemplates: "create:phone_templates",
    UpdatePhoneTemplates: "update:phone_templates",
    DeletePhoneTemplates: "delete:phone_templates",
    DeleteBranding: "delete:branding",
    ReadClientGrants: "read:client_grants",
    CreateClientGrants: "create:client_grants",
    UpdateClientGrants: "update:client_grants",
    DeleteClientGrants: "delete:client_grants",
    ReadOrganizationClientGrants: "read:organization_client_grants",
    ReadClients: "read:clients",
    ReadClientKeys: "read:client_keys",
    ReadClientCredentials: "read:client_credentials",
    ReadClientSummary: "read:client_summary",
    CreateClients: "create:clients",
    CreateClientCredentials: "create:client_credentials",
    UpdateClientCredentials: "update:client_credentials",
    DeleteClientCredentials: "delete:client_credentials",
    UpdateClients: "update:clients",
    UpdateClientKeys: "update:client_keys",
    DeleteClients: "delete:clients",
    ReadConnections: "read:connections",
    ReadConnectionProfiles: "read:connection_profiles",
    CreateConnectionProfiles: "create:connection_profiles",
    UpdateConnectionProfiles: "update:connection_profiles",
    DeleteConnectionProfiles: "delete:connection_profiles",
    CreateConnections: "create:connections",
    UpdateConnections: "update:connections",
    DeleteConnections: "delete:connections",
    ReadDirectoryProvisionings: "read:directory_provisionings",
    CreateDirectoryProvisionings: "create:directory_provisionings",
    UpdateDirectoryProvisionings: "update:directory_provisionings",
    DeleteDirectoryProvisionings: "delete:directory_provisionings",
    ReadUsers: "read:users",
    ReadConnectionsKeys: "read:connections_keys",
    CreateConnectionsKeys: "create:connections_keys",
    UpdateConnectionsKeys: "update:connections_keys",
    ReadScimConfig: "read:scim_config",
    CreateScimConfig: "create:scim_config",
    UpdateScimConfig: "update:scim_config",
    DeleteScimConfig: "delete:scim_config",
    ReadScimToken: "read:scim_token",
    CreateScimToken: "create:scim_token",
    DeleteScimToken: "delete:scim_token",
    DeleteUsers: "delete:users",
    ReadCustomDomains: "read:custom_domains",
    CreateCustomDomains: "create:custom_domains",
    UpdateCustomDomains: "update:custom_domains",
    DeleteCustomDomains: "delete:custom_domains",
    ReadDeviceCredentials: "read:device_credentials",
    CreateCurrentUserDeviceCredentials: "create:current_user_device_credentials",
    DeleteDeviceCredentials: "delete:device_credentials",
    DeleteCurrentUserDeviceCredentials: "delete:current_user_device_credentials",
    UpdateDeviceCodes: "update:device_codes",
    ReadDeviceCodes: "read:device_codes",
    CreateTestEmailDispatch: "create:test_email_dispatch",
    CreateEmailTemplates: "create:email_templates",
    ReadEmailTemplates: "read:email_templates",
    UpdateEmailTemplates: "update:email_templates",
    ReadEmailProvider: "read:email_provider",
    CreateEmailProvider: "create:email_provider",
    UpdateEmailProvider: "update:email_provider",
    DeleteEmailProvider: "delete:email_provider",
    ReadEntitlements: "read:entitlements",
    ReadEventStreams: "read:event_streams",
    CreateEventStreams: "create:event_streams",
    UpdateEventStreams: "update:event_streams",
    DeleteEventStreams: "delete:event_streams",
    ReadEventDeliveries: "read:event_deliveries",
    UpdateEventDeliveries: "update:event_deliveries",
    ReadExtensions: "read:extensions",
    ReadFlows: "read:flows",
    CreateFlows: "create:flows",
    ReadFlowsVaultConnections: "read:flows_vault_connections",
    CreateFlowsVaultConnections: "create:flows_vault_connections",
    UpdateFlowsVaultConnections: "update:flows_vault_connections",
    DeleteFlowsVaultConnections: "delete:flows_vault_connections",
    ReadFlowsExecutions: "read:flows_executions",
    DeleteFlowsExecutions: "delete:flows_executions",
    UpdateFlows: "update:flows",
    DeleteFlows: "delete:flows",
    ReadForms: "read:forms",
    CreateForms: "create:forms",
    UpdateForms: "update:forms",
    DeleteForms: "delete:forms",
    ReadGrants: "read:grants",
    DeleteGrants: "delete:grants",
    ReadGroups: "read:groups",
    ReadGroupMembers: "read:group_members",
    CreateGuardianEnrollmentTickets: "create:guardian_enrollment_tickets",
    ReadGuardianEnrollments: "read:guardian_enrollments",
    DeleteGuardianEnrollments: "delete:guardian_enrollments",
    ReadGuardianFactors: "read:guardian_factors",
    UpdateGuardianFactors: "update:guardian_factors",
    ReadMfaPolicies: "read:mfa_policies",
    UpdateMfaPolicies: "update:mfa_policies",
    ReadHooks: "read:hooks",
    CreateHooks: "create:hooks",
    UpdateHooks: "update:hooks",
    DeleteHooks: "delete:hooks",
    ReadInsights: "read:insights",
    ReadStats: "read:stats",
    ReadIntegrations: "read:integrations",
    CreateIntegrations: "create:integrations",
    UpdateIntegrations: "update:integrations",
    DeleteIntegrations: "delete:integrations",
    CreateUsers: "create:users",
    UpdateUsers: "update:users",
    ReadCustomSigningKeys: "read:custom_signing_keys",
    CreateCustomSigningKeys: "create:custom_signing_keys",
    UpdateCustomSigningKeys: "update:custom_signing_keys",
    DeleteCustomSigningKeys: "delete:custom_signing_keys",
    ReadEncryptionKeys: "read:encryption_keys",
    CreateEncryptionKeys: "create:encryption_keys",
    UpdateEncryptionKeys: "update:encryption_keys",
    DeleteEncryptionKeys: "delete:encryption_keys",
    ReadSigningKeys: "read:signing_keys",
    CreateSigningKeys: "create:signing_keys",
    UpdateSigningKeys: "update:signing_keys",
    ReadLogStreams: "read:log_streams",
    CreateLogStreams: "create:log_streams",
    UpdateLogStreams: "update:log_streams",
    DeleteLogStreams: "delete:log_streams",
    ReadLogs: "read:logs",
    ReadLogsUsers: "read:logs_users",
    ReadTenantSettings: "read:tenant_settings",
    UpdateTenantSettings: "update:tenant_settings",
    ReadNetworkAcls: "read:network_acls",
    CreateNetworkAcls: "create:network_acls",
    UpdateNetworkAcls: "update:network_acls",
    DeleteNetworkAcls: "delete:network_acls",
    ReadOrganizations: "read:organizations",
    ReadOrganizationsSummary: "read:organizations_summary",
    CreateOrganizations: "create:organizations",
    CreateOrganizationConnections: "create:organization_connections",
    UpdateOrganizations: "update:organizations",
    DeleteOrganizations: "delete:organizations",
    CreateOrganizationClientGrants: "create:organization_client_grants",
    DeleteOrganizationClientGrants: "delete:organization_client_grants",
    ReadOrganizationConnections: "read:organization_connections",
    UpdateOrganizationConnections: "update:organization_connections",
    DeleteOrganizationConnections: "delete:organization_connections",
    ReadOrganizationDiscoveryDomains: "read:organization_discovery_domains",
    CreateOrganizationDiscoveryDomains: "create:organization_discovery_domains",
    UpdateOrganizationDiscoveryDomains: "update:organization_discovery_domains",
    DeleteOrganizationDiscoveryDomains: "delete:organization_discovery_domains",
    ReadOrganizationInvitations: "read:organization_invitations",
    CreateOrganizationInvitations: "create:organization_invitations",
    DeleteOrganizationInvitations: "delete:organization_invitations",
    ReadOrganizationMembers: "read:organization_members",
    CreateOrganizationMembers: "create:organization_members",
    DeleteOrganizationMembers: "delete:organization_members",
    ReadOrganizationMemberRoles: "read:organization_member_roles",
    CreateOrganizationMemberRoles: "create:organization_member_roles",
    DeleteOrganizationMemberRoles: "delete:organization_member_roles",
    ReadPrompts: "read:prompts",
    UpdatePrompts: "update:prompts",
    ReadResourceServers: "read:resource_servers",
    UpdateResourceServers: "update:resource_servers",
    ReadRefreshTokens: "read:refresh_tokens",
    DeleteRefreshTokens: "delete:refresh_tokens",
    CreateResourceServers: "create:resource_servers",
    DeleteResourceServers: "delete:resource_servers",
    ReadRoles: "read:roles",
    CreateRoles: "create:roles",
    UpdateRoles: "update:roles",
    DeleteRoles: "delete:roles",
    ReadRoleMembers: "read:role_members",
    CreateRoleMembers: "create:role_members",
    ReadRules: "read:rules",
    CreateRules: "create:rules",
    UpdateRules: "update:rules",
    ReadRulesConfigs: "read:rules_configs",
    UpdateRulesConfigs: "update:rules_configs",
    DeleteRulesConfigs: "delete:rules_configs",
    DeleteRules: "delete:rules",
    ReadSecurityMetrics: "read:security_metrics",
    ReadSelfServiceProfiles: "read:self_service_profiles",
    CreateSelfServiceProfiles: "create:self_service_profiles",
    UpdateSelfServiceProfiles: "update:self_service_profiles",
    DeleteSelfServiceProfiles: "delete:self_service_profiles",
    ReadSelfServiceProfileCustomTexts: "read:self_service_profile_custom_texts",
    UpdateSelfServiceProfileCustomTexts: "update:self_service_profile_custom_texts",
    CreateSsoAccessTickets: "create:sso_access_tickets",
    DeleteSsoAccessTickets: "delete:sso_access_tickets",
    ReadSessions: "read:sessions",
    DeleteSessions: "delete:sessions",
    DeleteTenants: "delete:tenants",
    RunChecks: "run:checks",
    ReadChecks: "read:checks",
    ReadTenantFeatureFlags: "read:tenant_feature_flags",
    ReadTenantInvitations: "read:tenant_invitations",
    CreateTenantInvitations: "create:tenant_invitations",
    UpdateTenantInvitations: "update:tenant_invitations",
    DeleteTenantInvitations: "delete:tenant_invitations",
    ReadTenantMembers: "read:tenant_members",
    UpdateTenantMembers: "update:tenant_members",
    DeleteTenantMembers: "delete:tenant_members",
    ReadOwners: "read:owners",
    DeleteOwners: "delete:owners",
    CreateUserTickets: "create:user_tickets",
    ReadTokenExchangeProfiles: "read:token_exchange_profiles",
    CreateTokenExchangeProfiles: "create:token_exchange_profiles",
    UpdateTokenExchangeProfiles: "update:token_exchange_profiles",
    DeleteTokenExchangeProfiles: "delete:token_exchange_profiles",
    ReadEntityCounts: "read:entity_counts",
    ReadUserAttributeProfiles: "read:user_attribute_profiles",
    CreateUserAttributeProfiles: "create:user_attribute_profiles",
    UpdateUserAttributeProfiles: "update:user_attribute_profiles",
    DeleteUserAttributeProfiles: "delete:user_attribute_profiles",
    ReadUserIdpTokens: "read:user_idp_tokens",
    ReadCurrentUser: "read:current_user",
    UpdateUsersAppMetadata: "update:users_app_metadata",
    UpdateCurrentUserMetadata: "update:current_user_metadata",
    DeleteCurrentUser: "delete:current_user",
    ReadUserApplicationPasswords: "read:user_application_passwords",
    CreateUserApplicationPasswords: "create:user_application_passwords",
    DeleteUserApplicationPasswords: "delete:user_application_passwords",
    ReadAuthenticationMethods: "read:authentication_methods",
    UpdateAuthenticationMethods: "update:authentication_methods",
    CreateAuthenticationMethods: "create:authentication_methods",
    DeleteAuthenticationMethods: "delete:authentication_methods",
    ReadFederatedConnectionsTokens: "read:federated_connections_tokens",
    DeleteFederatedConnectionsTokens: "delete:federated_connections_tokens",
    UpdateCurrentUserIdentities: "update:current_user_identities",
    DeleteRoleMembers: "delete:role_members",
    ReadVdcsTemplates: "read:vdcs_templates",
    CreateVdcsTemplates: "create:vdcs_templates",
    UpdateVdcsTemplates: "update:vdcs_templates",
    DeleteVdcsTemplates: "delete:vdcs_templates",
};
